---
title: "BIO00056I"
subtitle: "Directed study 10 (worksheet): Phylogenies & Molecular Clock"

author: Daniel Jeffares
date: last-modified
format:
    html:
        code-copy: true
        code-tools: true
        code-link: true
        toc: true
        number-sections: true
        toc-location: right
        toc-depth: 2
        highlight-style: github
    
    pdf:
      toc: true
      number-sections: true
      colorlinks: true    

editor: source
---

::: callout-important
# This document now contains answers to questions.
:::


# Learning objectives

This worksheet is designed to prepare you for exam questions on phylogenies and molecular clocks.Here we show one example for how we can use and interpret a phylogenetic trees.

# Introduction

## Pathogens and epidemiology (the study of the causes of disease)

There are many types of pathogenic organisms; viruses, bacteria and protozoans are common types. Protozoans are single-celled eukaryotes, such as *Plasmodium* species that cause malaria. Here we consider *Leishmania* and *Endotrypanum*, protozoan pathogens that cause skin lesions that do not heal.

## Relevant facts about these protozoan pathogens

Three facts about these protozoan pathogens are important here:

-   **Breeding**: They undergo meiosis and sexual recombination. ie: they breed sexually.

-   **Vectors**: They are almost always transmitted by insects. The insects feed on human and/or animal blood. They pick up the pathogen from the skin of an infected person or animal when they feed, and transfer the pathogen to another person or animal. Such insects are called the vector of the pathogen.

-   **Reservoirs**: Some pathogens and the insect vectors feed on animals as well as people. A large number of pathogens can be present in the animal population. An animal that contains pathogens is called a reservoir species. A pathogen can have more than one reservoir species.

# The exam style questions

## The backgound

In a remote island in the pacific there appears to be a new disease. Its symptoms are similar to the cutaneous leishmaniasis, but with larger lesions. It is spreading, and doctors suspect it is caused by the spread of a pathogen because it comes in clusters of cases within specific villages and households. You have received a grant to investigate the cause. Could this be a new protozoan?

What is already known:

-   On this island, only two protozoans were known to cause symptoms like cutaneous leishmaniasis; *Leishmania major* and *Endotrypanum colombiensis*. They are quite different pathogens, and they can be distinguished genetically.

-   There are only three potential reservoirs: rats, dogs and pigs

-   There are two sandflies species that could be the vector

-   Both *L. major* and *E. colombiensis* and the mystery protozoan can be isolated from patients, cultured in a laboratory, and DNA extracted.

## The questions

You have a team of clinicians and laboratory technicians that can do parasite isolation, culturing and DNA extractions. Explain how you would use laboratory work and phylogenetic analysis to answer the following questions. The laboratory work should be simple, as we focus here on the phylogenetic analysis.

-   Question 1. Is this a new protozoan or a more virulent strain of an already known protozoan? (8 marks)

-   Question 2. Is there an insect vector? If so, what is it? (8 marks)

# Model answers

## Question 1. Is this a new protozoan or a more virulent strain of an already known protozoan? Also, is there an alternative to simple phylogentic methods? (10 marks)

To determine this, we could obtain the seuwnce of a gene rrom this new new protozoan and build a phylogeny. This would tell us if it is closely related to *L. major* or *E. colombiensis*, or if it is a new protozoan (1 mark). We can start by isolating the pathogen from patients, culturing it and extracting DNA (1 mark). Next, we need to PCR amplify and sequence two genes that are known to distinguish *L. major* and *E. colombiensis* (1 marks). If we have PCR primers for these genes already, great. If not, we will need to design some (1 marks). Once we have the sequences, we can build a phylogeny including sequences from known *L. major* and *E. colombiensis* samples (2 marks). If the sequences from the mystery pathogen cluster with one of these known pathogens, it is likely a more virulent strain of that pathogen (2 marks). If it forms a separate cluster, it is likely a new protozoan (2 marks).


If our PCR primers did not work, it is more of a challenge, and we might need to resort to sequencing the whole genome of the mystery pathogen, which will be more expensive and time consuming (2 marks).

## Question 2. Is there an insect vector? If so, what is it? (10 marks)

If our PCR primers worked, we can follow the same strategy for detecting pathogen DNA above. We would want to focus on potential vectors that live near the places where people are infected (1 mark). We would collect samples of these potential vectors, extract DNA and use our PCR primers to test for the presence of the pathogen DNA (2 marks). It would be wise to collect many vector samples (tens to hundreds) to increase our chances of finding the pathogen (1 marks). Then, we would sequence the PCR products from any positive samples and align the sequences (2 marks). This time,  we can add the sequences we obtained from the patients (1 mark). If we find that the sequences from patients match those from the local sand flies, we have found the vector (2 marks). If not, we may need to test other potential vectors or consider that there may not be an insect vector involved (1 marks).


