<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.33">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Daniel Jeffares">
<meta name="dcterms.date" content="2025-12-03">

<title>BIO00056I</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-6fbc3b7d0ac83ab8a043646d6c6d94a5.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-6f9ef9b5a12f5b8ef14302403eafb8d0.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">BIO00056I</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">BIO00056I</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./workshop5.html"> 
<span class="menu-text">workshop 5</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./workshop6.html"> 
<span class="menu-text">workshop 6</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./workshop7.html"> 
<span class="menu-text">workshop 7</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./workshop9.html"> 
<span class="menu-text">workshop 9</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./worksheet10.html"> 
<span class="menu-text">worksheet 10</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./worksheet11.html"> 
<span class="menu-text">worksheet 11</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./worksheet12.html"> 
<span class="menu-text">worksheet 12</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul class="collapse">
  <li><a href="#learning-objectives" id="toc-learning-objectives" class="nav-link active" data-scroll-target="#learning-objectives"><span class="header-section-number">1</span> Learning objectives</a></li>
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction"><span class="header-section-number">2</span> Introduction</a>
  <ul class="collapse">
  <li><a href="#the-molecular-clock" id="toc-the-molecular-clock" class="nav-link" data-scroll-target="#the-molecular-clock"><span class="header-section-number">2.1</span> The molecular clock</a></li>
  <li><a href="#evolutionary-change-over-deep-time" id="toc-evolutionary-change-over-deep-time" class="nav-link" data-scroll-target="#evolutionary-change-over-deep-time"><span class="header-section-number">2.2</span> Evolutionary change over deep time</a></li>
  </ul></li>
  <li><a href="#exercises" id="toc-exercises" class="nav-link" data-scroll-target="#exercises"><span class="header-section-number">3</span> Exercises</a>
  <ul class="collapse">
  <li><a href="#the-phylogeny" id="toc-the-phylogeny" class="nav-link" data-scroll-target="#the-phylogeny"><span class="header-section-number">3.1</span> The phylogeny</a></li>
  <li><a href="#genetic-distances-shallow-and-deep-time" id="toc-genetic-distances-shallow-and-deep-time" class="nav-link" data-scroll-target="#genetic-distances-shallow-and-deep-time"><span class="header-section-number">3.2</span> Genetic distances: shallow and deep time</a></li>
  <li><a href="#shallow-time-ns-ratios" id="toc-shallow-time-ns-ratios" class="nav-link" data-scroll-target="#shallow-time-ns-ratios"><span class="header-section-number">3.3</span> Shallow time: N/S ratios</a></li>
  <li><a href="#shallow-time-what-kinds-of-genes-have-high-or-low-ns-ratios" id="toc-shallow-time-what-kinds-of-genes-have-high-or-low-ns-ratios" class="nav-link" data-scroll-target="#shallow-time-what-kinds-of-genes-have-high-or-low-ns-ratios"><span class="header-section-number">3.4</span> Shallow time: what kinds of genes have high or low N/S ratios?</a></li>
  <li><a href="#deep-time-analysis" id="toc-deep-time-analysis" class="nav-link" data-scroll-target="#deep-time-analysis"><span class="header-section-number">3.5</span> Deep time analysis</a></li>
  <li><a href="#deep-time-analysis-discussion-points" id="toc-deep-time-analysis-discussion-points" class="nav-link" data-scroll-target="#deep-time-analysis-discussion-points"><span class="header-section-number">3.6</span> Deep time analysis: discussion points</a></li>
  <li><a href="#very-deep-time-analysis" id="toc-very-deep-time-analysis" class="nav-link" data-scroll-target="#very-deep-time-analysis"><span class="header-section-number">3.7</span> Very deep time analysis</a></li>
  <li><a href="#final-discussion-points" id="toc-final-discussion-points" class="nav-link" data-scroll-target="#final-discussion-points"><span class="header-section-number">3.8</span> Final discussion points</a></li>
  </ul></li>
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary"><span class="header-section-number">4</span> Summary</a></li>
  <li><a href="#after-the-workshop-exam-style-questions" id="toc-after-the-workshop-exam-style-questions" class="nav-link" data-scroll-target="#after-the-workshop-exam-style-questions"><span class="header-section-number">5</span> After the workshop: exam style questions</a></li>
  </ul>
<div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="workshop7-no-answers.pdf"><i class="bi bi-file-pdf"></i>PDF</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">BIO00056I</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
<p class="subtitle lead">Workshop 7: Comparative Genomics</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Daniel Jeffares </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">December 3, 2025</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="learning-objectives" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Learning objectives</h1>
<ul>
<li>Learn more about interpreting comparative genomics data</li>
<li>Learn about gene conservation over deep time (billions of years)</li>
<li>To observe an example of the principle that <em>diversity within species gives rise to the divergence between species</em></li>
</ul>
</section>
<section id="introduction" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Introduction</h1>
<p>Today, we examine whether we can observe consistent patterns of evolutionary rates across time scales.</p>
<section id="the-molecular-clock" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="the-molecular-clock"><span class="header-section-number">2.1</span> The molecular clock</h2>
<p>In the last workshop, we saw how several species of <em>Leishmania</em> parasites that can be found in the Amazon rainforest and other areas of South America were genetically different from each other. These species were all relatively closely related.</p>
<p>In this workshop, we continue looking at <em>Leishmania</em> data, but the observations we make will apply to any group of species.</p>
<p>This time, we look at how more <strong>distantly-related</strong> <em>Leishmania</em> species. We know from the <strong>molecular clock</strong> concept, that if two species are dissimilar genetically, this is because mutations have accumulated over time. Figure 2 below shows that the molecular clock also holds over very long periods of time.</p>
<p><img src="images/compgen1.jpg" class="img-fluid"> <strong>Figure 1.</strong> Evidence that molecular clock causes divergence between species over both short and long periods of time. Panel a: We saw from the influenza virus workshop that mutations occurred within influenza viruses at a regular rate with time over merely a few decades. Panel b: This also occurs over long periods of time. For example, geologists know that the Pacific islands of Hawaii have been formed over millions of years, emerging one after the other over the last 5 million years. Evolutionary biologists know that both honeycreeper birds and <em>Drosophila</em> fruit flies have populated these islands during this time, and from the geological data, we know <em>how long ago</em> they diverged. If we look at genetic data from the <em>Yp1</em> gene in honeycreepers (Panel c) and the Cytb gene from fruit flies (Panel d), we observe a very consistent correlation between genetic distance (vertical axis) and time between island formation (horizontal axis). Just like the data you saw from the influenza virus, it looks very much like there is a constant rate of genetic divergence with time, this time over millions of years. Plots b-d are from <a href="https://www.nature.com/articles/nrg1020">Bromham and Penny, 2003</a>.</p>
</section>
<section id="evolutionary-change-over-deep-time" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="evolutionary-change-over-deep-time"><span class="header-section-number">2.2</span> Evolutionary change over deep time</h2>
<p>We told you in the lectures that:</p>
<ul>
<li>The accumulation of mutations within species generates the genetic differences between species</li>
<li>The different kinds of genes have different evolutionary rates, for example:
<ul>
<li>In a single-celled organism, proteins that are located on the outside of the cells tend to evolve relatively rapidly, to adapt to the external environment</li>
<li>Proteins that are located deep on the insides of the cells, performing fundamental processes (like DNA replication), tend to change slowly because they are so important</li>
</ul></li>
</ul>
<p><br>
<br>
</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Are there consistent patterns of evolution within and between species?
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>The accumulation of mutations within species generates the genetic differences between species</li>
<li>Therefore, we might expect that the genes that evolve <em>evolve rapidly within species</em> would also <em>evolve rapidly between species</em></li>
<li>And the genes that change slowly <em>within species</em> should also change slowly <em>between species</em>.</li>
</ul>
<p><strong>Today, we will examine these predictions</strong></p>
</div>
</div>
</section>
</section>
<section id="exercises" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Exercises</h1>
<section id="the-phylogeny" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="the-phylogeny"><span class="header-section-number">3.1</span> The phylogeny</h2>
<p>This time, compare within species gene diversity of <em>Leishmania donovani</em>, then between species gene divergence between <em>Leishmania donovani</em>, and <em>Leishmania braziliensis</em>, and finally between species gene divergence with yeast. Figure 2 below shows a phylogenetic tree of <em>Leishmania</em> and related species.</p>
<p><img src="images/compgen2.jpg" class="img-fluid"></p>
<p><strong>Figure 2</strong>. A phylogenetic tree of <em>Leishmania</em> and related species. Two of species that we looked at last time in the Viannia clade are included (blue brackets). Today, we will study genetic differences within the <em>L. donovani</em> species (red arrow). We will also look at genetic differences <em>between species</em>, by comparing <em>L. braziliensis</em> to <em>L. donovani</em>. We will also compare <em>L. donovani</em> proteins to a yeast species that diverged from <em>Leishmania</em> over a billion years ago (but yeast is not shown on this tree).</p>
</section>
<section id="genetic-distances-shallow-and-deep-time" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="genetic-distances-shallow-and-deep-time"><span class="header-section-number">3.2</span> Genetic distances: shallow and deep time</h2>
<p>Today, we will examine patterns of evolutionary change over three time scales, as summarised in Table 1 below. Our aim is to examine whether the same kinds genes tend to evolve rapidly or slowly over all these time scales.</p>
<table class="table-striped table-hover caption-top table">
<caption><strong>Time scales</strong></caption>
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th>Time scale</th>
<th>Years (approximate)</th>
<th>Data comparison</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Shallow time</td>
<td>thousands</td>
<td>genetic differences within <em>L. donovani</em></td>
</tr>
<tr class="even">
<td>Intermediate</td>
<td>millions</td>
<td>genetic differences between <em>L. donovani</em> and <em>L. braziliensis</em></td>
</tr>
<tr class="odd">
<td>Deep time</td>
<td>billions</td>
<td>protein differences between <em>L. donovani</em> and yeast</td>
</tr>
</tbody>
</table>
<p>Note: These times scales are <em>very approximate</em>, but we certainly know that the within-species differences are much more recent than the between-species differences, and that the yeast divergence is <em>much older</em> still.</p>
<section id="synonymous-and-non-synonymous-mutations" class="level3" data-number="3.2.1">
<h3 data-number="3.2.1" class="anchored" data-anchor-id="synonymous-and-non-synonymous-mutations"><span class="header-section-number">3.2.1</span> Synonymous and non-synonymous mutations</h3>
<p>The SNPs we downloaded were classified as either:</p>
<ul>
<li><strong>synonymous</strong>: mutations that <strong>do not</strong> change the protein’s amino acid sequence. Sometimes these are called silent mutations.</li>
<li><strong>non-synonymous</strong>: mutations that <strong>do</strong> change the protein sequence.</li>
</ul>
<p>We expect the <strong>synonymous</strong> mutations to be more or less neutral (ie: not subject to selection), because they don’t change the protein, so we can think of these as representing what occurs without selection, like a control.</p>
<p>The <strong>non-synonymous</strong> mutations that do change the protein, will often be subject to selection. Since most new mutations are harmful, we expect that many non-synonymous mutations will be removed by <strong>purifying selection</strong>. Very occasionally, a non-synonymous mutation will be beneficial, and will be favoured by <strong>positive selection</strong>.</p>
<p>To understand selection within a gene, we can calculate ratio of non-synonymous (selection) / synonymous (control) ratio. We will call this the <strong>N/S ratio</strong>. I show some toy examples of N/S ratios in Table 2 below.</p>
<p><br>
<em>NB: More complex analyses call this ratio the dN/dS ratio, or the Ka/Ks ratio.</em></p>
<p><br>
</p>
<table class="caption-top table">
<caption><strong>N/S ratio examples</strong></caption>
<colgroup>
<col style="width: 5%">
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 60%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;">Gene</th>
<th style="text-align: center;">synonymous SNPs</th>
<th style="text-align: center;">non-synonymous SNPs</th>
<th style="text-align: left;">Interpretation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">A</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">10</td>
<td style="text-align: left;">far fewer non-synonymous than synonymous SNPs: <strong>purifying selection</strong> has removed many non-synonymous SNPs</td>
</tr>
<tr class="even">
<td style="text-align: center;">B</td>
<td style="text-align: center;">10</td>
<td style="text-align: center;">11</td>
<td style="text-align: left;">non-synonymous and synonymous SNPs about the same: <strong>no selection</strong></td>
</tr>
<tr class="odd">
<td style="text-align: center;">C</td>
<td style="text-align: center;">20</td>
<td style="text-align: center;">10</td>
<td style="text-align: left;">more non-synonymous than synonymous SNPs: <strong>positive selection</strong> has selected for non-synonymous SNPs</td>
</tr>
</tbody>
</table>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Interpreting the N/S ratios
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li><p><strong>Gene A</strong>: The N/S ratio is very low (N=1, S=10, N/S = 0.1). Since we observe many fewer non-synonymous (amino-acid changing) SNPs than synonymous SNPs <strong>strong purifying selection</strong> has probably been acting on this gene.</p></li>
<li><p><strong>Gene B</strong>: The N/S ratio is near to 1 (N=10, S=11, N/S = 0.9). Since we observe the same number of non-synonymous (amino-acid changing) SNPs as synonymous SNPs it is likely that <strong>no selection</strong> has been acting on this gene.</p></li>
<li><p><strong>Gene C</strong>: The N/S ratio is high (N=20, S=10, N/S = 2). Since we observe more non-synonymous (amino-acid changing) SNPs than synonymous SNPs <strong>strong positive selection</strong> has probably been acting on this gene</p></li>
</ul>
</div>
</div>
<p>NB: These are toy examples. We would seldom observe such extreme N/S values in real data.</p>
</section>
</section>
<section id="shallow-time-ns-ratios" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="shallow-time-ns-ratios"><span class="header-section-number">3.3</span> Shallow time: N/S ratios</h2>
<p>First, let’s look at Figure 3, which shows the distribution of N/S ratios within <em>L. donovani</em> genes.</p>
<p><img src="images/compgen3.jpg" class="img-fluid"></p>
<p><strong>Figure 3. The distribution of N/S ratios within L. donovani genes</strong>. We can see that most genes have N/S ratios less than 1 (left of the red dashed line), indicating that purifying selection is acting on most genes.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Discussion points: N/S ratios in L. donovani genes
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Most genes have N/S ratios less than 1 (left of the red dashed line). What does this indicate?</li>
<li>A few genes have N/S ratios greater than 1 (right of the red dashed line). What does this indicate?</li>
<li>In total, there are <code>57886</code> synonymous SNPs. We can assume that these are not subject to strong selection. In contrast, there are <code>48460</code> non-synonymous SNPs. So we have 83% as many <strong>non-synonymous SNPs</strong> as <strong>synonymous SNPs</strong>. Where did the other 17% of non-synonymous SNPs go?</li>
</ul>
</div>
</div>
</section>
<section id="shallow-time-what-kinds-of-genes-have-high-or-low-ns-ratios" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="shallow-time-what-kinds-of-genes-have-high-or-low-ns-ratios"><span class="header-section-number">3.4</span> Shallow time: what kinds of genes have high or low N/S ratios?</h2>
<p>If a gene has a low N/S ratio, this indicates that it is subject to strong purifying selection. If a gene has a high N/S ratio, this indicates that it is subject to positive selection.</p>
<p>To understand what kinds of genes have high or low N/S ratios, I divided the genes into 2 groups:</p>
<ul>
<li><strong>High N/S ratio genes</strong>, with N/S ratio &gt; 1.3, that <em>might be</em> subject to positive selection</li>
<li><strong>Low N/S ratio genes</strong>: with N/S ratio &lt; 0.15, that are probably subject to purifying selection</li>
</ul>
<p>Note: I say <em>might be</em>, because N/S ratios can be noisy, and positive selection can be quite difficult to prove.</p>
<p>There are 1714, genes with N/S ratio &gt; 1.3. Here is a list of those with the highest N/S ratios - it doesn’t tell us much does it!</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code> [1] "vacuolar-type Ca2+-ATPase, putative (fragment)"                                        
 [2] "Acyl CoA binding protein, putative"                                                    
 [3] "NUDIX hydrolase dihydroneopterin triphosphate pyrophosphohydrolase/hydrolase, putative"
 [4] "RNA-binding protein, putative"                                                         
 [5] "c2 domain protein, putative"                                                           
 [6] "methionyl-tRNA formyltransferase, putative"                                            
 [7] "pyroglutamyl-peptidase I, putative"                                                    
 [8] "Ssl1-like, putative"                                                                   
 [9] "DnaJ domain containing protein, putative"                                              
[10] "serine/threonine phosphatase, putative"                                                </code></pre>
</div>
</div>
<p>To explain what kinds of genes have high N/S ratios, we can use a technique called <strong>gene ontology (GO) enrichment analysis</strong>.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
gene ontology (GO)
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Gene Ontology (GO) is a system for classifying genes according to their biochemical functions, the biological processes they are involved in, and the cellular locations that the proteins operate in (eg: nucleus, membrane, etc).</li>
<li>If we have a gene list, we can use GO enrichment analysis to look for patterns in the kinds of genes that are present in the list.</li>
</ul>
</div>
</div>
<section id="shallow-time-gene-enrichment-analysis" class="level3" data-number="3.4.1">
<h3 data-number="3.4.1" class="anchored" data-anchor-id="shallow-time-gene-enrichment-analysis"><span class="header-section-number">3.4.1</span> Shallow time: gene enrichment analysis</h3>
<p>I ran a gene enrichment analysis, looking for enrichment in the <strong>cellular locations</strong> of the <strong>high N/S ratio genes</strong>, that might be subject to positive selection. I also ran an enrichment analysis on the <strong>low N/S ratio genes</strong>, that are subject to strong purifying selection. The results are shown in Figure 4 (positive selection) and Figure 5 (purifying selection) below.</p>
<p><img src="workshop7-data/table2-location.png" class="img-fluid"> <strong>Figure 4. A word cloud showing the cellular locations enriched in positive selection genes.</strong> I selected genes with high N/S ratios (&gt; 1.3), and ran the gene enrichment analysis using <a href="https://tritrypdb.org/tritrypdb/app">TriTrypDB</a>. Larger words indicate greater enrichment, and the grey text intensity indicates the statistical significance of the enrichment (darker = more significant, paler = less significant).</p>
<p><img src="workshop7-data/table3-location.png" class="img-fluid"> <strong>Figure 5. A word cloud showing the cellular locations enriched in genes that are subject to strong purifying selection</strong>. I selected genes with low N/S ratios (&lt; 0.15), and ran the gene enrichment analysis using <a href="https://tritrypdb.org/tritrypdb/app">TriTrypDB</a>. Larger words indicate greater enrichment, and the grey text intensity indicates the statistical significance of the enrichment (darker = more significant, paler = less significant).</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Discussion points: gene enrichment analysis
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Where do positive selection proteins localise (Figure 4)?</li>
<li>Where do slowly evolving proteins localise (Figure 5)?</li>
<li>In the same way that the MK test example showed us that different sites <em>within proteins</em> are subject to different types of selection, what patterns do you observe here?</li>
</ul>
</div>
</div>
</section>
</section>
<section id="deep-time-analysis" class="level2" data-number="3.5">
<h2 data-number="3.5" class="anchored" data-anchor-id="deep-time-analysis"><span class="header-section-number">3.5</span> Deep time analysis</h2>
<p>Since diversity within species gives rise to divergence between species, we might expect that the genes that evolve rapidly within species would also evolve rapidly between species over long periods of time. We’ll examine this now.</p>
<p>To study this, we will now compare how similar the proteins are between <em>L. donovani</em> and <em>L. braziliensis</em>. You can see that these are quite distant on the phylogenetic tree (Figure 2). We don’t know exactly how long ago they diverged, but it will be much longer ago that the within-species differences we saw earlier, so perhaps 100,000 years or even millions of years ago.</p>
<p>To compare the proteins between these two species, I calculated the <em>percent identity</em> of each <em>L. donovani</em> and <em>L. braziliensis</em> ortholog (ie: what percent of amino acids in the alignment are the same). The percent identities range from 22% to 100%. What is remarkable here is that some proteins have changed an enormous amount over this time, while others have not changed at all, showing that <strong>over evolutionary time, genes change at different rates</strong>.</p>
<p>Of the ~8,000 orthologous proteins I selected the most rapidly evolving 1500 genes (lowest percent identity) and the most slowly evolving 1500 genes (highest percent identity), and ran gene enrichment analyses on both sets of genes.</p>
<section id="deep-time-analysis-first-look-at-the-results" class="level3" data-number="3.5.1">
<h3 data-number="3.5.1" class="anchored" data-anchor-id="deep-time-analysis-first-look-at-the-results"><span class="header-section-number">3.5.1</span> Deep time analysis: first look at the results</h3>
<p>Let’s look at the most slowly evolving genes first. Here is a list of the most conserved, slowly evolving genes (with the highest percent identity):</p>
<hr>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code> [1] "calmodulin, putative"                          
 [2] "calmodulin, putative"                          
 [3] "40S ribosomal protein S5 (fragment)"           
 [4] "40S ribosomal protein S15A, putative"          
 [5] "60S ribosomal protein L34, putative (fragment)"
 [6] "ATG8/AUT7/APG8/PAZ2, putative"                 
 [7] "60S ribosomal protein L37a, putative"          
 [8] "40S ribosomal protein S8, putative (fragment)" 
 [9] "40S ribosomal protein S15A, putative"          
[10] "60S ribosomal protein L13, putative (fragment)"</code></pre>
</div>
</div>
<hr>
<p>This time, there is <strong>certainly</strong> a pattern!<br>
<br>
</p>
<p>And now here’s the list of the most rapidly evolving genes (with the <em>lowest</em> percent identity):</p>
<hr>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Present in the outer mitochondrial membrane proteome 22-1"        
 [2] "receptor-type adenylate cyclase a (fragment)"                     
 [3] "glutathionylspermidine synthase, putative"                        
 [4] "receptor-type adenylate cyclase, putative (fragment)"             
 [5] "polyprenyl synthase, putative"                                    
 [6] "ubiquitin-activating enzyme, putative"                            
 [7] "helicase-like protein"                                            
 [8] "Wee1-like protein kinase, putative"                               
 [9] "mitochondrial chaperone, putative (fragment)"                     
[10] "RNA recognition motif. (a.k.a. RRM, RBD, or RNP domain), putative"</code></pre>
</div>
</div>
<hr>
<p>I am not such a ‘Leishmaniac’ that I can see a pattern here, but do you notice anything that reminds me of the shallow time results?</p>
</section>
<section id="deep-time-analysis-go-enrichment-analysis" class="level3" data-number="3.5.2">
<h3 data-number="3.5.2" class="anchored" data-anchor-id="deep-time-analysis-go-enrichment-analysis"><span class="header-section-number">3.5.2</span> Deep time analysis: GO enrichment analysis</h3>
<p>To understand these large lists of genes, I again ran gene enrichment analyses on both the most slowly evolving genes (highest percent identity) and the most rapidly evolving genes (lowest percent identity). The results are shown in Figure 6 (slowly evolving genes) and Figure 7 (rapidly evolving genes) below.</p>
<p><img src="images/table5-wordcloud.png" class="img-fluid"></p>
<p><strong>Figure 6. A word cloud showing the cellular locations enriched in slowly evolving genes.</strong> I selected genes with high percent identity (most slowly evolving), and ran the gene enrichment analysis using <a href="https://tritrypdb.org/tritrypdb/app">TriTrypDB</a>. Larger words indicate greater enrichment, and the grey text intensity indicates the statistical significance of the enrichment (darker = more significant, paler = less significant).</p>
<p><img src="images/table4-wordcloud.png" class="img-fluid"> <strong>Figure 7. A word cloud showing the cellular locations enriched in rapidly evolving genes.</strong> I selected genes with low percent identity (most rapidly evolving), and ran the gene enrichment analysis using <a href="https://tritrypdb.org/tritrypdb/app">TriTrypDB</a>. Larger words indicate greater enrichment, and the grey text intensity indicates the statistical significance of the enrichment (darker = more significant, paler = less significant).</p>
</section>
</section>
<section id="deep-time-analysis-discussion-points" class="level2" data-number="3.6">
<h2 data-number="3.6" class="anchored" data-anchor-id="deep-time-analysis-discussion-points"><span class="header-section-number">3.6</span> Deep time analysis: discussion points</h2>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Discussion points: deep time gene enrichment analysis
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Consider what kinds of genes are slowly evolving (Figure 6) and rapidly evolving (Figure 7).</li>
<li>Is this consistent with what we saw in the shallow time analysis?</li>
<li>We used slightly different methods (N/S ratios vs percent identity), and different species comparisons (<em>L. donovani</em> within-species vs <em>L. donovani</em> vs <em>L. braziliensis</em> between-species).</li>
<li>Yet we see similar patterns. What does this tell us about the processes of selection that operate within species (<em>L. donovani</em>) and between species?</li>
<li>Are rates of change higher for proteins the insides of cells, or the outsides of cells? Why might this be?</li>
</ul>
</div>
</div>
</section>
<section id="very-deep-time-analysis" class="level2" data-number="3.7">
<h2 data-number="3.7" class="anchored" data-anchor-id="very-deep-time-analysis"><span class="header-section-number">3.7</span> Very deep time analysis</h2>
<p>So far we have looked at genetic differences within <em>L. donovani</em> (shallow time) and between <em>L. donovani</em> and <em>L. braziliensis</em> (deep time). Now we will compare Leishmania proteins to the fission yeast <em>Schizosaccharomyces pombe</em>, which diverged from Leishmania a very long time ago, perhaps a billion years ago.</p>
<p>Again, I calculated the percent identity of each <em>L. donovani</em> and <em>S. pombe</em> ortholog (ie: what percent of amino acids in the alignment are the same).</p>
<p>This time, we will only look at the conserved, slowly evolving proteins, because the rapidly evolving proteins have changed so much that we cannot identify orthologs between these very distantly-related species. The top 150 conserved proteins (with the highest percent identity) are between 58% and 83% identical between <em>L. donovani</em> and <em>S. pombe</em>.</p>
<p><img src="images/table6-wordcloud.png" class="img-fluid"> <strong>Figure 8. A word cloud showing the cellular locations enriched in very slowly evolving genes.</strong> I selected genes with high percent identity (most slowly evolving), and ran the gene enrichment analysis using <a href="https://tritrypdb.org/tritrypdb/app">TriTrypDB</a>. Larger words indicate greater enrichment, and the grey text intensity indicates the statistical significance of the enrichment (darker = more significant, paler = less significant).</p>
<p>Here there is one very clear outline: <strong>the ribosome</strong>.</p>
</section>
<section id="final-discussion-points" class="level2" data-number="3.8">
<h2 data-number="3.8" class="anchored" data-anchor-id="final-discussion-points"><span class="header-section-number">3.8</span> Final discussion points</h2>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Discussion points
</div>
</div>
<div class="callout-body-container callout-body">
<p>We observe a fairly consistent pattern across all time scales that we have examined: genes that are located on the outsides of cells tend to evolve rapidly, while genes that are located deep within cells tend to evolve slowly. of course, there are exceptions, and the rates of protein change are affected by many other factors, but this general pattern is very clear. Let’s consider this.</p>
<ul>
<li>Why do you think that ribosomal proteins are so highly conserved over such long periods of time (Figure 8)?</li>
<li>Given the function of ribosomes in the cell, what might be the consequences of a mutation that reduced protein translation accuracy by 0.1%?</li>
<li>Why do you think that proteins located on the outsides of cells tend to evolve rapidly?</li>
</ul>
</div>
</div>
<p>For those that are curious <a href="https://docs.google.com/spreadsheets/d/1KmtZizHfnWdP1V2PJQZXV5G0p_NLEIM2vR8yGiTule0/edit?usp=sharing">the genes lists that I used for this analysis are here</a>. You’ll need to be signed in with your University of York Google account to access it.</p>
</section>
</section>
<section id="summary" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Summary</h1>
<p>Today, we examined the patterns of evolutionary change within and between species. We observed that:</p>
<ul>
<li>Different kinds of genes evolve at different rates, both within species and between species</li>
<li>These patterns are consistent over both short time scales (within species) and long time scales</li>
<li>Genes that are located on the outsides of cells tend to evolve rapidly, while genes that are located deep within cells tend to evolve slowly</li>
</ul>
<p><strong>That is the end for today.</strong></p>
</section>
<section id="after-the-workshop-exam-style-questions" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> After the workshop: exam style questions</h1>
<p><strong>Question 1. If the rate of mutation is the same for all genes in a genome, and we compare two very closely related species, we should observe the same numbers of mutations in each gene. Is this correct?</strong></p>
<p>In your answer, explain:</p>
<ul>
<li>If we <em>do</em> see the same numbers of mutations in each gene, or not</li>
<li>What genetic and evolutionary factors might cause us to <em>not</em> observe the same numbers of mutations in each gene</li>
</ul>
<p>Hint: consider both stochastic factors (randomness) and biological processes of evolutionary change.</p>
<p><strong>Question 2. Given what you know about the fitness effects of mutations, which population genetic process would we expect to have the main impact on differences in evolutionary rates between genes?</strong></p>
<p>In your answer, explain:</p>
<ul>
<li>What the main population genetic process is</li>
<li>Why this results in differences in evolutionary rates between genes</li>
<li>How we could test this hypothesis using genetic data from within and between species</li>
</ul>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb4" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "BIO00056I"</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Workshop 7: Comparative Genomics"</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> Daniel Jeffares</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> last-modified</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co">    html:</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co">      </span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co">        code-copy: true</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="co">        code-tools: true</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co">        code-link: true</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co">        toc: true</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="co">        number-sections: true</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="co">        toc-location: right</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="co">        toc-depth: 2</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="co">        highlight-style: github</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a><span class="co">            </span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="co">    pdf:</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a><span class="co">      toc: true</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="co">      number-sections: true</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a><span class="co">      colorlinks: true</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> source</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a><span class="co">    </span></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a><span class="fu"># Learning objectives</span></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Learn more about interpreting comparative genomics data</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Learn about gene conservation over deep time (billions of years)</span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>To observe an example of the principle that *diversity within species gives rise to the divergence between species*</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a><span class="fu"># Introduction</span></span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a>Today, we examine whether we can observe consistent patterns of evolutionary rates across time scales.</span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a><span class="fu">## The molecular clock</span></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a>In the last workshop, we saw how several species of *Leishmania* parasites that can be found in the Amazon rainforest and other areas of South America were genetically different from each other. These species were all relatively closely related.</span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a>In this workshop, we continue looking at *Leishmania* data, but the observations we make will apply to any group of species.</span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a>This time, we look at how more **distantly-related** *Leishmania* species. We know from the **molecular clock** concept, that if two species are dissimilar genetically, this is because mutations have accumulated over time. Figure 2 below shows that the molecular clock also holds over very long periods of time. </span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/compgen1.jpg)</span></span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a>**Figure 1.** Evidence that molecular clock causes divergence between species over both short and long periods of time.  Panel a: We saw from the influenza virus workshop that mutations occurred within influenza viruses at a regular rate with time over merely a few decades. Panel b: This also occurs over long periods of time. For example, geologists know that the Pacific islands of Hawaii have been formed over millions of years, emerging one after the other over the last 5 million years. Evolutionary biologists know that both honeycreeper birds and *Drosophila* fruit flies have populated these islands during this time, and from the geological data, we know *how long ago* they diverged. If we look at genetic data from the *Yp1* gene in honeycreepers (Panel c) and the Cytb gene from fruit flies (Panel d), we observe a very consistent correlation between genetic distance (vertical axis) and time between island formation (horizontal axis). Just like the data you saw from the influenza virus, it looks very much like there is a constant rate of genetic divergence with time, this time over millions of years. Plots b-d are from <span class="co">[</span><span class="ot">Bromham and Penny, 2003</span><span class="co">](https://www.nature.com/articles/nrg1020)</span>.</span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a><span class="fu">## Evolutionary change over deep time </span></span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a>We told you in the lectures that:</span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>The accumulation of mutations within species generates the genetic differences between species</span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>The   different kinds of genes have different evolutionary rates, for example:</span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>In a single-celled organism, proteins that are located on the outside of the cells tend to  evolve relatively rapidly, to adapt to the external environment</span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>Proteins that are located deep on the insides of the cells, performing fundamental processes (like DNA replication), tend to change slowly because they are so important</span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a>\</span>
<span id="cb4-63"><a href="#cb4-63" aria-hidden="true" tabindex="-1"></a>\</span>
<span id="cb4-64"><a href="#cb4-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-65"><a href="#cb4-65" aria-hidden="true" tabindex="-1"></a>:::callout-note</span>
<span id="cb4-66"><a href="#cb4-66" aria-hidden="true" tabindex="-1"></a><span class="fu"># Are there consistent patterns of evolution within and between species?</span></span>
<span id="cb4-67"><a href="#cb4-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-68"><a href="#cb4-68" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>The accumulation of mutations within species generates the genetic differences between species </span>
<span id="cb4-69"><a href="#cb4-69" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Therefore, we might expect that the genes that evolve *evolve rapidly within species* would also *evolve rapidly between species*</span>
<span id="cb4-70"><a href="#cb4-70" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>And the genes that change slowly *within species* should also change slowly *between species*.</span>
<span id="cb4-71"><a href="#cb4-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-72"><a href="#cb4-72" aria-hidden="true" tabindex="-1"></a>**Today, we will examine these predictions**</span>
<span id="cb4-73"><a href="#cb4-73" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb4-74"><a href="#cb4-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-75"><a href="#cb4-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-76"><a href="#cb4-76" aria-hidden="true" tabindex="-1"></a><span class="fu"># Exercises</span></span>
<span id="cb4-77"><a href="#cb4-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-78"><a href="#cb4-78" aria-hidden="true" tabindex="-1"></a><span class="fu">## The phylogeny</span></span>
<span id="cb4-79"><a href="#cb4-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-80"><a href="#cb4-80" aria-hidden="true" tabindex="-1"></a>This time, compare within species gene diversity of *Leishmania donovani*, then between species gene divergence between *Leishmania donovani*, and *Leishmania braziliensis*, and finally  between species gene divergence with yeast. Figure 2 below shows a phylogenetic tree of *Leishmania* and related species.</span>
<span id="cb4-81"><a href="#cb4-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-82"><a href="#cb4-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-83"><a href="#cb4-83" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/compgen2.jpg)</span></span>
<span id="cb4-84"><a href="#cb4-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-85"><a href="#cb4-85" aria-hidden="true" tabindex="-1"></a>**Figure 2**. A phylogenetic tree of *Leishmania* and related species. Two of species that we looked at last time in the Viannia clade are included (blue brackets). Today, we will study genetic differences within the *L. donovani* species (red arrow). We will also look at genetic differences *between species*, by comparing *L. braziliensis* to  *L. donovani*. We will also compare *L. donovani* proteins to a yeast species that diverged from *Leishmania* over a billion years ago (but yeast is not shown on this tree).</span>
<span id="cb4-86"><a href="#cb4-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-87"><a href="#cb4-87" aria-hidden="true" tabindex="-1"></a><span class="fu">## Genetic distances: shallow and deep time</span></span>
<span id="cb4-88"><a href="#cb4-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-89"><a href="#cb4-89" aria-hidden="true" tabindex="-1"></a>Today, we will examine patterns of evolutionary change over three time scales, as summarised in Table 1 below. Our aim is to examine whether the same kinds genes tend to evolve rapidly or slowly over all these time scales.</span>
<span id="cb4-90"><a href="#cb4-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-91"><a href="#cb4-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-92"><a href="#cb4-92" aria-hidden="true" tabindex="-1"></a>| Time scale  | Years (approximate)  | Data comparison |</span>
<span id="cb4-93"><a href="#cb4-93" aria-hidden="true" tabindex="-1"></a>|--------|--------|--------|</span>
<span id="cb4-94"><a href="#cb4-94" aria-hidden="true" tabindex="-1"></a>| Shallow time  | thousands | genetic differences within *L. donovani* |</span>
<span id="cb4-95"><a href="#cb4-95" aria-hidden="true" tabindex="-1"></a>| Intermediate   | millions  |  genetic differences between *L. donovani* and *L. braziliensis*     |</span>
<span id="cb4-96"><a href="#cb4-96" aria-hidden="true" tabindex="-1"></a>| Deep time | billions  | protein differences between *L. donovani* and yeast   |</span>
<span id="cb4-97"><a href="#cb4-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-98"><a href="#cb4-98" aria-hidden="true" tabindex="-1"></a>: **Time scales** {.striped .hover}</span>
<span id="cb4-99"><a href="#cb4-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-100"><a href="#cb4-100" aria-hidden="true" tabindex="-1"></a>Note: These times scales are *very approximate*, but we certainly know that the within-species differences are much more recent than the between-species differences, and that the yeast divergence is *much older* still.</span>
<span id="cb4-101"><a href="#cb4-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-102"><a href="#cb4-102" aria-hidden="true" tabindex="-1"></a><span class="fu">### Synonymous and non-synonymous mutations</span></span>
<span id="cb4-103"><a href="#cb4-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-104"><a href="#cb4-104" aria-hidden="true" tabindex="-1"></a>The SNPs we downloaded were classified as either:</span>
<span id="cb4-105"><a href="#cb4-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-106"><a href="#cb4-106" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**synonymous**: mutations that **do not** change the protein’s amino acid sequence. Sometimes these are called silent mutations. </span>
<span id="cb4-107"><a href="#cb4-107" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**non-synonymous**: mutations that **do** change the protein sequence.</span>
<span id="cb4-108"><a href="#cb4-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-109"><a href="#cb4-109" aria-hidden="true" tabindex="-1"></a>We expect the **synonymous** mutations to be more or less neutral (ie: not subject to selection), because they don't change the protein, so we can think of these as representing what occurs without selection, like a control. </span>
<span id="cb4-110"><a href="#cb4-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-111"><a href="#cb4-111" aria-hidden="true" tabindex="-1"></a>The **non-synonymous** mutations that do change the protein, will often be subject to selection. Since most new mutations are harmful, we expect that many non-synonymous mutations will be removed by **purifying selection**. Very occasionally, a non-synonymous mutation will be beneficial, and will be favoured by **positive selection**.</span>
<span id="cb4-112"><a href="#cb4-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-113"><a href="#cb4-113" aria-hidden="true" tabindex="-1"></a>To understand selection within a gene, we can calculate ratio of non-synonymous (selection) / synonymous (control) ratio. We will call this the **N/S ratio**. I show some toy examples of N/S ratios in Table 2 below.</span>
<span id="cb4-114"><a href="#cb4-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-115"><a href="#cb4-115" aria-hidden="true" tabindex="-1"></a>\</span>
<span id="cb4-116"><a href="#cb4-116" aria-hidden="true" tabindex="-1"></a>*NB: More complex analyses call this ratio the dN/dS ratio, or the Ka/Ks ratio.*</span>
<span id="cb4-117"><a href="#cb4-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-118"><a href="#cb4-118" aria-hidden="true" tabindex="-1"></a>\</span>
<span id="cb4-119"><a href="#cb4-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-120"><a href="#cb4-120" aria-hidden="true" tabindex="-1"></a>| Gene | synonymous SNPs | non-synonymous SNPs | Interpretation |</span>
<span id="cb4-121"><a href="#cb4-121" aria-hidden="true" tabindex="-1"></a>|:------:|:------:|:------:|:-------|</span>
<span id="cb4-122"><a href="#cb4-122" aria-hidden="true" tabindex="-1"></a>| A | 1 | 10 | far fewer non-synonymous than synonymous SNPs: **purifying selection** has removed many non-synonymous SNPs |</span>
<span id="cb4-123"><a href="#cb4-123" aria-hidden="true" tabindex="-1"></a>| B | 10 | 11 | non-synonymous and synonymous SNPs about the same: **no selection**  |</span>
<span id="cb4-124"><a href="#cb4-124" aria-hidden="true" tabindex="-1"></a>| C | 20 | 10 | more non-synonymous than synonymous SNPs: **positive selection** has selected for non-synonymous SNPs |</span>
<span id="cb4-125"><a href="#cb4-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-126"><a href="#cb4-126" aria-hidden="true" tabindex="-1"></a>: **N/S ratio examples** {.striped .hover}{tbl-colwidths="<span class="co">[</span><span class="ot">5,15,15,60</span><span class="co">]</span>"}</span>
<span id="cb4-127"><a href="#cb4-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-128"><a href="#cb4-128" aria-hidden="true" tabindex="-1"></a>:::callout-note</span>
<span id="cb4-129"><a href="#cb4-129" aria-hidden="true" tabindex="-1"></a><span class="fu"># Interpreting the N/S ratios</span></span>
<span id="cb4-130"><a href="#cb4-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-131"><a href="#cb4-131" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**Gene A**: The N/S ratio is very low (N=1, S=10, N/S = 0.1). Since we observe many fewer non-synonymous (amino-acid changing) SNPs than synonymous SNPs **strong purifying selection** has probably been acting on this gene.</span>
<span id="cb4-132"><a href="#cb4-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-133"><a href="#cb4-133" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**Gene B**: The N/S ratio is near to 1 (N=10, S=11, N/S = 0.9). Since we observe the same number of non-synonymous (amino-acid changing) SNPs as synonymous SNPs it is likely that **no selection** has been acting on this gene.</span>
<span id="cb4-134"><a href="#cb4-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-135"><a href="#cb4-135" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**Gene C**: The N/S ratio is high (N=20, S=10, N/S = 2). Since we observe more non-synonymous (amino-acid changing) SNPs than synonymous SNPs **strong positive selection** has probably been acting on this gene</span>
<span id="cb4-136"><a href="#cb4-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-137"><a href="#cb4-137" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb4-138"><a href="#cb4-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-139"><a href="#cb4-139" aria-hidden="true" tabindex="-1"></a>NB: These are toy examples. We would seldom observe such extreme N/S values in real data.</span>
<span id="cb4-140"><a href="#cb4-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-141"><a href="#cb4-141" aria-hidden="true" tabindex="-1"></a><span class="fu">## Shallow time: N/S ratios</span></span>
<span id="cb4-142"><a href="#cb4-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-143"><a href="#cb4-143" aria-hidden="true" tabindex="-1"></a>First, let's look at Figure 3, which shows the distribution of N/S ratios within *L. donovani* genes.</span>
<span id="cb4-144"><a href="#cb4-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-145"><a href="#cb4-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-148"><a href="#cb4-148" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-149"><a href="#cb4-149" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb4-150"><a href="#cb4-150" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb4-151"><a href="#cb4-151" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb4-152"><a href="#cb4-152" aria-hidden="true" tabindex="-1"></a><span class="co">#| output: false</span></span>
<span id="cb4-153"><a href="#cb4-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-154"><a href="#cb4-154" aria-hidden="true" tabindex="-1"></a><span class="co">#load libraries</span></span>
<span id="cb4-155"><a href="#cb4-155" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb4-156"><a href="#cb4-156" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb4-157"><a href="#cb4-157" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(janitor)</span>
<span id="cb4-158"><a href="#cb4-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-159"><a href="#cb4-159" aria-hidden="true" tabindex="-1"></a><span class="co">#set wd</span></span>
<span id="cb4-160"><a href="#cb4-160" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/Users/dj757/gd/modules/BIO56I"</span>)</span>
<span id="cb4-161"><a href="#cb4-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-162"><a href="#cb4-162" aria-hidden="true" tabindex="-1"></a><span class="co">#load data</span></span>
<span id="cb4-163"><a href="#cb4-163" aria-hidden="true" tabindex="-1"></a>table1 <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">"workshop7-data/BIO00056I-workshop7.xlsx"</span>, <span class="at">sheet =</span> <span class="dv">1</span>, <span class="at">skip=</span><span class="dv">6</span>)</span>
<span id="cb4-164"><a href="#cb4-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-165"><a href="#cb4-165" aria-hidden="true" tabindex="-1"></a><span class="co">#clean data</span></span>
<span id="cb4-166"><a href="#cb4-166" aria-hidden="true" tabindex="-1"></a>table1 <span class="ot">&lt;-</span> table1 <span class="sc">%&gt;%</span></span>
<span id="cb4-167"><a href="#cb4-167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">clean_names</span>()</span>
<span id="cb4-168"><a href="#cb4-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-169"><a href="#cb4-169" aria-hidden="true" tabindex="-1"></a><span class="co">#histgram of table1$n_s_ratio</span></span>
<span id="cb4-170"><a href="#cb4-170" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(table1, <span class="fu">aes</span>(<span class="at">x=</span>n_s_ratio)) <span class="sc">+</span></span>
<span id="cb4-171"><a href="#cb4-171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb4-172"><a href="#cb4-172" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"N/S ratio within L. donovani"</span>) <span class="sc">+</span></span>
<span id="cb4-173"><a href="#cb4-173" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Number of genes"</span>) <span class="sc">+</span></span>
<span id="cb4-174"><a href="#cb4-174" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">""</span>)<span class="sc">+</span></span>
<span id="cb4-175"><a href="#cb4-175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">1</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"red"</span>)</span>
<span id="cb4-176"><a href="#cb4-176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-177"><a href="#cb4-177" aria-hidden="true" tabindex="-1"></a><span class="co">#save plot</span></span>
<span id="cb4-178"><a href="#cb4-178" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"images/compgen3.jpg"</span>, <span class="at">width=</span><span class="dv">6</span>, <span class="at">height=</span><span class="dv">4</span>)</span>
<span id="cb4-179"><a href="#cb4-179" aria-hidden="true" tabindex="-1"></a>       </span>
<span id="cb4-180"><a href="#cb4-180" aria-hidden="true" tabindex="-1"></a><span class="co">#totals</span></span>
<span id="cb4-181"><a href="#cb4-181" aria-hidden="true" tabindex="-1"></a>nonsyn_total <span class="ot">&lt;-</span> <span class="fu">sum</span>(table1<span class="sc">$</span>nonsynonymous_snps)</span>
<span id="cb4-182"><a href="#cb4-182" aria-hidden="true" tabindex="-1"></a><span class="co">#48460</span></span>
<span id="cb4-183"><a href="#cb4-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-184"><a href="#cb4-184" aria-hidden="true" tabindex="-1"></a>syn_total <span class="ot">&lt;-</span> <span class="fu">sum</span>(table1<span class="sc">$</span>synonymous_snps)</span>
<span id="cb4-185"><a href="#cb4-185" aria-hidden="true" tabindex="-1"></a><span class="co">#57886</span></span>
<span id="cb4-186"><a href="#cb4-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-187"><a href="#cb4-187" aria-hidden="true" tabindex="-1"></a>total_ratio <span class="ot">&lt;-</span> <span class="fu">sum</span>(nonsyn_total <span class="sc">/</span> syn_total)</span>
<span id="cb4-188"><a href="#cb4-188" aria-hidden="true" tabindex="-1"></a><span class="co">#0.8376</span></span>
<span id="cb4-189"><a href="#cb4-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-190"><a href="#cb4-190" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-191"><a href="#cb4-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-192"><a href="#cb4-192" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/compgen3.jpg)</span></span>
<span id="cb4-193"><a href="#cb4-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-194"><a href="#cb4-194" aria-hidden="true" tabindex="-1"></a>**Figure 3. The distribution of N/S ratios within L. donovani genes**. We can see that most genes have N/S ratios less than 1 (left of the red dashed line), indicating that purifying selection is acting on most genes.</span>
<span id="cb4-195"><a href="#cb4-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-196"><a href="#cb4-196" aria-hidden="true" tabindex="-1"></a>:::callout-tip</span>
<span id="cb4-197"><a href="#cb4-197" aria-hidden="true" tabindex="-1"></a><span class="fu"># Discussion points: N/S ratios in L. donovani genes</span></span>
<span id="cb4-198"><a href="#cb4-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-199"><a href="#cb4-199" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Most genes have N/S ratios less than 1 (left of the red dashed line). What does this indicate?</span>
<span id="cb4-200"><a href="#cb4-200" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>A few genes have N/S ratios greater than 1 (right of the red dashed line). What does this indicate?</span>
<span id="cb4-201"><a href="#cb4-201" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>In total, there are <span class="in">`57886`</span> synonymous SNPs. We can assume that these are not subject to strong selection. In contrast, there are <span class="in">`48460`</span> non-synonymous SNPs. So we have 83% as many **non-synonymous SNPs** as **synonymous SNPs**. Where did the other 17% of non-synonymous SNPs go? </span>
<span id="cb4-202"><a href="#cb4-202" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb4-203"><a href="#cb4-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-204"><a href="#cb4-204" aria-hidden="true" tabindex="-1"></a><span class="fu">## Shallow time: what kinds of genes have high or low N/S ratios?</span></span>
<span id="cb4-205"><a href="#cb4-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-206"><a href="#cb4-206" aria-hidden="true" tabindex="-1"></a>If a gene has a low N/S ratio, this indicates that it is subject to strong purifying selection. If a gene has a high N/S ratio, this indicates that it is subject to positive selection.</span>
<span id="cb4-207"><a href="#cb4-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-208"><a href="#cb4-208" aria-hidden="true" tabindex="-1"></a>To understand what kinds of genes have high or low N/S ratios, I divided the genes into 2 groups:</span>
<span id="cb4-209"><a href="#cb4-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-210"><a href="#cb4-210" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**High N/S ratio genes**, with N/S ratio &gt; 1.3, that *might be* subject to positive selection</span>
<span id="cb4-211"><a href="#cb4-211" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**Low N/S ratio genes**: with N/S ratio &lt; 0.15, that are probably subject to purifying selection</span>
<span id="cb4-212"><a href="#cb4-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-213"><a href="#cb4-213" aria-hidden="true" tabindex="-1"></a>Note: I say *might be*, because N/S ratios can be noisy, and positive selection can be quite difficult to prove. </span>
<span id="cb4-214"><a href="#cb4-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-215"><a href="#cb4-215" aria-hidden="true" tabindex="-1"></a>There are 1714, genes with N/S ratio &gt; 1.3. Here is a list of those with the highest N/S ratios - it doesn't tell us much does it!</span>
<span id="cb4-216"><a href="#cb4-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-219"><a href="#cb4-219" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-220"><a href="#cb4-220" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb4-221"><a href="#cb4-221" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb4-222"><a href="#cb4-222" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb4-223"><a href="#cb4-223" aria-hidden="true" tabindex="-1"></a><span class="co">#| output: true</span></span>
<span id="cb4-224"><a href="#cb4-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-225"><a href="#cb4-225" aria-hidden="true" tabindex="-1"></a><span class="co">#high N/S ratio genes</span></span>
<span id="cb4-226"><a href="#cb4-226" aria-hidden="true" tabindex="-1"></a>high_ns_genes <span class="ot">&lt;-</span> table1 <span class="sc">|&gt;</span></span>
<span id="cb4-227"><a href="#cb4-227" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n_s_ratio <span class="sc">&gt;</span> <span class="fl">1.3</span>) <span class="sc">|&gt;</span></span>
<span id="cb4-228"><a href="#cb4-228" aria-hidden="true" tabindex="-1"></a>  <span class="co">#arrange by n_s_ratio descending</span></span>
<span id="cb4-229"><a href="#cb4-229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_s_ratio))</span>
<span id="cb4-230"><a href="#cb4-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-231"><a href="#cb4-231" aria-hidden="true" tabindex="-1"></a>high_ns_genes_named_only <span class="ot">&lt;-</span> high_ns_genes <span class="sc">|&gt;</span></span>
<span id="cb4-232"><a href="#cb4-232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"hypothetical"</span>, genedescription, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>))</span>
<span id="cb4-233"><a href="#cb4-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-234"><a href="#cb4-234" aria-hidden="true" tabindex="-1"></a><span class="co">#show a list of the top 10 genedescriptions</span></span>
<span id="cb4-235"><a href="#cb4-235" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(high_ns_genes_named_only<span class="sc">$</span>genedescription, <span class="dv">10</span>)</span>
<span id="cb4-236"><a href="#cb4-236" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-237"><a href="#cb4-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-238"><a href="#cb4-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-239"><a href="#cb4-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-240"><a href="#cb4-240" aria-hidden="true" tabindex="-1"></a>To explain what kinds of genes have high N/S ratios, we can use a technique called **gene ontology (GO) enrichment analysis**. </span>
<span id="cb4-241"><a href="#cb4-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-242"><a href="#cb4-242" aria-hidden="true" tabindex="-1"></a>:::callout-note</span>
<span id="cb4-243"><a href="#cb4-243" aria-hidden="true" tabindex="-1"></a><span class="fu"># gene ontology (GO)</span></span>
<span id="cb4-244"><a href="#cb4-244" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Gene Ontology (GO) is a system for classifying genes according to their biochemical functions, the biological processes they are involved in, and the cellular locations that the proteins operate in (eg: nucleus, membrane, etc).</span>
<span id="cb4-245"><a href="#cb4-245" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>If we have a gene list, we can use GO enrichment analysis to look for patterns in the kinds of genes that are present in the list.</span>
<span id="cb4-246"><a href="#cb4-246" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb4-247"><a href="#cb4-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-248"><a href="#cb4-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-249"><a href="#cb4-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-250"><a href="#cb4-250" aria-hidden="true" tabindex="-1"></a><span class="fu">### Shallow time: gene enrichment analysis</span></span>
<span id="cb4-251"><a href="#cb4-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-252"><a href="#cb4-252" aria-hidden="true" tabindex="-1"></a>I ran a gene enrichment analysis, looking for enrichment in the **cellular locations** of the **high N/S ratio genes**, that might be subject to positive selection. I also ran an enrichment analysis on the **low N/S ratio genes**, that are subject to strong purifying selection.</span>
<span id="cb4-253"><a href="#cb4-253" aria-hidden="true" tabindex="-1"></a>The results are shown in Figure 4 (positive selection) and Figure 5 (purifying selection) below.</span>
<span id="cb4-254"><a href="#cb4-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-255"><a href="#cb4-255" aria-hidden="true" tabindex="-1"></a><span class="al">![](workshop7-data/table2-location.png)</span></span>
<span id="cb4-256"><a href="#cb4-256" aria-hidden="true" tabindex="-1"></a>**Figure 4. A word cloud showing the cellular locations enriched in positive selection genes.** I selected genes with high N/S ratios (&gt; 1.3), and ran the gene enrichment analysis using <span class="co">[</span><span class="ot">TriTrypDB</span><span class="co">](https://tritrypdb.org/tritrypdb/app)</span>. Larger words indicate greater enrichment, and the grey text intensity indicates the statistical significance of the enrichment (darker = more significant, paler = less significant).</span>
<span id="cb4-257"><a href="#cb4-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-258"><a href="#cb4-258" aria-hidden="true" tabindex="-1"></a><span class="al">![](workshop7-data/table3-location.png)</span></span>
<span id="cb4-259"><a href="#cb4-259" aria-hidden="true" tabindex="-1"></a>**Figure 5. A word cloud showing the cellular locations enriched in genes that are subject to strong purifying selection**. I selected genes with low N/S ratios (&lt; 0.15), and ran the gene enrichment analysis using <span class="co">[</span><span class="ot">TriTrypDB</span><span class="co">](https://tritrypdb.org/tritrypdb/app)</span>. Larger words indicate greater enrichment, and the grey text intensity indicates the statistical significance of the enrichment (darker = more significant, paler = less significant).</span>
<span id="cb4-260"><a href="#cb4-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-261"><a href="#cb4-261" aria-hidden="true" tabindex="-1"></a>:::callout-tip</span>
<span id="cb4-262"><a href="#cb4-262" aria-hidden="true" tabindex="-1"></a><span class="fu"># Discussion points: gene enrichment analysis</span></span>
<span id="cb4-263"><a href="#cb4-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-264"><a href="#cb4-264" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Where do positive selection proteins localise (Figure 4)?</span>
<span id="cb4-265"><a href="#cb4-265" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Where do slowly evolving proteins localise (Figure 5)?</span>
<span id="cb4-266"><a href="#cb4-266" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>In the same way that the MK test example showed us that different sites *within proteins* are subject to different types of selection, what patterns do you observe here?</span>
<span id="cb4-267"><a href="#cb4-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-268"><a href="#cb4-268" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb4-269"><a href="#cb4-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-270"><a href="#cb4-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-271"><a href="#cb4-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-272"><a href="#cb4-272" aria-hidden="true" tabindex="-1"></a><span class="fu">## Deep time analysis</span></span>
<span id="cb4-273"><a href="#cb4-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-274"><a href="#cb4-274" aria-hidden="true" tabindex="-1"></a>Since diversity within species gives rise to divergence between species, we might expect that the genes that evolve rapidly within species would also evolve rapidly between species over long periods of time. We'll examine this now.</span>
<span id="cb4-275"><a href="#cb4-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-276"><a href="#cb4-276" aria-hidden="true" tabindex="-1"></a>To study this, we will now compare how similar the proteins are between *L. donovani* and *L. braziliensis*. You can see that these are quite distant on the phylogenetic tree (Figure 2). We don't know exactly how long ago they diverged, but it will be much longer ago that the within-species differences we saw earlier, so perhaps 100,000 years or even millions of years ago.</span>
<span id="cb4-277"><a href="#cb4-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-278"><a href="#cb4-278" aria-hidden="true" tabindex="-1"></a>To compare the proteins between these two species, I calculated the *percent identity* of each *L. donovani* and *L. braziliensis* ortholog (ie: what percent of amino acids in the alignment are the same). The percent identities range from 22% to 100%. What is remarkable here is that some proteins have changed an enormous amount over this time, while others have not changed at all, showing that **over evolutionary time, genes change at different rates**.</span>
<span id="cb4-279"><a href="#cb4-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-280"><a href="#cb4-280" aria-hidden="true" tabindex="-1"></a>Of the ~8,000 orthologous proteins I selected the most rapidly evolving 1500 genes (lowest percent identity) and the most slowly evolving 1500 genes (highest percent identity), and ran gene enrichment analyses on both sets of genes. </span>
<span id="cb4-281"><a href="#cb4-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-282"><a href="#cb4-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-283"><a href="#cb4-283" aria-hidden="true" tabindex="-1"></a><span class="fu">### Deep time analysis: first look at the results</span></span>
<span id="cb4-284"><a href="#cb4-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-285"><a href="#cb4-285" aria-hidden="true" tabindex="-1"></a>Let's look at the most slowly evolving genes first. Here is a list of the most conserved, slowly evolving genes (with the highest percent identity):</span>
<span id="cb4-286"><a href="#cb4-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-287"><a href="#cb4-287" aria-hidden="true" tabindex="-1"></a>----</span>
<span id="cb4-288"><a href="#cb4-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-291"><a href="#cb4-291" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-292"><a href="#cb4-292" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb4-293"><a href="#cb4-293" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb4-294"><a href="#cb4-294" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb4-295"><a href="#cb4-295" aria-hidden="true" tabindex="-1"></a><span class="co">#| output: true</span></span>
<span id="cb4-296"><a href="#cb4-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-297"><a href="#cb4-297" aria-hidden="true" tabindex="-1"></a><span class="co">#load libraries</span></span>
<span id="cb4-298"><a href="#cb4-298" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb4-299"><a href="#cb4-299" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb4-300"><a href="#cb4-300" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(janitor)</span>
<span id="cb4-301"><a href="#cb4-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-302"><a href="#cb4-302" aria-hidden="true" tabindex="-1"></a><span class="co">#load gene function data</span></span>
<span id="cb4-303"><a href="#cb4-303" aria-hidden="true" tabindex="-1"></a>t5gene <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"week8/table5-genes.tsv"</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-304"><a href="#cb4-304" aria-hidden="true" tabindex="-1"></a>t5gene <span class="ot">&lt;-</span> <span class="fu">clean_names</span>(t5gene)</span>
<span id="cb4-305"><a href="#cb4-305" aria-hidden="true" tabindex="-1"></a><span class="co">#load gene percent id data</span></span>
<span id="cb4-306"><a href="#cb4-306" aria-hidden="true" tabindex="-1"></a>t5id <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"week8/table5-percent-id.tsv"</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-307"><a href="#cb4-307" aria-hidden="true" tabindex="-1"></a>t5id <span class="ot">&lt;-</span> <span class="fu">clean_names</span>(t5id)</span>
<span id="cb4-308"><a href="#cb4-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-309"><a href="#cb4-309" aria-hidden="true" tabindex="-1"></a><span class="co">#merge data: join on ldon_gene (in t4id) matching gene_id (in t4gene)</span></span>
<span id="cb4-310"><a href="#cb4-310" aria-hidden="true" tabindex="-1"></a>table5 <span class="ot">&lt;-</span> <span class="fu">left_join</span>(t5id, t5gene, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ldon_gene"</span> <span class="ot">=</span> <span class="st">"gene_id"</span>))</span>
<span id="cb4-311"><a href="#cb4-311" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-312"><a href="#cb4-312" aria-hidden="true" tabindex="-1"></a><span class="co">#arrange and remove hypothetical proteins</span></span>
<span id="cb4-313"><a href="#cb4-313" aria-hidden="true" tabindex="-1"></a>table5_most_conserved_genes <span class="ot">&lt;-</span> table5 <span class="sc">|&gt;</span></span>
<span id="cb4-314"><a href="#cb4-314" aria-hidden="true" tabindex="-1"></a>  <span class="co">#arrange by percent)_identity ascending</span></span>
<span id="cb4-315"><a href="#cb4-315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(percent_identity,<span class="at">reverse =</span> T) <span class="sc">|&gt;</span></span>
<span id="cb4-316"><a href="#cb4-316" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"hypothetical"</span>, product_description, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>))</span>
<span id="cb4-317"><a href="#cb4-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-318"><a href="#cb4-318" aria-hidden="true" tabindex="-1"></a><span class="co">#show a list of the top 10 product_descriptions</span></span>
<span id="cb4-319"><a href="#cb4-319" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(table5_most_conserved_genes<span class="sc">$</span>product_description, <span class="dv">10</span>)</span>
<span id="cb4-320"><a href="#cb4-320" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-321"><a href="#cb4-321" aria-hidden="true" tabindex="-1"></a>----</span>
<span id="cb4-322"><a href="#cb4-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-323"><a href="#cb4-323" aria-hidden="true" tabindex="-1"></a>This time, there is **certainly** a pattern!</span>
<span id="cb4-324"><a href="#cb4-324" aria-hidden="true" tabindex="-1"></a>\</span>
<span id="cb4-325"><a href="#cb4-325" aria-hidden="true" tabindex="-1"></a>\</span>
<span id="cb4-326"><a href="#cb4-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-327"><a href="#cb4-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-328"><a href="#cb4-328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-329"><a href="#cb4-329" aria-hidden="true" tabindex="-1"></a>And now here's the list of the most rapidly evolving genes (with the *lowest* percent identity):</span>
<span id="cb4-330"><a href="#cb4-330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-331"><a href="#cb4-331" aria-hidden="true" tabindex="-1"></a>----</span>
<span id="cb4-332"><a href="#cb4-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-335"><a href="#cb4-335" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-336"><a href="#cb4-336" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb4-337"><a href="#cb4-337" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb4-338"><a href="#cb4-338" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb4-339"><a href="#cb4-339" aria-hidden="true" tabindex="-1"></a><span class="co">#| output: true</span></span>
<span id="cb4-340"><a href="#cb4-340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-341"><a href="#cb4-341" aria-hidden="true" tabindex="-1"></a><span class="co">#load libraries</span></span>
<span id="cb4-342"><a href="#cb4-342" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb4-343"><a href="#cb4-343" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb4-344"><a href="#cb4-344" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(janitor)</span>
<span id="cb4-345"><a href="#cb4-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-346"><a href="#cb4-346" aria-hidden="true" tabindex="-1"></a><span class="co">#load gene function data</span></span>
<span id="cb4-347"><a href="#cb4-347" aria-hidden="true" tabindex="-1"></a>t4gene <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"week8/table4-genes.tsv"</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-348"><a href="#cb4-348" aria-hidden="true" tabindex="-1"></a>t4gene <span class="ot">&lt;-</span> <span class="fu">clean_names</span>(t4gene)</span>
<span id="cb4-349"><a href="#cb4-349" aria-hidden="true" tabindex="-1"></a><span class="co">#load gene percent id data</span></span>
<span id="cb4-350"><a href="#cb4-350" aria-hidden="true" tabindex="-1"></a>t4id <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"week8/table4-percent-id.tsv"</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-351"><a href="#cb4-351" aria-hidden="true" tabindex="-1"></a>t4id <span class="ot">&lt;-</span> <span class="fu">clean_names</span>(t4id)</span>
<span id="cb4-352"><a href="#cb4-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-353"><a href="#cb4-353" aria-hidden="true" tabindex="-1"></a><span class="co">#merge data: join on ldon_gene (in t4id) matching gene_id (in t4gene)</span></span>
<span id="cb4-354"><a href="#cb4-354" aria-hidden="true" tabindex="-1"></a>table4 <span class="ot">&lt;-</span> <span class="fu">left_join</span>(t4id, t4gene, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ldon_gene"</span> <span class="ot">=</span> <span class="st">"gene_id"</span>))</span>
<span id="cb4-355"><a href="#cb4-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-356"><a href="#cb4-356" aria-hidden="true" tabindex="-1"></a><span class="co">#arrange and remove hypothetical proteins</span></span>
<span id="cb4-357"><a href="#cb4-357" aria-hidden="true" tabindex="-1"></a>table4_least_conserved_genes <span class="ot">&lt;-</span> table4 <span class="sc">|&gt;</span></span>
<span id="cb4-358"><a href="#cb4-358" aria-hidden="true" tabindex="-1"></a>  <span class="co">#arrange by percent)_identity ascending</span></span>
<span id="cb4-359"><a href="#cb4-359" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(percent_identity) <span class="sc">|&gt;</span></span>
<span id="cb4-360"><a href="#cb4-360" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"hypothetical"</span>, product_description, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>))</span>
<span id="cb4-361"><a href="#cb4-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-362"><a href="#cb4-362" aria-hidden="true" tabindex="-1"></a><span class="co">#show a list of the top 10 product_descriptions</span></span>
<span id="cb4-363"><a href="#cb4-363" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(table4_least_conserved_genes<span class="sc">$</span>product_description, <span class="dv">10</span>)</span>
<span id="cb4-364"><a href="#cb4-364" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-365"><a href="#cb4-365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-366"><a href="#cb4-366" aria-hidden="true" tabindex="-1"></a>----</span>
<span id="cb4-367"><a href="#cb4-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-368"><a href="#cb4-368" aria-hidden="true" tabindex="-1"></a>I am not such a 'Leishmaniac' that I can see a pattern here, but do you notice anything that reminds me of the shallow time results?</span>
<span id="cb4-369"><a href="#cb4-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-370"><a href="#cb4-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-371"><a href="#cb4-371" aria-hidden="true" tabindex="-1"></a><span class="fu">### Deep time analysis: GO enrichment analysis</span></span>
<span id="cb4-372"><a href="#cb4-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-373"><a href="#cb4-373" aria-hidden="true" tabindex="-1"></a>To understand these large lists of genes, I again ran gene enrichment analyses on both the most slowly evolving genes (highest percent identity) and the most rapidly evolving genes (lowest percent identity). The results are shown in Figure 6 (slowly evolving genes) and Figure 7 (rapidly evolving genes) below.</span>
<span id="cb4-374"><a href="#cb4-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-375"><a href="#cb4-375" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/table5-wordcloud.png)</span></span>
<span id="cb4-376"><a href="#cb4-376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-377"><a href="#cb4-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-378"><a href="#cb4-378" aria-hidden="true" tabindex="-1"></a>**Figure 6. A word cloud showing the cellular locations enriched in slowly evolving genes.** I selected genes with high percent identity (most slowly evolving), and ran the gene enrichment analysis using <span class="co">[</span><span class="ot">TriTrypDB</span><span class="co">](https://tritrypdb.org/tritrypdb/app)</span>. Larger words indicate greater enrichment, and the grey text intensity indicates the statistical significance of the enrichment (darker = more significant, paler = less significant).</span>
<span id="cb4-379"><a href="#cb4-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-380"><a href="#cb4-380" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/table4-wordcloud.png)</span></span>
<span id="cb4-381"><a href="#cb4-381" aria-hidden="true" tabindex="-1"></a>**Figure 7. A word cloud showing the cellular locations enriched in rapidly evolving genes.** I selected genes with low percent identity (most rapidly evolving), and ran the gene enrichment analysis using <span class="co">[</span><span class="ot">TriTrypDB</span><span class="co">](https://tritrypdb.org/tritrypdb/app)</span>. Larger words indicate greater enrichment, and the grey text intensity indicates the statistical significance of the enrichment (darker = more significant, paler = less significant).</span>
<span id="cb4-382"><a href="#cb4-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-383"><a href="#cb4-383" aria-hidden="true" tabindex="-1"></a><span class="fu">## Deep time analysis: discussion points</span></span>
<span id="cb4-384"><a href="#cb4-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-385"><a href="#cb4-385" aria-hidden="true" tabindex="-1"></a>:::callout-tip</span>
<span id="cb4-386"><a href="#cb4-386" aria-hidden="true" tabindex="-1"></a><span class="fu"># Discussion points: deep time gene enrichment analysis</span></span>
<span id="cb4-387"><a href="#cb4-387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-388"><a href="#cb4-388" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Consider what kinds of genes are slowly evolving (Figure 6) and rapidly evolving (Figure 7).</span>
<span id="cb4-389"><a href="#cb4-389" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Is this consistent with what we saw in the shallow time analysis?</span>
<span id="cb4-390"><a href="#cb4-390" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>We used slightly different methods (N/S ratios vs percent identity), and different species comparisons (*L. donovani* within-species vs *L. donovani* vs *L. braziliensis* between-species). </span>
<span id="cb4-391"><a href="#cb4-391" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Yet we see similar patterns. What does this tell us about the processes of selection that operate within species (*L. donovani*) and between species?</span>
<span id="cb4-392"><a href="#cb4-392" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Are rates of change higher for proteins the insides of cells, or the outsides of cells? Why might this be?</span>
<span id="cb4-393"><a href="#cb4-393" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb4-394"><a href="#cb4-394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-395"><a href="#cb4-395" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb4-396"><a href="#cb4-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-397"><a href="#cb4-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-398"><a href="#cb4-398" aria-hidden="true" tabindex="-1"></a><span class="fu">## Very deep time analysis</span></span>
<span id="cb4-399"><a href="#cb4-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-400"><a href="#cb4-400" aria-hidden="true" tabindex="-1"></a>So far we have looked at genetic differences within *L. donovani* (shallow time) and between *L. donovani* and *L. braziliensis* (deep time). Now we will compare Leishmania proteins to the fission yeast *Schizosaccharomyces pombe*, which diverged from Leishmania a very long time ago, perhaps a billion years ago.</span>
<span id="cb4-401"><a href="#cb4-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-402"><a href="#cb4-402" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-403"><a href="#cb4-403" aria-hidden="true" tabindex="-1"></a>Again, I calculated the percent identity of each *L. donovani* and *S. pombe* ortholog (ie: what percent of amino acids in the alignment are the same). </span>
<span id="cb4-404"><a href="#cb4-404" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-405"><a href="#cb4-405" aria-hidden="true" tabindex="-1"></a>This time, we will only look at the conserved, slowly evolving proteins, because the rapidly evolving proteins have changed so much that we cannot identify orthologs between these very distantly-related species. The top 150 conserved proteins (with the highest percent identity) are between 58% and 83% identical between *L. donovani* and *S. pombe*.</span>
<span id="cb4-406"><a href="#cb4-406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-407"><a href="#cb4-407" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/table6-wordcloud.png)</span></span>
<span id="cb4-408"><a href="#cb4-408" aria-hidden="true" tabindex="-1"></a>**Figure 8. A word cloud showing the cellular locations enriched in very slowly evolving genes.** I selected genes with high percent identity (most slowly evolving), and ran the gene enrichment analysis using <span class="co">[</span><span class="ot">TriTrypDB</span><span class="co">](https://tritrypdb.org/tritrypdb/app)</span>. Larger words indicate greater enrichment, and the grey text intensity indicates the statistical significance of the enrichment (darker = more significant, paler = less significant). </span>
<span id="cb4-409"><a href="#cb4-409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-410"><a href="#cb4-410" aria-hidden="true" tabindex="-1"></a>Here there is one very clear outline: **the ribosome**.</span>
<span id="cb4-411"><a href="#cb4-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-412"><a href="#cb4-412" aria-hidden="true" tabindex="-1"></a><span class="fu">## Final discussion points</span></span>
<span id="cb4-413"><a href="#cb4-413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-414"><a href="#cb4-414" aria-hidden="true" tabindex="-1"></a>:::callout-tip</span>
<span id="cb4-415"><a href="#cb4-415" aria-hidden="true" tabindex="-1"></a><span class="fu"># Discussion points</span></span>
<span id="cb4-416"><a href="#cb4-416" aria-hidden="true" tabindex="-1"></a>We observe a fairly consistent pattern across all time scales that we have examined: genes that are located on the outsides of cells tend to evolve rapidly, while genes that are located deep within cells tend to evolve slowly. of course, there are exceptions, and the rates of protein change are affected by many other factors, but this general pattern is very clear. Let's consider this.</span>
<span id="cb4-417"><a href="#cb4-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-418"><a href="#cb4-418" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Why do you think that ribosomal proteins are so highly conserved over such long periods of time (Figure 8)?</span>
<span id="cb4-419"><a href="#cb4-419" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Given the function of ribosomes in the cell, what might be the consequences of a mutation that reduced protein translation accuracy by 0.1%?</span>
<span id="cb4-420"><a href="#cb4-420" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Why do you think that proteins located on the outsides of cells tend to evolve rapidly?</span>
<span id="cb4-421"><a href="#cb4-421" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb4-422"><a href="#cb4-422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-423"><a href="#cb4-423" aria-hidden="true" tabindex="-1"></a>For those that are curious <span class="co">[</span><span class="ot">the genes lists that I used for this analysis are here</span><span class="co">](https://docs.google.com/spreadsheets/d/1KmtZizHfnWdP1V2PJQZXV5G0p_NLEIM2vR8yGiTule0/edit?usp=sharing)</span>. You'll need to be signed in with your University of York Google account to access it.</span>
<span id="cb4-424"><a href="#cb4-424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-425"><a href="#cb4-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-426"><a href="#cb4-426" aria-hidden="true" tabindex="-1"></a><span class="fu"># Summary</span></span>
<span id="cb4-427"><a href="#cb4-427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-428"><a href="#cb4-428" aria-hidden="true" tabindex="-1"></a>Today, we examined the patterns of evolutionary change within and between species. We observed that:</span>
<span id="cb4-429"><a href="#cb4-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-430"><a href="#cb4-430" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Different kinds of genes evolve at different rates, both within species and between species</span>
<span id="cb4-431"><a href="#cb4-431" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>These patterns are consistent over both short time scales (within species) and long time scales</span>
<span id="cb4-432"><a href="#cb4-432" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Genes that are located on the outsides of cells tend to evolve rapidly, while genes that are located deep within cells tend to evolve slowly</span>
<span id="cb4-433"><a href="#cb4-433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-434"><a href="#cb4-434" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-435"><a href="#cb4-435" aria-hidden="true" tabindex="-1"></a>**That is the end for today.**</span>
<span id="cb4-436"><a href="#cb4-436" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-437"><a href="#cb4-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-438"><a href="#cb4-438" aria-hidden="true" tabindex="-1"></a><span class="fu"># After the workshop: exam style questions</span></span>
<span id="cb4-439"><a href="#cb4-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-440"><a href="#cb4-440" aria-hidden="true" tabindex="-1"></a>**Question 1. If the rate of mutation is the same for all genes in a genome, and we compare two very closely related species, we should observe the same numbers of mutations in each gene. Is this correct?**</span>
<span id="cb4-441"><a href="#cb4-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-442"><a href="#cb4-442" aria-hidden="true" tabindex="-1"></a>In your answer, explain:</span>
<span id="cb4-443"><a href="#cb4-443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-444"><a href="#cb4-444" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>If we *do* see the same numbers of mutations in each gene, or not</span>
<span id="cb4-445"><a href="#cb4-445" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>What genetic and evolutionary factors might cause us to *not* observe the same numbers of mutations in each gene</span>
<span id="cb4-446"><a href="#cb4-446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-447"><a href="#cb4-447" aria-hidden="true" tabindex="-1"></a>Hint: consider both stochastic factors (randomness) and biological processes of evolutionary change. </span>
<span id="cb4-448"><a href="#cb4-448" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-449"><a href="#cb4-449" aria-hidden="true" tabindex="-1"></a>**Question 2. Given what you know about the fitness effects of mutations, which population genetic process would we expect to have the main impact on differences in evolutionary rates between genes?**</span>
<span id="cb4-450"><a href="#cb4-450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-451"><a href="#cb4-451" aria-hidden="true" tabindex="-1"></a>In your answer, explain:</span>
<span id="cb4-452"><a href="#cb4-452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-453"><a href="#cb4-453" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>What the main population genetic process is</span>
<span id="cb4-454"><a href="#cb4-454" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Why this results in differences in evolutionary rates between genes</span>
<span id="cb4-455"><a href="#cb4-455" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>How we could test this hypothesis using genetic data from within and between species</span>
<span id="cb4-456"><a href="#cb4-456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-457"><a href="#cb4-457" aria-hidden="true" tabindex="-1"></a></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>