<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.33">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Daniel Jeffares">
<meta name="dcterms.date" content="2025-02-24">

<title>BIO00056I</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-6fbc3b7d0ac83ab8a043646d6c6d94a5.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-6f9ef9b5a12f5b8ef14302403eafb8d0.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">BIO00056I</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">BIO00056I</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./workshop5.html" aria-current="page"> 
<span class="menu-text">workshop 5</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./workshop6.html"> 
<span class="menu-text">workshop 6</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./workshop7.html"> 
<span class="menu-text">workshop 7</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./workshop9.html"> 
<span class="menu-text">workshop 9</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./worksheet10.html"> 
<span class="menu-text">worksheet 10</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul class="collapse">
  <li><a href="#learning-objectives" id="toc-learning-objectives" class="nav-link active" data-scroll-target="#learning-objectives"><span class="header-section-number">1</span> Learning objectives</a></li>
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction"><span class="header-section-number">2</span> Introduction</a>
  <ul class="collapse">
  <li><a href="#phylogenetic-trees-and-the-molecular-clock" id="toc-phylogenetic-trees-and-the-molecular-clock" class="nav-link" data-scroll-target="#phylogenetic-trees-and-the-molecular-clock"><span class="header-section-number">2.1</span> Phylogenetic trees and the molecular clock</a></li>
  <li><a href="#the-molecular-clock-and-the-real-world" id="toc-the-molecular-clock-and-the-real-world" class="nav-link" data-scroll-target="#the-molecular-clock-and-the-real-world"><span class="header-section-number">2.2</span> The molecular clock and the real world</a></li>
  </ul></li>
  <li><a href="#exercises" id="toc-exercises" class="nav-link" data-scroll-target="#exercises"><span class="header-section-number">3</span> Exercises</a>
  <ul class="collapse">
  <li><a href="#obtaining-data" id="toc-obtaining-data" class="nav-link" data-scroll-target="#obtaining-data"><span class="header-section-number">3.1</span> Obtaining data</a></li>
  <li><a href="#visualising-and-analysing-data" id="toc-visualising-and-analysing-data" class="nav-link" data-scroll-target="#visualising-and-analysing-data"><span class="header-section-number">3.2</span> Visualising and analysing data</a></li>
  <li><a href="#starting-analysis-in-mega" id="toc-starting-analysis-in-mega" class="nav-link" data-scroll-target="#starting-analysis-in-mega"><span class="header-section-number">3.3</span> Starting analysis in Mega</a></li>
  <li><a href="#group-the-strains-by-year." id="toc-group-the-strains-by-year." class="nav-link" data-scroll-target="#group-the-strains-by-year."><span class="header-section-number">3.4</span> Group the strains by year.</a></li>
  <li><a href="#building-a-phylogenetic-tree" id="toc-building-a-phylogenetic-tree" class="nav-link" data-scroll-target="#building-a-phylogenetic-tree"><span class="header-section-number">3.5</span> Building a phylogenetic tree</a></li>
  <li><a href="#measuring-molecular-change-over-time" id="toc-measuring-molecular-change-over-time" class="nav-link" data-scroll-target="#measuring-molecular-change-over-time"><span class="header-section-number">3.6</span> Measuring molecular change over time</a></li>
  <li><a href="#analysis-of-genetic-change-with-time" id="toc-analysis-of-genetic-change-with-time" class="nav-link" data-scroll-target="#analysis-of-genetic-change-with-time"><span class="header-section-number">3.7</span> Analysis of genetic change with time</a></li>
  <li><a href="#optional-does-the-influenza-virus-spread-across-the-tasman-sea" id="toc-optional-does-the-influenza-virus-spread-across-the-tasman-sea" class="nav-link" data-scroll-target="#optional-does-the-influenza-virus-spread-across-the-tasman-sea"><span class="header-section-number">3.8</span> Optional: Does the influenza virus spread across the Tasman Sea?</a></li>
  <li><a href="#interpreting-the-tasman-sea-data" id="toc-interpreting-the-tasman-sea-data" class="nav-link" data-scroll-target="#interpreting-the-tasman-sea-data"><span class="header-section-number">3.9</span> Interpreting the Tasman sea data</a></li>
  </ul></li>
  <li><a href="#summary-what-we-have-learned" id="toc-summary-what-we-have-learned" class="nav-link" data-scroll-target="#summary-what-we-have-learned"><span class="header-section-number">4</span> Summary: what we have learned</a></li>
  <li><a href="#exam-style-questions" id="toc-exam-style-questions" class="nav-link" data-scroll-target="#exam-style-questions"><span class="header-section-number">5</span> Exam style questions</a>
  <ul class="collapse">
  <li><a href="#question-1-10-points" id="toc-question-1-10-points" class="nav-link" data-scroll-target="#question-1-10-points"><span class="header-section-number">5.1</span> Question 1 (10 points)</a></li>
  <li><a href="#question-2-10-points" id="toc-question-2-10-points" class="nav-link" data-scroll-target="#question-2-10-points"><span class="header-section-number">5.2</span> Question 2 (10 points)</a></li>
  </ul></li>
  <li><a href="#further-reading" id="toc-further-reading" class="nav-link" data-scroll-target="#further-reading"><span class="header-section-number">6</span> Further reading</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">BIO00056I</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
<p class="subtitle lead">Workshop 5: Phylogenies &amp; Molecular Clock</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Daniel Jeffares </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 24, 2025</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p>We do not advise doing this workshop on a Mac since Mega software does not work very well on the Mac OS. It is best to use the University PCs which have the software already installed.</p>
</div>
</div>
<section id="learning-objectives" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Learning objectives</h1>
<p>The aim of this practical is to learn to visualise and interpret the data using phylogenetic analyses. You will see that we can extract a lot of information about evolutionary processes and populations from a phylogeny.</p>
<p>By the end of this workshop, you should be able to:</p>
<ul>
<li>Build phylogenetic trees using using MEGA software</li>
<li>Infer evolutionary relationships based on phylogeny</li>
<li>Correlate molecular evolutionary changes with time using R</li>
</ul>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Glossary
</div>
</div>
<div class="callout-body-container callout-body">
<p>Technical definitions for this workshop.</p>
<ul>
<li><p><strong>molecular clock</strong>: The observation that mutations accumulate at a roughly constant rate over time.</p></li>
<li><p><strong>node</strong> (of a phylogenetic tree): A point where branches split, representing either a common ancestor (internal node) or a modern species (terminal node).</p></li>
<li><p><strong>branch</strong> (of a phylogenetic tree): A line connecting nodes, representing an evolutionary lineage. Branch length indicates the amount of evolutionary change. Due to the molecular clock hypothesis, branch lengths can be proportional to time.</p></li>
<li><p><strong>mutation rate</strong>: The frequency at which genetic changes occur in DNA over time, typically measured as substitutions per site per year.</p></li>
<li><p><strong>purifying selection</strong>: Natural selection that removes harmful mutations from a population, acting to preserve functional DNA sequences.</p></li>
</ul>
</div>
</div>
</section>
<section id="introduction" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Introduction</h1>
<section id="phylogenetic-trees-and-the-molecular-clock" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="phylogenetic-trees-and-the-molecular-clock"><span class="header-section-number">2.1</span> Phylogenetic trees and the molecular clock</h2>
<p>Much of what we can infer from phylogenies uses the model of the ‘molecular clock’. This is the observation that there tends to be a uniform, ‘clock-like’ rate of genetic change per year both <em>within</em> and <em>between</em> species. The <em>rate</em> of change differs between very different species and between genes.</p>
<p>Rates of change between species differ because:</p>
<ul>
<li>small organisms (like viruses and bacteria) have short generation times</li>
<li>small organisms have high mutation rates</li>
</ul>
<p>Rates of change between genes within a genome differ because the amount of <em>purifying selection</em> acting on a gene differs.</p>
<p>You can refresh your knowledge of Phylogenetic trees and the molecular clock from these two recordings:</p>
<ul>
<li><p><a href="https://york.cloud.panopto.eu/Panopto/Pages/Viewer.aspx?id=327fb8cc-9f70-4207-8afc-b0b2012cd0d4">Phylogenies and molecular clock Part 1</a></p></li>
<li><p><a href="https://york.cloud.panopto.eu/Panopto/Pages/Viewer.aspx?id=22c2aa19-d1eb-4819-88d0-b0b2013211ee">Phylogenies and molecular clock Part 2</a></p></li>
</ul>
</section>
<section id="the-molecular-clock-and-the-real-world" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="the-molecular-clock-and-the-real-world"><span class="header-section-number">2.2</span> The molecular clock and the real world</h2>
<p>Evolutionary thinking can also help us to understand disease. For example, the global pandemic of the coronavirus SARS-Cov2 (which causes COVID-19 disease) is an RNA virus. As with all viruses, SARS-Cov2 is constantly undergoing evolutionary change.</p>
<p>The initial outbreak of the SARS-Cov2 virus in Wuhan China, was dated with amazing accuracy (to the month) to August 2019 <a href="https://doi.org/10.3390/v13091790">using a time tree</a>. Being able to tell when outbreaks occur, as well as where, can be a very powerful tool as it can supplement what we can see from epidemiology. Analysis of DNA sequences is the only way we can obtain this information.</p>
</section>
</section>
<section id="exercises" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Exercises</h1>
<p>In this workshop, we will study the evolution of the influenza virus. Influenza is an RNA virus that causes seasonal epidemics and occasional pandemics in humans. The influenza virus evolves very rapidly, which is why we need a new flu vaccine every year.</p>
<section id="obtaining-data" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="obtaining-data"><span class="header-section-number">3.1</span> Obtaining data</h2>
<p>There is a wealth of freely available sequence data available on GenBank. The downside is that most data is often poorly annotated and misses the critical associated information (e.g.&nbsp;date and place of collection). Here we will take advantage of the <a href="https://www.ncbi.nlm.nih.gov/genomes/FLU/Database/nph-select.cgi?go=database">influenza virus resource database</a>, which is relatively well curated.</p>
<p>We have downloaded all the human full-length H3N2 type sequences from the hemagglutinin (HA) sequences from Wellington (New Zealand). Hemagglutinin (HA) a glycoprotein gene that is on the influenza virus surface.</p>
<p>You can <a href="https://drive.google.com/file/d/1cv4OAmLBRRad_7ObJrlljysWKO9KhX8l/view?usp=sharing">download the file H3N2_wellington.fasta here</a>. This file has HA sequences from 72 influenza strains from 1985 to 2005. Sequences are already aligned, and sorted by date.</p>
</section>
<section id="visualising-and-analysing-data" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="visualising-and-analysing-data"><span class="header-section-number">3.2</span> Visualising and analysing data</h2>
</section>
<section id="starting-analysis-in-mega" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="starting-analysis-in-mega"><span class="header-section-number">3.3</span> Starting analysis in Mega</h2>
<p>We will use the Mega software for all analyses. It should be installed on your computers. If you would like to use your own computer, you can download it from <a href="http://www.megasoftware.net/">the mega wesbsite</a>. Use the latest version, Mega 11.</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p>Mega software does not work very well on the Mac OS.</p>
</div>
</div>
<p>Once you start Mega, you will see the following window. Be careful here, because Mega has multiple windows.</p>
<p><img src="mega1.jpg" class="img-fluid"></p>
<p>Next, load the H3N2_Wellington_seq.fa file through the <code>Data &gt; Open a File/Session Menu</code> (Ctrl+O).</p>
<p>Then click:</p>
<ul>
<li><code>Analyze</code> (not align)</li>
<li>OK to <code>Nucleotide Sequences</code></li>
<li>Yes to <code>Protein-coding nucleotide sequence data</code></li>
<li>OK to <code>Standard Genetic Code</code>.</li>
</ul>
<p>You shuld see the window shown below</p>
<p><img src="mega2.jpg" class="img-fluid"></p>
</section>
<section id="group-the-strains-by-year." class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="group-the-strains-by-year."><span class="header-section-number">3.4</span> Group the strains by year.</h2>
<p>Click on the <code>TA</code> window at top left to see the Sequence Data Explorer.</p>
<p>You can scroll around and resize this window. This data file comprises 72 complete HA sequences from influenza H3N2 sampled between 1985 to 2005.</p>
<p>Now let’s group the strains by year. This will be fairly easy because we sorted the data by year before we downloaded it. Samples have the year embedded in the name.</p>
<p>For example:</p>
<ul>
<li>this one is from 1985: <code>Influenza A virus (A/Wellington/4/1985(H3N2))</code></li>
<li>and this from 2004: <code>Influenza A virus (A/Wellington/34/2004(H3N2))</code></li>
</ul>
<p>First, go to the <code>Display</code> menu, and click on <code>Show Group Names</code>. This will help you to keep track of what you‘ve done.</p>
<p><img src="mega3.jpg" class="img-fluid" width="300"></p>
<p>To group strains, select one or more strains from a year by clicking on the name of the first strain in the grey part of the window, and shift+click to select the last one, as below:</p>
<p>Then, go to the <code>Edit</code> menu, and click on <code>Create Group from Selection</code>. A new group will appear in the left-hand panel. Rename it to the year of sampling (e.g.&nbsp;1985).</p>
<p>Repeat this for all years from 1985 to 2005.</p>
<p><img src="mega4.jpg" class="img-fluid" width="600"></p>
</section>
<section id="building-a-phylogenetic-tree" class="level2" data-number="3.5">
<h2 data-number="3.5" class="anchored" data-anchor-id="building-a-phylogenetic-tree"><span class="header-section-number">3.5</span> Building a phylogenetic tree</h2>
<p>As you will have seen, there are a large number of possible options for analysis in MEGA. However, we will build one tree along a single, robust methodology.</p>
<p>Go to the <code>Phylogeny</code> menu on the main menu:</p>
<p><code>Phylogeny -&gt; Construct/Test Neighbor-Joining</code></p>
<p>Choose these options:</p>
<ul>
<li>For Model/Method, choose the <code>Kimura 2-parameter model</code> of evolution</li>
<li>For Test of Phylogeny, choose <code>Bootstrap method</code> and select 1000 bootstraps.</li>
<li>Leave all other options as default</li>
</ul>
<p>Then click OK.</p>
<p><strong>Bootstraps</strong> represent statistical support for individual clades, values in excess of 70-80% are considered well supported.</p>
<p>After less than a minute, you will get a phylogenetic tree, a new Mega window.</p>
<p>It does not look very informative, so change the presentation of the tree using the menus on the left:</p>
<ul>
<li>First, deselect <code>Taxon Names</code></li>
<li>Under Layout, select <code>Auto-size tree</code></li>
<li>Define a root. The branch leading to the sequence from 1985 makes a biologically reasonable root. To define a root, simply right click with your mouse on the chosen branch and select <code>Place root</code>.</li>
<li>You may also need to click toggle <code>Scaling</code> of the tree in the <code>Layout</code> menu.</li>
</ul>
<p>You should see a tree like the one shown below.</p>
<p><img src="mega5.jpg" class="img-fluid" width="700"></p>
<section id="understanding-the-tree" class="level3" data-number="3.5.1">
<h3 data-number="3.5.1" class="anchored" data-anchor-id="understanding-the-tree"><span class="header-section-number">3.5.1</span> Understanding the tree</h3>
<p>What do you notice about the clustering of years on this phylogeny?</p>
<ul>
<li>In particular, look at the branch lengths (ie: genetic distance) from the root of the tree, to the tips.</li>
<li>This indicates how many mutations have occurred since the common ancestor.</li>
</ul>
</section>
</section>
<section id="measuring-molecular-change-over-time" class="level2" data-number="3.6">
<h2 data-number="3.6" class="anchored" data-anchor-id="measuring-molecular-change-over-time"><span class="header-section-number">3.6</span> Measuring molecular change over time</h2>
<p>An important question in phylogenetics is whether the accumulation of mutation (substitution rate) is constant over time. One way to test for this would be to count the number of mutations from the root to each tip. Estimating the most likely root is not completely straightforward and counting the number of mutations from root to tip would require some scripting well beyond the scope of this practical. However, we can test whether genetic distances increase linearly with time.</p>
<p>To address the issue of linearity between substitution rates and time, we can first compute genetic distances between the sequences grouped by years. To do this:</p>
<ul>
<li>Go to the <code>Distance</code> menu (at the top of the main Mega window)</li>
<li>Click <code>Compute Between Group Mean Distance</code> (see image below)</li>
<li>Keep the <code>Kimura 2 parameter model</code></li>
<li>Click <code>OK</code></li>
</ul>
<p><img src="mega6.jpg" class="img-fluid" width="400"></p>
<p>You should see the matrix shown below.</p>
<p><img src="mega7.jpg" class="img-fluid" width="600"></p>
<p>Next, export this data to a file:</p>
<ul>
<li>In the matrix window, click on the <code>File</code> menu</li>
<li>Then <code>Export/Print Distances</code></li>
<li>Output format: <code>text</code></li>
<li>Export Type: <code>column</code></li>
</ul>
<p>You should see a window something like the image below.</p>
<p><img src="mega8.jpg" class="img-fluid" width="400"></p>
<p>Finally edit and save this file: - Remove all the text above Species1 Species2 Distance - Remove the text below the data table, “Table. Estimates of Evolutionary Divergence over Sequence …” etc</p>
<ul>
<li>Edit the <code>Species1 Species2 Distance</code> text to <code>date1 date1 Distance</code></li>
<li>Choose <code>File &gt; Save As</code></li>
<li>Name the file <code>H3N2_wellington.distances.txt</code></li>
</ul>
<p>You can also <a href="https://drive.google.com/file/d/1Um4R5L_BeKCy9qzdTM10zbyuGi50a_o6/view?usp=sharing">obtain a copy of this file here</a>.</p>
</section>
<section id="analysis-of-genetic-change-with-time" class="level2" data-number="3.7">
<h2 data-number="3.7" class="anchored" data-anchor-id="analysis-of-genetic-change-with-time"><span class="header-section-number">3.7</span> Analysis of genetic change with time</h2>
<p>The remainder of this part of the workshop will use R Studio, so start RStudio now.</p>
<p>All the commands you’ll need are in the R Script <code>BIO00056I-influenza-practical-2024.R</code>, which is <a href="https://drive.google.com/file/d/1j-9ktSYBZiPBxiwZpy8QaBWX7aqA54c8/view?usp=sharing">available to download here</a>.</p>
<p>Or you can create you own script, and copy the comnands from here as we go along.</p>
<section id="set-up-the-script" class="level3" data-number="3.7.1">
<h3 data-number="3.7.1" class="anchored" data-anchor-id="set-up-the-script"><span class="header-section-number">3.7.1</span> Set up the script</h3>
<p>First, set up your script:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#BIO00056I Influenza Virus Practical ----</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co"># SET UP ----</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">#set your working directory</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Your working directory will be different!</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"~/path/to/your/files/"</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co">#You can use the Session menu to set this with:</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Session menu  &gt; Set Working Directory &gt; Choose Directory</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co">#then select the directory where your files are.</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="co">#load the tidyverse</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
<section id="loading-and-plotting-the-data" class="level3" data-number="3.7.2">
<h3 data-number="3.7.2" class="anchored" data-anchor-id="loading-and-plotting-the-data"><span class="header-section-number">3.7.2</span> Loading and plotting the data</h3>
<p>Then load the Wellgonton data, and make a plot</p>
<p>&nbsp;</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># WELLINGTON DATA ----</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co">#open your distances file, into a data frame:</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Your distance file may have a different name</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>flu <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(<span class="fu">read.table</span>(<span class="st">"H3N2_wellington.distances.txt"</span>,<span class="at">h=</span>T))</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co">#calculate how many years between two groups</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>flu<span class="sc">$</span>time.passed <span class="ot">&lt;-</span> flu<span class="sc">$</span>date2 <span class="sc">-</span> flu<span class="sc">$</span>date1</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co">#see if distance increases with time, as mutations accumulate</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="co">#make the plot</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>flu_plot<span class="ot">&lt;-</span><span class="fu">ggplot</span>(flu,<span class="fu">aes</span>(<span class="at">x=</span>time.passed, <span class="at">y=</span> Distance))<span class="sc">+</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"time distance between groups (years)"</span>)<span class="sc">+</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"genetic distance between groups"</span>)<span class="sc">+</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="co">#view the plot</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>flu_plot </span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a><span class="co">#now add a line of best fit:</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>flu_plot <span class="sc">+</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>)</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a><span class="co">#examine whether the correlation is statistically significant: </span></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(flu<span class="sc">$</span>time.passed,flu<span class="sc">$</span>Distance)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="interpret-the-data" class="level3" data-number="3.7.3">
<h3 data-number="3.7.3" class="anchored" data-anchor-id="interpret-the-data"><span class="header-section-number">3.7.3</span> Interpret the data</h3>
<p>You should see a good correlation between genetic distance and time distance.</p>
<ul>
<li>What causes this? (which process of evolution)</li>
<li>Why is there such a strong correlation?</li>
</ul>
<p>We can observe this relationship over just a few decades? - What does this tell us about the rate of change?</p>
</section>
</section>
<section id="optional-does-the-influenza-virus-spread-across-the-tasman-sea" class="level2" data-number="3.8">
<h2 data-number="3.8" class="anchored" data-anchor-id="optional-does-the-influenza-virus-spread-across-the-tasman-sea"><span class="header-section-number">3.8</span> Optional: Does the influenza virus spread across the Tasman Sea?</h2>
<p>Another use of virus genetic data is to determine how fast pathogens spread. Here, we’ll show an example, where we examine whether people in New Zealand share the same strains as their neighbors in Australia across the Tasman Sea.</p>
<p>We have downloaded and aligned 118 strains from Australia and New Zealand. You can download the file H3N2_tasman.fasta here. These sequences are already assigned.</p>
<p>Then open MEGA, selecting <code>Analyse</code>, and</p>
<ul>
<li>Select the Distance menu, and Compute Pairwise Distances.</li>
<li>You will see a matrix of distances</li>
<li>In this window, select the <code>File</code> menu, and:
<ul>
<li>Export/Print Distances</li>
<li>Output format: CSV</li>
<li>Export type: Column</li>
</ul></li>
</ul>
<p>As before remove all the text above and below the data:</p>
<p>Above:</p>
<ul>
<li>Title: fasta file</li>
<li>Description:</li>
</ul>
<p>Below:</p>
<ul>
<li>Table. Estimates of Evolutionary Divergence between Sequences</li>
<li>The number of base substitutions per site from between sequences are shown.</li>
</ul>
<p>Then edit the header line to be: <code>region1 region2 Distance</code>, ensuring there is a space character between each column names.</p>
<p>Finally, save the file as <code>H3N2_tasman.distances.csv</code>.</p>
<p>You can also obtain <a href="https://drive.google.com/file/d/15xnaP5Ju9oxfYEugDqtZ8YDsokYUumml/view?usp=sharing">a copy of the H3N2_tasman.distances.csv file here</a>.</p>
<p>Then examine the between-country differences and compare these to within-country differences, using the code in the R script, as below:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># TASMAN SEA DATA ----</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co">#open the Tasman genetic distance data</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>tas <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"H3N2_tasman.distances.csv"</span>)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co">#add country1 and country2 columns</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>tas<span class="sc">$</span>country1 <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>tas<span class="sc">$</span>country2 <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="co">#label the countries using grep</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>tas[<span class="fu">grep</span>(<span class="st">"NZ"</span>, tas<span class="sc">$</span>region1),]<span class="sc">$</span>country1 <span class="ot">=</span> <span class="st">"NZ"</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>tas[<span class="fu">grep</span>(<span class="st">"AUS"</span>, tas<span class="sc">$</span>region1),]<span class="sc">$</span>country1 <span class="ot">=</span> <span class="st">"AUS"</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>tas[<span class="fu">grep</span>(<span class="st">"NZ"</span>, tas<span class="sc">$</span>region2),]<span class="sc">$</span>country2 <span class="ot">=</span> <span class="st">"NZ"</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>tas[<span class="fu">grep</span>(<span class="st">"AUS"</span>, tas<span class="sc">$</span>region2),]<span class="sc">$</span>country2 <span class="ot">=</span> <span class="st">"AUS"</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="co">#add a comparison.type column</span></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>tas<span class="sc">$</span>comparison.type <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a><span class="co">#mark the within country data</span></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>tas[<span class="fu">which</span>(tas<span class="sc">$</span>region1 <span class="sc">==</span> tas<span class="sc">$</span>region2),]<span class="sc">$</span>comparison.type <span class="ot">&lt;-</span> <span class="st">'within country'</span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a><span class="co">#mark the between country data</span></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>tas[<span class="fu">which</span>(tas<span class="sc">$</span>region1 <span class="sc">!=</span> tas<span class="sc">$</span>region2),]<span class="sc">$</span>comparison.type <span class="ot">&lt;-</span> <span class="st">'between country'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now we make some plots:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co">#plot this data </span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co">#to examine whether genetic distances differ</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co">#within a country vs between countries</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(tas,<span class="fu">aes</span>(<span class="at">y=</span>Distance,<span class="at">x=</span>comparison.type))<span class="sc">+</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"genetic distance comparison"</span>)<span class="sc">+</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"genetic distance"</span>)</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co">#test whether there is any significant difference</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co">#between the Distance within vs between countries</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(Distance <span class="sc">~</span> comparison.type, <span class="at">data =</span> tas)</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Now go back to the exercises, to think about this data.</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="re">END</span><span class="co"> ---</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="interpreting-the-tasman-sea-data" class="level2" data-number="3.9">
<h2 data-number="3.9" class="anchored" data-anchor-id="interpreting-the-tasman-sea-data"><span class="header-section-number">3.9</span> Interpreting the Tasman sea data</h2>
<ul>
<li>Do genetic distances differ very much within vs between countries in the box plot?</li>
<li>What would you expect to see if there was one lineage of influenza viruses circulating in Australia and another lineage circulating only in New Zealand?</li>
<li>What would you expect to see if influenza viruses were spread freely between Australia and New Zealand?</li>
</ul>
</section>
</section>
<section id="summary-what-we-have-learned" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Summary: what we have learned</h1>
<ul>
<li>Phylogenetic trees can be used to visualise much more than merely evolutionary relationships between species</li>
<li>We used the phylogenetic tree to examine whether mutation accumulation in the influenza virus is clock-like</li>
<li>There seems to be a considerable amounts of evolutionary change occurring within the influenza viruses over just a few decades</li>
<li>This change is approximately clock-like, with a roughly constant rate of mutation accumulation over time</li>
<li>We can answer other questions about the spread of viruses using genetic data, such as whether influenza viruses spread freely between one place (Australia) and another (New Zealand)</li>
</ul>
</section>
<section id="exam-style-questions" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Exam style questions</h1>
<section id="question-1-10-points" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="question-1-10-points"><span class="header-section-number">5.1</span> Question 1 (10 points)</h2>
<p>How could we use the molecular clock to understand the spread of a plant pathogenic virus? In this scenario, we have access to infected crops from throughout the United Kingdom, sampled from different fields over a period of 10 years, and if we invest more funds we can access samples from up to 30 years ago. We are interested to know how to control the virus.</p>
<p>In your answer, explain:</p>
<ul>
<li>What <em>useful</em> information we could gather from the phylogeny and molecular clock analyses? (4 points)</li>
<li>How we would collect samples so that we could obtain this information? (4 points)</li>
<li>What analysis we would perform? (2 points)</li>
</ul>
</section>
<section id="question-2-10-points" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="question-2-10-points"><span class="header-section-number">5.2</span> Question 2 (10 points)</h2>
<p>Evidence and implications of the molecular clock (10 points)</p>
<ul>
<li>What aspects of RNA viruses make them so suitable for observing the molecular clock ? (3 points)</li>
<li>Give an example of another sitation in a multicellular species (plant, animal or fungi), where we might be able to observe the molecular clock over a longer period of time (e.g.&nbsp;decades, or centuries). (5 points)</li>
<li>Now consider the genetic change between the multicellular species, once they that have been diverging for millions of years. Assuming that you showed the genetic change eas clock-like, what does this tell you about the effects of mutations? (2 points)</li>
</ul>
</section>
</section>
<section id="further-reading" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Further reading</h1>
<p>These two articles will be useful for those that want to extend their knowledge.</p>
<ul>
<li>Baum (2005): <a href="https://www.science.org/doi/10.1126/science.1117727">The Tree-Thinking Challenge</a></li>
<li>Shao (2017): <a href="https://www.mdpi.com/1422-0067/18/8/1650">Evolution of Influenza A Virus by Mutation and Re-Assortment</a></li>
</ul>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Acknowledegment
</div>
</div>
<div class="callout-body-container callout-body">
<p>This workshop was designed and shared by <a href="https://en.wikipedia.org/wiki/Fran%C3%A7ois_Balloux">François Balloux</a>.</p>
</div>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb5" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "BIO00056I"</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Workshop 5: Phylogenies &amp; Molecular Clock"</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> Daniel Jeffares</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2025-02-24</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">    html:</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co">      </span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co">        code-copy: true</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="co">        code-tools: true</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="co">        code-link: true</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="co">        toc: true</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="co">        number-sections: true</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="co">        toc-location: right</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="co">        toc-depth: 2</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="co">        highlight-style: github</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> source</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a><span class="co">    </span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>:::callout-important</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>We do not advise doing this workshop on a Mac since Mega software does not work very well on the Mac OS.</span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>It is best to use the University PCs which have the software already installed.</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a><span class="fu"># Learning objectives</span></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a>The aim of this practical is to learn to visualise and interpret the data using phylogenetic analyses. You will see that we can extract a lot of information about evolutionary processes and populations from a phylogeny.</span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>By the end of this workshop, you should be able to:</span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a><span class="ss">*   </span>Build phylogenetic trees using using MEGA software</span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a><span class="ss">*   </span>Infer evolutionary relationships based on phylogeny</span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a><span class="ss">*   </span>Correlate molecular evolutionary changes with time using R</span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>:::callout-note</span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a><span class="fu"># Glossary</span></span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a>Technical definitions for this workshop.</span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>**molecular clock**: </span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a>The observation that mutations accumulate at a roughly constant rate over time.</span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>**node** (of a phylogenetic tree):</span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a>A point where branches split, representing either a common ancestor (internal node) or a modern species (terminal node).</span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>**branch** (of a phylogenetic tree):</span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a>A line connecting nodes, representing an evolutionary lineage. Branch length indicates the amount of evolutionary change. Due to the molecular clock hypothesis, branch lengths can be proportional to time.</span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>**mutation rate**:</span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a>The frequency at which genetic changes occur in DNA over time, typically measured as substitutions per site per year.</span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>**purifying selection**:</span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a>Natural selection that removes harmful mutations from a population, acting to preserve functional DNA sequences.</span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a><span class="fu"># Introduction</span></span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a><span class="fu">##  Phylogenetic trees and the molecular clock </span></span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a>Much of what we can infer from phylogenies uses the model of the 'molecular clock'. This is the observation that there tends to be a uniform, 'clock-like' rate of genetic change per year both *within* and *between* species.  The *rate* of change differs between very different species and between genes. </span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-69"><a href="#cb5-69" aria-hidden="true" tabindex="-1"></a>Rates of change between species differ because:</span>
<span id="cb5-70"><a href="#cb5-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-71"><a href="#cb5-71" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>small organisms (like viruses and bacteria) have short generation times</span>
<span id="cb5-72"><a href="#cb5-72" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>small organisms have high mutation rates</span>
<span id="cb5-73"><a href="#cb5-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-74"><a href="#cb5-74" aria-hidden="true" tabindex="-1"></a>Rates of change between genes within a genome differ because the amount of *purifying selection* acting on a gene differs.</span>
<span id="cb5-75"><a href="#cb5-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-76"><a href="#cb5-76" aria-hidden="true" tabindex="-1"></a>You can refresh your knowledge of Phylogenetic trees and the molecular clock from these two recordings:</span>
<span id="cb5-77"><a href="#cb5-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-78"><a href="#cb5-78" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="co">[</span><span class="ot">Phylogenies and molecular clock Part 1</span><span class="co">](https://york.cloud.panopto.eu/Panopto/Pages/Viewer.aspx?id=327fb8cc-9f70-4207-8afc-b0b2012cd0d4)</span> </span>
<span id="cb5-79"><a href="#cb5-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-80"><a href="#cb5-80" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="co">[</span><span class="ot">Phylogenies and molecular clock Part 2</span><span class="co">](https://york.cloud.panopto.eu/Panopto/Pages/Viewer.aspx?id=22c2aa19-d1eb-4819-88d0-b0b2013211ee)</span> </span>
<span id="cb5-81"><a href="#cb5-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-82"><a href="#cb5-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-83"><a href="#cb5-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-84"><a href="#cb5-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-85"><a href="#cb5-85" aria-hidden="true" tabindex="-1"></a><span class="fu">## The molecular clock and the real world</span></span>
<span id="cb5-86"><a href="#cb5-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-87"><a href="#cb5-87" aria-hidden="true" tabindex="-1"></a>Evolutionary thinking can also help us to understand disease. For example, the global pandemic of the coronavirus SARS-Cov2 (which causes COVID-19 disease) is an RNA virus. As with all viruses, SARS-Cov2 is constantly undergoing evolutionary change.</span>
<span id="cb5-88"><a href="#cb5-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-89"><a href="#cb5-89" aria-hidden="true" tabindex="-1"></a>The initial outbreak of the SARS-Cov2 virus in Wuhan China, was dated with amazing accuracy (to the month) to August 2019 <span class="co">[</span><span class="ot">using a time tree</span><span class="co">](https://doi.org/10.3390/v13091790)</span>. Being able to tell when outbreaks occur, as well as where, can be a very powerful tool as it can supplement what we can see from epidemiology. Analysis of DNA sequences is the only way we can obtain this information.</span>
<span id="cb5-90"><a href="#cb5-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-91"><a href="#cb5-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-92"><a href="#cb5-92" aria-hidden="true" tabindex="-1"></a><span class="fu"># Exercises</span></span>
<span id="cb5-93"><a href="#cb5-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-94"><a href="#cb5-94" aria-hidden="true" tabindex="-1"></a>In this workshop, we will study the evolution of the influenza virus. Influenza is an RNA virus that causes seasonal epidemics and occasional pandemics in humans. The influenza virus evolves very rapidly, which is why we need a new flu vaccine every year.</span>
<span id="cb5-95"><a href="#cb5-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-96"><a href="#cb5-96" aria-hidden="true" tabindex="-1"></a><span class="fu">## Obtaining data</span></span>
<span id="cb5-97"><a href="#cb5-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-98"><a href="#cb5-98" aria-hidden="true" tabindex="-1"></a>There is a wealth of freely available sequence data available on GenBank. The downside is that most data is often poorly annotated and misses the critical associated information (e.g. date and place of collection). Here we will take advantage of the <span class="co">[</span><span class="ot">influenza virus resource database</span><span class="co">](https://www.ncbi.nlm.nih.gov/genomes/FLU/Database/nph-select.cgi?go=database)</span>, which is relatively well curated.</span>
<span id="cb5-99"><a href="#cb5-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-100"><a href="#cb5-100" aria-hidden="true" tabindex="-1"></a>We have downloaded all the human full-length H3N2 type sequences from the hemagglutinin (HA) sequences from Wellington (New Zealand). Hemagglutinin (HA) a glycoprotein gene that is on the influenza virus surface.</span>
<span id="cb5-101"><a href="#cb5-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-102"><a href="#cb5-102" aria-hidden="true" tabindex="-1"></a>You can <span class="co">[</span><span class="ot">download the file H3N2_wellington.fasta here</span><span class="co">](https://drive.google.com/file/d/1cv4OAmLBRRad_7ObJrlljysWKO9KhX8l/view?usp=sharing)</span>.  This file has HA sequences from 72 influenza strains from 1985 to 2005. Sequences are already aligned, and sorted by date.</span>
<span id="cb5-103"><a href="#cb5-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-104"><a href="#cb5-104" aria-hidden="true" tabindex="-1"></a><span class="fu">## Visualising and analysing data</span></span>
<span id="cb5-105"><a href="#cb5-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-106"><a href="#cb5-106" aria-hidden="true" tabindex="-1"></a><span class="fu">## Starting analysis in Mega</span></span>
<span id="cb5-107"><a href="#cb5-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-108"><a href="#cb5-108" aria-hidden="true" tabindex="-1"></a>We will use the Mega software for all analyses. It should be installed on your computers. If you would like to use your own computer, you can download it from <span class="co">[</span><span class="ot">the mega wesbsite</span><span class="co">](http://www.megasoftware.net/)</span>. Use the latest version, Mega 11. </span>
<span id="cb5-109"><a href="#cb5-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-110"><a href="#cb5-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-111"><a href="#cb5-111" aria-hidden="true" tabindex="-1"></a>:::callout-important</span>
<span id="cb5-112"><a href="#cb5-112" aria-hidden="true" tabindex="-1"></a>Mega software does not work very well on the Mac OS.</span>
<span id="cb5-113"><a href="#cb5-113" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb5-114"><a href="#cb5-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-115"><a href="#cb5-115" aria-hidden="true" tabindex="-1"></a>Once you start Mega, you will see the following window. Be careful here, because Mega has multiple windows.</span>
<span id="cb5-116"><a href="#cb5-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-117"><a href="#cb5-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-118"><a href="#cb5-118" aria-hidden="true" tabindex="-1"></a><span class="al">![](mega1.jpg)</span></span>
<span id="cb5-119"><a href="#cb5-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-120"><a href="#cb5-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-121"><a href="#cb5-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-122"><a href="#cb5-122" aria-hidden="true" tabindex="-1"></a>Next, load the H3N2_Wellington_seq.fa file through the <span class="in">`Data &gt; Open a File/Session Menu`</span> (Ctrl+O). </span>
<span id="cb5-123"><a href="#cb5-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-124"><a href="#cb5-124" aria-hidden="true" tabindex="-1"></a>Then click:</span>
<span id="cb5-125"><a href="#cb5-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-126"><a href="#cb5-126" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="in">`Analyze`</span> (not align)</span>
<span id="cb5-127"><a href="#cb5-127" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>OK to <span class="in">`Nucleotide Sequences`</span></span>
<span id="cb5-128"><a href="#cb5-128" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Yes to <span class="in">`Protein-coding nucleotide sequence data`</span></span>
<span id="cb5-129"><a href="#cb5-129" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>OK to <span class="in">`Standard Genetic Code`</span>. </span>
<span id="cb5-130"><a href="#cb5-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-131"><a href="#cb5-131" aria-hidden="true" tabindex="-1"></a>You shuld see the window shown below</span>
<span id="cb5-132"><a href="#cb5-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-133"><a href="#cb5-133" aria-hidden="true" tabindex="-1"></a><span class="al">![](mega2.jpg)</span></span>
<span id="cb5-134"><a href="#cb5-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-135"><a href="#cb5-135" aria-hidden="true" tabindex="-1"></a><span class="fu">## Group the strains by year.</span></span>
<span id="cb5-136"><a href="#cb5-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-137"><a href="#cb5-137" aria-hidden="true" tabindex="-1"></a>Click on the <span class="in">`TA`</span> window at top left to see the Sequence Data Explorer. </span>
<span id="cb5-138"><a href="#cb5-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-139"><a href="#cb5-139" aria-hidden="true" tabindex="-1"></a>You can scroll around and resize this window. This data file comprises 72 complete HA</span>
<span id="cb5-140"><a href="#cb5-140" aria-hidden="true" tabindex="-1"></a>sequences from influenza H3N2 sampled between 1985 to 2005.</span>
<span id="cb5-141"><a href="#cb5-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-142"><a href="#cb5-142" aria-hidden="true" tabindex="-1"></a>Now let’s group the strains by year. This will be fairly easy because we sorted the</span>
<span id="cb5-143"><a href="#cb5-143" aria-hidden="true" tabindex="-1"></a>data by year before we downloaded it. Samples have the year embedded in the</span>
<span id="cb5-144"><a href="#cb5-144" aria-hidden="true" tabindex="-1"></a>name. </span>
<span id="cb5-145"><a href="#cb5-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-146"><a href="#cb5-146" aria-hidden="true" tabindex="-1"></a>For example:</span>
<span id="cb5-147"><a href="#cb5-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-148"><a href="#cb5-148" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>this one is from 1985: <span class="in">`Influenza A virus (A/Wellington/4/1985(H3N2))`</span> </span>
<span id="cb5-149"><a href="#cb5-149" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>and this from 2004: <span class="in">`Influenza A virus (A/Wellington/34/2004(H3N2))`</span></span>
<span id="cb5-150"><a href="#cb5-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-151"><a href="#cb5-151" aria-hidden="true" tabindex="-1"></a>First, go to the <span class="in">`Display`</span> menu, and click on <span class="in">`Show Group Names`</span>. This will help you to keep track of what you‘ve done.</span>
<span id="cb5-152"><a href="#cb5-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-153"><a href="#cb5-153" aria-hidden="true" tabindex="-1"></a><span class="al">![](mega3.jpg)</span>{width="300"}</span>
<span id="cb5-154"><a href="#cb5-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-155"><a href="#cb5-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-156"><a href="#cb5-156" aria-hidden="true" tabindex="-1"></a>To group strains, select one or more strains from a year by clicking on the name of the first strain in the grey part of the window, and shift+click to select the last one, as below:</span>
<span id="cb5-157"><a href="#cb5-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-158"><a href="#cb5-158" aria-hidden="true" tabindex="-1"></a>Then, go to the <span class="in">`Edit`</span> menu, and click on <span class="in">`Create Group from Selection`</span>. A new group will appear in the left-hand panel. Rename it to the year of sampling (e.g. 1985). </span>
<span id="cb5-159"><a href="#cb5-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-160"><a href="#cb5-160" aria-hidden="true" tabindex="-1"></a>Repeat this for all years from 1985 to 2005.</span>
<span id="cb5-161"><a href="#cb5-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-162"><a href="#cb5-162" aria-hidden="true" tabindex="-1"></a><span class="al">![](mega4.jpg)</span>{width="600"}</span>
<span id="cb5-163"><a href="#cb5-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-164"><a href="#cb5-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-165"><a href="#cb5-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-166"><a href="#cb5-166" aria-hidden="true" tabindex="-1"></a><span class="fu">## Building a phylogenetic tree</span></span>
<span id="cb5-167"><a href="#cb5-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-168"><a href="#cb5-168" aria-hidden="true" tabindex="-1"></a>As you will have seen, there are a large number of possible options for analysis in MEGA. However, we will build one tree along a single, robust methodology. </span>
<span id="cb5-169"><a href="#cb5-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-170"><a href="#cb5-170" aria-hidden="true" tabindex="-1"></a>Go to the <span class="in">`Phylogeny`</span> menu on the main menu: </span>
<span id="cb5-171"><a href="#cb5-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-172"><a href="#cb5-172" aria-hidden="true" tabindex="-1"></a><span class="in">`Phylogeny -&gt; Construct/Test Neighbor-Joining`</span> </span>
<span id="cb5-173"><a href="#cb5-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-174"><a href="#cb5-174" aria-hidden="true" tabindex="-1"></a>Choose these options:</span>
<span id="cb5-175"><a href="#cb5-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-176"><a href="#cb5-176" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>For Model/Method, choose the <span class="in">`Kimura 2-parameter model`</span> of evolution</span>
<span id="cb5-177"><a href="#cb5-177" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>For Test of Phylogeny, choose <span class="in">`Bootstrap method`</span> and select 1000 bootstraps.</span>
<span id="cb5-178"><a href="#cb5-178" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Leave all other options as default</span>
<span id="cb5-179"><a href="#cb5-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-180"><a href="#cb5-180" aria-hidden="true" tabindex="-1"></a>Then click OK. </span>
<span id="cb5-181"><a href="#cb5-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-182"><a href="#cb5-182" aria-hidden="true" tabindex="-1"></a>**Bootstraps** represent statistical support for individual clades, values in excess of 70-80% are considered well supported. </span>
<span id="cb5-183"><a href="#cb5-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-184"><a href="#cb5-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-185"><a href="#cb5-185" aria-hidden="true" tabindex="-1"></a>After less than a minute, you will get a phylogenetic tree, a new Mega window. </span>
<span id="cb5-186"><a href="#cb5-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-187"><a href="#cb5-187" aria-hidden="true" tabindex="-1"></a>It does not look very informative, so change the presentation of the tree using the menus on the left:</span>
<span id="cb5-188"><a href="#cb5-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-189"><a href="#cb5-189" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>First, deselect <span class="in">`Taxon Names`</span></span>
<span id="cb5-190"><a href="#cb5-190" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Under Layout, select <span class="in">`Auto-size tree`</span></span>
<span id="cb5-191"><a href="#cb5-191" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Define a root. The branch leading to the sequence from 1985 makes a biologically reasonable root. To define a root, simply right click with your mouse on the chosen branch and select <span class="in">`Place root`</span>. </span>
<span id="cb5-192"><a href="#cb5-192" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>You may also need to click toggle <span class="in">`Scaling`</span> of the tree in the <span class="in">`Layout`</span> menu.</span>
<span id="cb5-193"><a href="#cb5-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-194"><a href="#cb5-194" aria-hidden="true" tabindex="-1"></a>You should see a tree like the one shown below.</span>
<span id="cb5-195"><a href="#cb5-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-196"><a href="#cb5-196" aria-hidden="true" tabindex="-1"></a><span class="al">![](mega5.jpg)</span>{width="700"}</span>
<span id="cb5-197"><a href="#cb5-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-198"><a href="#cb5-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-199"><a href="#cb5-199" aria-hidden="true" tabindex="-1"></a><span class="fu">### Understanding the tree</span></span>
<span id="cb5-200"><a href="#cb5-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-201"><a href="#cb5-201" aria-hidden="true" tabindex="-1"></a>What do you notice about the clustering of years on this phylogeny? </span>
<span id="cb5-202"><a href="#cb5-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-203"><a href="#cb5-203" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>In particular, look at the branch lengths (ie: genetic distance) from the root of the tree, to the tips. </span>
<span id="cb5-204"><a href="#cb5-204" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>This indicates how many mutations have occurred since the common ancestor.</span>
<span id="cb5-205"><a href="#cb5-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-206"><a href="#cb5-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-207"><a href="#cb5-207" aria-hidden="true" tabindex="-1"></a><span class="fu">## Measuring molecular change over time</span></span>
<span id="cb5-208"><a href="#cb5-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-209"><a href="#cb5-209" aria-hidden="true" tabindex="-1"></a>An important question in phylogenetics is whether the accumulation of mutation (substitution rate) is constant over time. One way to test for this would be to count the number of mutations from the root to each tip. Estimating the most likely root is not completely straightforward and counting the number of mutations from root to tip would require some scripting well beyond the scope of this practical. However, we can test whether genetic distances increase linearly with time.</span>
<span id="cb5-210"><a href="#cb5-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-211"><a href="#cb5-211" aria-hidden="true" tabindex="-1"></a>To address the issue of linearity between substitution rates and time, we can first compute genetic distances between the sequences grouped by years. </span>
<span id="cb5-212"><a href="#cb5-212" aria-hidden="true" tabindex="-1"></a>To do this:</span>
<span id="cb5-213"><a href="#cb5-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-214"><a href="#cb5-214" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Go to the <span class="in">`Distance`</span> menu (at the top of the main Mega window)</span>
<span id="cb5-215"><a href="#cb5-215" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Click <span class="in">`Compute Between Group Mean Distance`</span> (see image below) </span>
<span id="cb5-216"><a href="#cb5-216" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Keep the <span class="in">`Kimura 2 parameter model`</span></span>
<span id="cb5-217"><a href="#cb5-217" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Click <span class="in">`OK`</span></span>
<span id="cb5-218"><a href="#cb5-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-219"><a href="#cb5-219" aria-hidden="true" tabindex="-1"></a><span class="al">![](mega6.jpg)</span>{width="400"}</span>
<span id="cb5-220"><a href="#cb5-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-221"><a href="#cb5-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-222"><a href="#cb5-222" aria-hidden="true" tabindex="-1"></a>You should see the matrix shown below.</span>
<span id="cb5-223"><a href="#cb5-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-224"><a href="#cb5-224" aria-hidden="true" tabindex="-1"></a><span class="al">![](mega7.jpg)</span>{width="600"}</span>
<span id="cb5-225"><a href="#cb5-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-226"><a href="#cb5-226" aria-hidden="true" tabindex="-1"></a>Next, export this data to a file:</span>
<span id="cb5-227"><a href="#cb5-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-228"><a href="#cb5-228" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>In the matrix window, click on the <span class="in">`File`</span> menu</span>
<span id="cb5-229"><a href="#cb5-229" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Then <span class="in">`Export/Print Distances`</span></span>
<span id="cb5-230"><a href="#cb5-230" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Output format: <span class="in">`text`</span></span>
<span id="cb5-231"><a href="#cb5-231" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Export Type: <span class="in">`column`</span></span>
<span id="cb5-232"><a href="#cb5-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-233"><a href="#cb5-233" aria-hidden="true" tabindex="-1"></a>You should see a window something like the image below.</span>
<span id="cb5-234"><a href="#cb5-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-235"><a href="#cb5-235" aria-hidden="true" tabindex="-1"></a><span class="al">![](mega8.jpg)</span>{width="400"}</span>
<span id="cb5-236"><a href="#cb5-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-237"><a href="#cb5-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-238"><a href="#cb5-238" aria-hidden="true" tabindex="-1"></a>Finally edit and save this file:</span>
<span id="cb5-239"><a href="#cb5-239" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Remove all the text above Species1 Species2 Distance</span>
<span id="cb5-240"><a href="#cb5-240" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Remove the text below the data table, "Table. Estimates of Evolutionary Divergence over Sequence ..." etc</span>
<span id="cb5-241"><a href="#cb5-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-242"><a href="#cb5-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-243"><a href="#cb5-243" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Edit the <span class="in">`Species1 Species2 Distance`</span> text to <span class="in">`date1 date1 Distance`</span> </span>
<span id="cb5-244"><a href="#cb5-244" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Choose <span class="in">`File &gt; Save As`</span></span>
<span id="cb5-245"><a href="#cb5-245" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Name the file <span class="in">`H3N2_wellington.distances.txt`</span></span>
<span id="cb5-246"><a href="#cb5-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-247"><a href="#cb5-247" aria-hidden="true" tabindex="-1"></a>You can also <span class="co">[</span><span class="ot">obtain a copy of this file here</span><span class="co">](https://drive.google.com/file/d/1Um4R5L_BeKCy9qzdTM10zbyuGi50a_o6/view?usp=sharing)</span>.</span>
<span id="cb5-248"><a href="#cb5-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-249"><a href="#cb5-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-250"><a href="#cb5-250" aria-hidden="true" tabindex="-1"></a><span class="fu">## Analysis of genetic change with time</span></span>
<span id="cb5-251"><a href="#cb5-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-252"><a href="#cb5-252" aria-hidden="true" tabindex="-1"></a>The remainder of this part of the workshop will use R Studio, so start RStudio now.</span>
<span id="cb5-253"><a href="#cb5-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-254"><a href="#cb5-254" aria-hidden="true" tabindex="-1"></a>All the commands you’ll need are in the R Script <span class="in">`BIO00056I-influenza-practical-2024.R`</span>, which is <span class="co">[</span><span class="ot">available to download here</span><span class="co">](https://drive.google.com/file/d/1j-9ktSYBZiPBxiwZpy8QaBWX7aqA54c8/view?usp=sharing)</span>.</span>
<span id="cb5-255"><a href="#cb5-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-256"><a href="#cb5-256" aria-hidden="true" tabindex="-1"></a>Or you can create you own script, and copy the comnands from here as we go along.</span>
<span id="cb5-257"><a href="#cb5-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-258"><a href="#cb5-258" aria-hidden="true" tabindex="-1"></a><span class="fu">### Set up the script</span></span>
<span id="cb5-259"><a href="#cb5-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-260"><a href="#cb5-260" aria-hidden="true" tabindex="-1"></a>First, set up your script:</span>
<span id="cb5-261"><a href="#cb5-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-264"><a href="#cb5-264" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb5-265"><a href="#cb5-265" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb5-266"><a href="#cb5-266" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb5-267"><a href="#cb5-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-268"><a href="#cb5-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-269"><a href="#cb5-269" aria-hidden="true" tabindex="-1"></a><span class="co">#BIO00056I Influenza Virus Practical ----</span></span>
<span id="cb5-270"><a href="#cb5-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-271"><a href="#cb5-271" aria-hidden="true" tabindex="-1"></a><span class="co"># SET UP ----</span></span>
<span id="cb5-272"><a href="#cb5-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-273"><a href="#cb5-273" aria-hidden="true" tabindex="-1"></a><span class="co">#set your working directory</span></span>
<span id="cb5-274"><a href="#cb5-274" aria-hidden="true" tabindex="-1"></a><span class="co">#Your working directory will be different!</span></span>
<span id="cb5-275"><a href="#cb5-275" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"~/path/to/your/files/"</span>)</span>
<span id="cb5-276"><a href="#cb5-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-277"><a href="#cb5-277" aria-hidden="true" tabindex="-1"></a><span class="co">#You can use the Session menu to set this with:</span></span>
<span id="cb5-278"><a href="#cb5-278" aria-hidden="true" tabindex="-1"></a><span class="co">#Session menu  &gt; Set Working Directory &gt; Choose Directory</span></span>
<span id="cb5-279"><a href="#cb5-279" aria-hidden="true" tabindex="-1"></a><span class="co">#then select the directory where your files are.</span></span>
<span id="cb5-280"><a href="#cb5-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-281"><a href="#cb5-281" aria-hidden="true" tabindex="-1"></a><span class="co">#load the tidyverse</span></span>
<span id="cb5-282"><a href="#cb5-282" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb5-283"><a href="#cb5-283" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb5-284"><a href="#cb5-284" aria-hidden="true" tabindex="-1"></a>\</span>
<span id="cb5-285"><a href="#cb5-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-286"><a href="#cb5-286" aria-hidden="true" tabindex="-1"></a><span class="fu">### Loading and plotting the data</span></span>
<span id="cb5-287"><a href="#cb5-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-288"><a href="#cb5-288" aria-hidden="true" tabindex="-1"></a>Then load the Wellgonton data, and make a plot</span>
<span id="cb5-289"><a href="#cb5-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-290"><a href="#cb5-290" aria-hidden="true" tabindex="-1"></a>\ </span>
<span id="cb5-291"><a href="#cb5-291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-294"><a href="#cb5-294" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb5-295"><a href="#cb5-295" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb5-296"><a href="#cb5-296" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb5-297"><a href="#cb5-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-298"><a href="#cb5-298" aria-hidden="true" tabindex="-1"></a><span class="co"># WELLINGTON DATA ----</span></span>
<span id="cb5-299"><a href="#cb5-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-300"><a href="#cb5-300" aria-hidden="true" tabindex="-1"></a><span class="co">#open your distances file, into a data frame:</span></span>
<span id="cb5-301"><a href="#cb5-301" aria-hidden="true" tabindex="-1"></a><span class="co">#Your distance file may have a different name</span></span>
<span id="cb5-302"><a href="#cb5-302" aria-hidden="true" tabindex="-1"></a>flu <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(<span class="fu">read.table</span>(<span class="st">"H3N2_wellington.distances.txt"</span>,<span class="at">h=</span>T))</span>
<span id="cb5-303"><a href="#cb5-303" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-304"><a href="#cb5-304" aria-hidden="true" tabindex="-1"></a><span class="co">#calculate how many years between two groups</span></span>
<span id="cb5-305"><a href="#cb5-305" aria-hidden="true" tabindex="-1"></a>flu<span class="sc">$</span>time.passed <span class="ot">&lt;-</span> flu<span class="sc">$</span>date2 <span class="sc">-</span> flu<span class="sc">$</span>date1</span>
<span id="cb5-306"><a href="#cb5-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-307"><a href="#cb5-307" aria-hidden="true" tabindex="-1"></a><span class="co">#see if distance increases with time, as mutations accumulate</span></span>
<span id="cb5-308"><a href="#cb5-308" aria-hidden="true" tabindex="-1"></a><span class="co">#make the plot</span></span>
<span id="cb5-309"><a href="#cb5-309" aria-hidden="true" tabindex="-1"></a>flu_plot<span class="ot">&lt;-</span><span class="fu">ggplot</span>(flu,<span class="fu">aes</span>(<span class="at">x=</span>time.passed, <span class="at">y=</span> Distance))<span class="sc">+</span></span>
<span id="cb5-310"><a href="#cb5-310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb5-311"><a href="#cb5-311" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"time distance between groups (years)"</span>)<span class="sc">+</span></span>
<span id="cb5-312"><a href="#cb5-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"genetic distance between groups"</span>)<span class="sc">+</span></span>
<span id="cb5-313"><a href="#cb5-313" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb5-314"><a href="#cb5-314" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-315"><a href="#cb5-315" aria-hidden="true" tabindex="-1"></a><span class="co">#view the plot</span></span>
<span id="cb5-316"><a href="#cb5-316" aria-hidden="true" tabindex="-1"></a>flu_plot </span>
<span id="cb5-317"><a href="#cb5-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-318"><a href="#cb5-318" aria-hidden="true" tabindex="-1"></a><span class="co">#now add a line of best fit:</span></span>
<span id="cb5-319"><a href="#cb5-319" aria-hidden="true" tabindex="-1"></a>flu_plot <span class="sc">+</span></span>
<span id="cb5-320"><a href="#cb5-320" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>)</span>
<span id="cb5-321"><a href="#cb5-321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-322"><a href="#cb5-322" aria-hidden="true" tabindex="-1"></a><span class="co">#examine whether the correlation is statistically significant: </span></span>
<span id="cb5-323"><a href="#cb5-323" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(flu<span class="sc">$</span>time.passed,flu<span class="sc">$</span>Distance)</span>
<span id="cb5-324"><a href="#cb5-324" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb5-325"><a href="#cb5-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-326"><a href="#cb5-326" aria-hidden="true" tabindex="-1"></a><span class="fu">### Interpret the data</span></span>
<span id="cb5-327"><a href="#cb5-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-328"><a href="#cb5-328" aria-hidden="true" tabindex="-1"></a>You should see a good correlation between genetic distance and time distance.</span>
<span id="cb5-329"><a href="#cb5-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-330"><a href="#cb5-330" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>What causes this? (which process of evolution) </span>
<span id="cb5-331"><a href="#cb5-331" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Why is there such a strong correlation?</span>
<span id="cb5-332"><a href="#cb5-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-333"><a href="#cb5-333" aria-hidden="true" tabindex="-1"></a>We can observe this relationship over just a few decades?</span>
<span id="cb5-334"><a href="#cb5-334" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>What does this tell us about the rate of change?</span>
<span id="cb5-335"><a href="#cb5-335" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-336"><a href="#cb5-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-337"><a href="#cb5-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-338"><a href="#cb5-338" aria-hidden="true" tabindex="-1"></a><span class="fu">## Optional: Does the influenza virus spread across the Tasman Sea?</span></span>
<span id="cb5-339"><a href="#cb5-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-340"><a href="#cb5-340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-341"><a href="#cb5-341" aria-hidden="true" tabindex="-1"></a>Another use of virus genetic data is to determine how fast pathogens spread.</span>
<span id="cb5-342"><a href="#cb5-342" aria-hidden="true" tabindex="-1"></a>Here, we’ll show an example, where we examine whether people in New Zealand</span>
<span id="cb5-343"><a href="#cb5-343" aria-hidden="true" tabindex="-1"></a>share the same strains as their neighbors in Australia across the Tasman Sea.</span>
<span id="cb5-344"><a href="#cb5-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-345"><a href="#cb5-345" aria-hidden="true" tabindex="-1"></a>We have downloaded and aligned 118 strains from Australia and New Zealand. You</span>
<span id="cb5-346"><a href="#cb5-346" aria-hidden="true" tabindex="-1"></a>can download the file H3N2_tasman.fasta here. These sequences are already</span>
<span id="cb5-347"><a href="#cb5-347" aria-hidden="true" tabindex="-1"></a>assigned.</span>
<span id="cb5-348"><a href="#cb5-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-349"><a href="#cb5-349" aria-hidden="true" tabindex="-1"></a>Then open MEGA, selecting <span class="in">`Analyse`</span>, and</span>
<span id="cb5-350"><a href="#cb5-350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-351"><a href="#cb5-351" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Select the Distance menu, and Compute Pairwise Distances.</span>
<span id="cb5-352"><a href="#cb5-352" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>You will see a matrix of distances</span>
<span id="cb5-353"><a href="#cb5-353" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>In this window, select the <span class="in">`File`</span> menu, and:</span>
<span id="cb5-354"><a href="#cb5-354" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>Export/Print Distances</span>
<span id="cb5-355"><a href="#cb5-355" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>Output format: CSV</span>
<span id="cb5-356"><a href="#cb5-356" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>Export type: Column</span>
<span id="cb5-357"><a href="#cb5-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-358"><a href="#cb5-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-359"><a href="#cb5-359" aria-hidden="true" tabindex="-1"></a>As before remove all the text above and below the data:</span>
<span id="cb5-360"><a href="#cb5-360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-361"><a href="#cb5-361" aria-hidden="true" tabindex="-1"></a>Above:</span>
<span id="cb5-362"><a href="#cb5-362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-363"><a href="#cb5-363" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>Title: fasta file</span>
<span id="cb5-364"><a href="#cb5-364" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>Description:</span>
<span id="cb5-365"><a href="#cb5-365" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-366"><a href="#cb5-366" aria-hidden="true" tabindex="-1"></a>Below:</span>
<span id="cb5-367"><a href="#cb5-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-368"><a href="#cb5-368" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>Table. Estimates of Evolutionary Divergence between Sequences</span>
<span id="cb5-369"><a href="#cb5-369" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span>The number of base substitutions per site from between sequences are shown.</span>
<span id="cb5-370"><a href="#cb5-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-371"><a href="#cb5-371" aria-hidden="true" tabindex="-1"></a>Then edit the header line to be: <span class="in">`region1 region2 Distance`</span>, ensuring there is a space character between each column names.</span>
<span id="cb5-372"><a href="#cb5-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-373"><a href="#cb5-373" aria-hidden="true" tabindex="-1"></a>Finally, save the file as <span class="in">`H3N2_tasman.distances.csv`</span>. </span>
<span id="cb5-374"><a href="#cb5-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-375"><a href="#cb5-375" aria-hidden="true" tabindex="-1"></a>You can also obtain <span class="co">[</span><span class="ot">a copy of the H3N2_tasman.distances.csv file here</span><span class="co">](https://drive.google.com/file/d/15xnaP5Ju9oxfYEugDqtZ8YDsokYUumml/view?usp=sharing)</span>.</span>
<span id="cb5-376"><a href="#cb5-376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-377"><a href="#cb5-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-378"><a href="#cb5-378" aria-hidden="true" tabindex="-1"></a>Then examine the between-country differences and compare these to within-country</span>
<span id="cb5-379"><a href="#cb5-379" aria-hidden="true" tabindex="-1"></a>differences, using the code in the R script, as below:</span>
<span id="cb5-380"><a href="#cb5-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-383"><a href="#cb5-383" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb5-384"><a href="#cb5-384" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb5-385"><a href="#cb5-385" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb5-386"><a href="#cb5-386" aria-hidden="true" tabindex="-1"></a><span class="co">#| </span></span>
<span id="cb5-387"><a href="#cb5-387" aria-hidden="true" tabindex="-1"></a><span class="co"># TASMAN SEA DATA ----</span></span>
<span id="cb5-388"><a href="#cb5-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-389"><a href="#cb5-389" aria-hidden="true" tabindex="-1"></a><span class="co">#open the Tasman genetic distance data</span></span>
<span id="cb5-390"><a href="#cb5-390" aria-hidden="true" tabindex="-1"></a>tas <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"H3N2_tasman.distances.csv"</span>)</span>
<span id="cb5-391"><a href="#cb5-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-392"><a href="#cb5-392" aria-hidden="true" tabindex="-1"></a><span class="co">#add country1 and country2 columns</span></span>
<span id="cb5-393"><a href="#cb5-393" aria-hidden="true" tabindex="-1"></a>tas<span class="sc">$</span>country1 <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb5-394"><a href="#cb5-394" aria-hidden="true" tabindex="-1"></a>tas<span class="sc">$</span>country2 <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb5-395"><a href="#cb5-395" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-396"><a href="#cb5-396" aria-hidden="true" tabindex="-1"></a><span class="co">#label the countries using grep</span></span>
<span id="cb5-397"><a href="#cb5-397" aria-hidden="true" tabindex="-1"></a>tas[<span class="fu">grep</span>(<span class="st">"NZ"</span>, tas<span class="sc">$</span>region1),]<span class="sc">$</span>country1 <span class="ot">=</span> <span class="st">"NZ"</span></span>
<span id="cb5-398"><a href="#cb5-398" aria-hidden="true" tabindex="-1"></a>tas[<span class="fu">grep</span>(<span class="st">"AUS"</span>, tas<span class="sc">$</span>region1),]<span class="sc">$</span>country1 <span class="ot">=</span> <span class="st">"AUS"</span></span>
<span id="cb5-399"><a href="#cb5-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-400"><a href="#cb5-400" aria-hidden="true" tabindex="-1"></a>tas[<span class="fu">grep</span>(<span class="st">"NZ"</span>, tas<span class="sc">$</span>region2),]<span class="sc">$</span>country2 <span class="ot">=</span> <span class="st">"NZ"</span></span>
<span id="cb5-401"><a href="#cb5-401" aria-hidden="true" tabindex="-1"></a>tas[<span class="fu">grep</span>(<span class="st">"AUS"</span>, tas<span class="sc">$</span>region2),]<span class="sc">$</span>country2 <span class="ot">=</span> <span class="st">"AUS"</span></span>
<span id="cb5-402"><a href="#cb5-402" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-403"><a href="#cb5-403" aria-hidden="true" tabindex="-1"></a><span class="co">#add a comparison.type column</span></span>
<span id="cb5-404"><a href="#cb5-404" aria-hidden="true" tabindex="-1"></a>tas<span class="sc">$</span>comparison.type <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb5-405"><a href="#cb5-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-406"><a href="#cb5-406" aria-hidden="true" tabindex="-1"></a><span class="co">#mark the within country data</span></span>
<span id="cb5-407"><a href="#cb5-407" aria-hidden="true" tabindex="-1"></a>tas[<span class="fu">which</span>(tas<span class="sc">$</span>region1 <span class="sc">==</span> tas<span class="sc">$</span>region2),]<span class="sc">$</span>comparison.type <span class="ot">&lt;-</span> <span class="st">'within country'</span></span>
<span id="cb5-408"><a href="#cb5-408" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-409"><a href="#cb5-409" aria-hidden="true" tabindex="-1"></a><span class="co">#mark the between country data</span></span>
<span id="cb5-410"><a href="#cb5-410" aria-hidden="true" tabindex="-1"></a>tas[<span class="fu">which</span>(tas<span class="sc">$</span>region1 <span class="sc">!=</span> tas<span class="sc">$</span>region2),]<span class="sc">$</span>comparison.type <span class="ot">&lt;-</span> <span class="st">'between country'</span></span>
<span id="cb5-411"><a href="#cb5-411" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb5-412"><a href="#cb5-412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-413"><a href="#cb5-413" aria-hidden="true" tabindex="-1"></a>Now we make some plots:</span>
<span id="cb5-414"><a href="#cb5-414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-417"><a href="#cb5-417" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb5-418"><a href="#cb5-418" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb5-419"><a href="#cb5-419" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb5-420"><a href="#cb5-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-421"><a href="#cb5-421" aria-hidden="true" tabindex="-1"></a><span class="co">#plot this data </span></span>
<span id="cb5-422"><a href="#cb5-422" aria-hidden="true" tabindex="-1"></a><span class="co">#to examine whether genetic distances differ</span></span>
<span id="cb5-423"><a href="#cb5-423" aria-hidden="true" tabindex="-1"></a><span class="co">#within a country vs between countries</span></span>
<span id="cb5-424"><a href="#cb5-424" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(tas,<span class="fu">aes</span>(<span class="at">y=</span>Distance,<span class="at">x=</span>comparison.type))<span class="sc">+</span></span>
<span id="cb5-425"><a href="#cb5-425" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb5-426"><a href="#cb5-426" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"genetic distance comparison"</span>)<span class="sc">+</span></span>
<span id="cb5-427"><a href="#cb5-427" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"genetic distance"</span>)</span>
<span id="cb5-428"><a href="#cb5-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-429"><a href="#cb5-429" aria-hidden="true" tabindex="-1"></a><span class="co">#test whether there is any significant difference</span></span>
<span id="cb5-430"><a href="#cb5-430" aria-hidden="true" tabindex="-1"></a><span class="co">#between the Distance within vs between countries</span></span>
<span id="cb5-431"><a href="#cb5-431" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(Distance <span class="sc">~</span> comparison.type, <span class="at">data =</span> tas)</span>
<span id="cb5-432"><a href="#cb5-432" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-433"><a href="#cb5-433" aria-hidden="true" tabindex="-1"></a><span class="co">#Now go back to the exercises, to think about this data.</span></span>
<span id="cb5-434"><a href="#cb5-434" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-435"><a href="#cb5-435" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="re">END</span><span class="co"> ---</span></span>
<span id="cb5-436"><a href="#cb5-436" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb5-437"><a href="#cb5-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-438"><a href="#cb5-438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-439"><a href="#cb5-439" aria-hidden="true" tabindex="-1"></a><span class="fu">## Interpreting the Tasman sea data</span></span>
<span id="cb5-440"><a href="#cb5-440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-441"><a href="#cb5-441" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Do genetic distances differ very much within vs between countries in the box</span>
<span id="cb5-442"><a href="#cb5-442" aria-hidden="true" tabindex="-1"></a>plot?</span>
<span id="cb5-443"><a href="#cb5-443" aria-hidden="true" tabindex="-1"></a><span class="ss">-  </span>What would you expect to see if there was one lineage of influenza viruses circulating in Australia and another lineage circulating only in New Zealand?</span>
<span id="cb5-444"><a href="#cb5-444" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>What would you expect to see if influenza viruses were spread freely between Australia and New Zealand?</span>
<span id="cb5-445"><a href="#cb5-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-446"><a href="#cb5-446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-447"><a href="#cb5-447" aria-hidden="true" tabindex="-1"></a><span class="fu"># Summary: what we have learned</span></span>
<span id="cb5-448"><a href="#cb5-448" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-449"><a href="#cb5-449" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Phylogenetic trees can be used to visualise much more than merely evolutionary relationships between species</span>
<span id="cb5-450"><a href="#cb5-450" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>We used the phylogenetic tree to examine whether mutation accumulation in the influenza virus is clock-like</span>
<span id="cb5-451"><a href="#cb5-451" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>There seems to be a considerable amounts of evolutionary change occurring within the influenza viruses over just a few decades</span>
<span id="cb5-452"><a href="#cb5-452" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>This change is approximately clock-like, with a roughly constant rate of mutation accumulation over time</span>
<span id="cb5-453"><a href="#cb5-453" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>We can answer other questions about the spread of viruses using genetic data, such as whether influenza viruses spread freely between one place (Australia) and another (New Zealand)</span>
<span id="cb5-454"><a href="#cb5-454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-455"><a href="#cb5-455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-456"><a href="#cb5-456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-457"><a href="#cb5-457" aria-hidden="true" tabindex="-1"></a><span class="fu"># Exam style questions</span></span>
<span id="cb5-458"><a href="#cb5-458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-459"><a href="#cb5-459" aria-hidden="true" tabindex="-1"></a><span class="fu">## Question 1 (10 points)</span></span>
<span id="cb5-460"><a href="#cb5-460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-461"><a href="#cb5-461" aria-hidden="true" tabindex="-1"></a>How could we use the molecular clock to understand the spread of a plant pathogenic virus? In this scenario, we have access to infected crops from throughout the United Kingdom, sampled from different fields over a period of 10 years, and if we invest more funds we can access samples from up to 30 years ago. We are interested to know how to control the virus.</span>
<span id="cb5-462"><a href="#cb5-462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-463"><a href="#cb5-463" aria-hidden="true" tabindex="-1"></a>In your answer, explain:</span>
<span id="cb5-464"><a href="#cb5-464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-465"><a href="#cb5-465" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>What *useful* information we could gather from the phylogeny and molecular clock analyses? (4 points)</span>
<span id="cb5-466"><a href="#cb5-466" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>How we would collect samples so that we could obtain this information? (4 points)</span>
<span id="cb5-467"><a href="#cb5-467" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>What analysis we would perform? (2 points)</span>
<span id="cb5-468"><a href="#cb5-468" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-469"><a href="#cb5-469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-470"><a href="#cb5-470" aria-hidden="true" tabindex="-1"></a><span class="fu">## Question 2 (10 points)  </span></span>
<span id="cb5-471"><a href="#cb5-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-472"><a href="#cb5-472" aria-hidden="true" tabindex="-1"></a>Evidence and implications of the molecular clock (10 points)</span>
<span id="cb5-473"><a href="#cb5-473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-474"><a href="#cb5-474" aria-hidden="true" tabindex="-1"></a><span class="ss"> - </span>What aspects of RNA viruses make them so suitable for observing the molecular clock ? (3 points)</span>
<span id="cb5-475"><a href="#cb5-475" aria-hidden="true" tabindex="-1"></a><span class="ss"> - </span>Give an example of another sitation in a multicellular species (plant, animal or fungi), where we might be able to observe the molecular clock over a longer period of time (e.g. decades, or centuries). (5 points)</span>
<span id="cb5-476"><a href="#cb5-476" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Now consider the genetic change between the multicellular species, once they that have been diverging for millions of years. Assuming that you showed the genetic change eas clock-like, what does this tell you about the effects of mutations? (2 points)</span>
<span id="cb5-477"><a href="#cb5-477" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-478"><a href="#cb5-478" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-479"><a href="#cb5-479" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-480"><a href="#cb5-480" aria-hidden="true" tabindex="-1"></a><span class="fu"># Further reading</span></span>
<span id="cb5-481"><a href="#cb5-481" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-482"><a href="#cb5-482" aria-hidden="true" tabindex="-1"></a>These two articles will be useful for those that want to extend their knowledge.</span>
<span id="cb5-483"><a href="#cb5-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-484"><a href="#cb5-484" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Baum (2005): <span class="co">[</span><span class="ot">The Tree-Thinking Challenge</span><span class="co">](https://www.science.org/doi/10.1126/science.1117727)</span></span>
<span id="cb5-485"><a href="#cb5-485" aria-hidden="true" tabindex="-1"></a><span class="ss">*   </span>Shao (2017): <span class="co">[</span><span class="ot">Evolution of Influenza A Virus by Mutation and Re-Assortment</span><span class="co">](https://www.mdpi.com/1422-0067/18/8/1650)</span></span>
<span id="cb5-486"><a href="#cb5-486" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-487"><a href="#cb5-487" aria-hidden="true" tabindex="-1"></a>:::callout-note</span>
<span id="cb5-488"><a href="#cb5-488" aria-hidden="true" tabindex="-1"></a><span class="fu"># Acknowledegment</span></span>
<span id="cb5-489"><a href="#cb5-489" aria-hidden="true" tabindex="-1"></a>This workshop was designed and shared by <span class="co">[</span><span class="ot">François Balloux</span><span class="co">](https://en.wikipedia.org/wiki/Fran%C3%A7ois_Balloux)</span>.</span>
<span id="cb5-490"><a href="#cb5-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-491"><a href="#cb5-491" aria-hidden="true" tabindex="-1"></a>:::</span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>