<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.33">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Daniel Jeffares">
<meta name="dcterms.date" content="2025-11-04">

<title>BIO00056I GWAS Workshop – BIO00056I</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-6fbc3b7d0ac83ab8a043646d6c6d94a5.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-6f9ef9b5a12f5b8ef14302403eafb8d0.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">BIO00056I</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">BIO00056I</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./workshop5.html"> 
<span class="menu-text">workshop 5</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./workshop6.html"> 
<span class="menu-text">workshop 6</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./workshop7.html"> 
<span class="menu-text">workshop 7</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./workshop9.html" aria-current="page"> 
<span class="menu-text">workshop 9</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./worksheet10.html"> 
<span class="menu-text">worksheet 10</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./worksheet11.html"> 
<span class="menu-text">worksheet 11</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul class="collapse">
  <li><a href="#learning-objectives" id="toc-learning-objectives" class="nav-link active" data-scroll-target="#learning-objectives"><span class="header-section-number">1</span> Learning objectives</a></li>
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction"><span class="header-section-number">2</span> Introduction</a>
  <ul class="collapse">
  <li><a href="#background" id="toc-background" class="nav-link" data-scroll-target="#background"><span class="header-section-number">2.1</span> Background</a></li>
  <li><a href="#fission-yeast" id="toc-fission-yeast" class="nav-link" data-scroll-target="#fission-yeast"><span class="header-section-number">2.2</span> Fission yeast</a></li>
  </ul></li>
  </ul>
<div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="workshop9.pdf"><i class="bi bi-file-pdf"></i>PDF</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">BIO00056I GWAS Workshop</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Daniel Jeffares </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 4, 2025</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="learning-objectives" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Learning objectives</h1>
<ol type="1">
<li>Understand the principles of quantitative genetics.</li>
<li>Appreciate how this alters our perspective on evolution.</li>
<li>Learn how to design experiments that explore quantitative trait variation.</li>
</ol>
</section>
<section id="introduction" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Introduction</h1>
<section id="background" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="background"><span class="header-section-number">2.1</span> Background</h2>
<p>In the lectures, we learnt about quantitative traits and how to analyse them. To recap, quantitative traits are often described as complex traits because they are affected by genetic variation in many genes. The genetic variants in those genes each account for a small proportion of the trait variation. We usually assume that they contribute in an additive way (rather than more complex ways).</p>
<p>One good way to identify loci controlling quantitative traits is to perform a genome-wide association study (GWAS). GWAS experiments use diversity panels of individuals to identify genetic markers that are significantly associated with a trait of interest.</p>
<p>Using a diversity panel instead of a biparental mapping population ensures that linkage disequilibrium is low (segments of co-inherited DNA are small), and therefore resolution is high (QTL sizes are small), which helps to narrow down the regions containing the genes controlling the trait. Diversity panels also capture much more genetic variation.</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Quantitative traits and evolution
</div>
</div>
<div class="callout-body-container callout-body">
<p>Most traits are quantitative, and are caused by multiple genetic variants. This means that:</p>
<ul>
<li>We expect traits to adapt <em>gradually</em>, as a few variants change</li>
<li>We do not expect to observe many strong <em>selective sweeps</em></li>
</ul>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Glossary
</div>
</div>
<div class="callout-body-container callout-body">
<p>Technical definitions for this workshop.</p>
<ul>
<li><em>Schizosaccharomyces pombe</em>: Fission yeast; a haploid, unicellular fungus widely used as a model organism.</li>
<li><em>GWAS</em>: Genome-wide association study; a statistical approach to associate genetic variants with traits across a genome.</li>
<li><em>trait</em>: A measurable characteristic (phenotype), such as growth rate at high temperature.</li>
<li><em>Manhattan plot</em>: A plot of genomic position (x) versus −log10(P-value) (y) showing association signals across the genome.</li>
<li><em>QTL</em>: Quantitative trait locus; a genomic region containing variants that influence a quantitative trait.</li>
<li><em>haplotype</em>: A set of genetic variants along a single chromosome that tend to be inherited together.</li>
<li><em>selective sweep</em>: Rapid increase in frequency of a beneficial allele, reducing nearby genetic variation via linkage.</li>
</ul>
</div>
</div>
</section>
<section id="fission-yeast" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="fission-yeast"><span class="header-section-number">2.2</span> Fission yeast</h2>
<p>In this workshop we will use data from the fission yeast <em>Schizosaccharomyces pombe</em>, which is a model species for cell biology. You can read more about <em>S. pombe</em> <a href="http://www-users.york.ac.uk/~dj757/BIO00047I/BIO00056I_index.html">here</a>.</p>
<p>When fission yeast is not being used in labs as a model species for cell biology, it usually grows on fruit. It is most often found in high-sugar fruits or in fermented beverages (wine, beer, and quite often in cachaça in Brazil).</p>
<p><img src="images/pombe1.jpg" class="img-fluid" style="width:50.0%"></p>
<div style="font-size: 80%;">
<p><strong>Figure 1. The fission yeast <em>Schizosaccharomyces pombe</em>.</strong> Fission yeast is an excellent model system. It is easy to grow in the laboratory, easy to transform, and it has a very small genome (12 Mb). It has ~5,000 protein-coding genes and ~1,500 non-coding RNAs. It is haploid. Fission yeast and the budding yeast <em>Saccharomyces cerevisiae</em> are not closely related fungi, so study of both has contributed greatly to our understanding of molecular biology, evolutionary biology, and quantitative genetics.</p>
<div style="font-size: 100%;">
<p><br>
</p>
<section id="gwas-for-yeast" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="gwas-for-yeast"><span class="header-section-number">2.3</span> GWAS for yeast</h2>
<p>GWAS is often used for human or crop traits, but we can apply this method to any sexually recombining eukaryotic species for which we have trait data and polymorphism data (e.g., SNPs).</p>
<p>The data we will use come from two papers: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25665008">Nature Genetics</a> and <a href="http://europepmc.org/abstract/MED/28117401">Nature Communications</a>. These articles described the genetic diversity of 160 strains of this species by sequencing their genomes, and measured over 200 traits.</p>
<p>In this workshop we will examine GWAS results for four traits and consider what they could mean for the evolution of this species.</p>
</section>
<section id="the-traits" class="level2" data-number="2.4">
<h2 data-number="2.4" class="anchored" data-anchor-id="the-traits"><span class="header-section-number">2.4</span> The traits</h2>
<p>The traits we will look at today are:</p>
<p><strong>Growth in 400 mM NaCl</strong> The growth rate of the strains in 400 mM salt. This could be important if strains grew in fruit near the sea or in salty substrates.</p>
<p><strong>Growth at elevated temperature</strong> The growth rate of the strains at 40 °C. This is quite warm for this yeast, so it is a stressful condition.</p>
<p><strong>Maximal cell density in rich medium</strong> This trait measures how well each strain grows when given plenty of sugar and other resources.</p>
<p><strong>Glucose/fructose utilisation in wine</strong> How much of the glucose and fructose each strain used in grape must (used to make wine).</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Traits and trade-offs
</div>
</div>
<div class="callout-body-container callout-body">
<p>A high value in one of these traits could be adaptive in one environment but suboptimal in another. Also, optimising one trait may limit another trait. This could be because one mutation results in multiple changes, or the alleles that influence two traits are closely linked.</p>
</div>
</div>
<div style="page-break-after: always;"></div>
</section>
<section id="exercises" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Exercises</h1>
<section id="exploring-the-data" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="exploring-the-data"><span class="header-section-number">3.1</span> Exploring the data</h2>
<p>First, clear all the previous data you may have in R, set your working directory, and load the GWAS data.</p>
<p><strong>To set a working directory</strong> you can either use a command (as below), or use the menus;</p>
<p>Session &gt; Set Working Directory &gt; Choose Directory.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#clear all the previous data</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#set working directory</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">#your working directory will be different!</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/Users/dj757/gd/modules/BIO56I/workshops/gwas"</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co">#load the data</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">url</span>(<span class="st">"http://www-users.york.ac.uk/~dj757/BIO00056I/data/gwas-data.Rda"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>After loading these data you will have four data frames called <code>gluc</code>, <code>heat</code>, <code>rich</code>, and <code>salt</code>. Each contains the GWAS results for one trait. Start by looking at <code>salt</code>.</p>
<p>Try these commands to see what they contain:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#examine what is in the salt data frame</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(salt)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(salt)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(salt)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(salt)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This table contains the output from a GWAS tool called <a href="https://dougspeed.com/">LDAK</a>. This is one of many software tools that run GWAS analyses. It describes the association of each SNP with the trait of salt tolerance.</p>
<p>Each row of the table contains information about one SNP. For our purposes, the important columns are:</p>
<ul>
<li><code>Wald_P</code>, the P-value, the probability that this SNP is associated with this trait</li>
<li><code>Chr</code>, which chromosome the SNP is on</li>
<li><code>BP</code>, the position of the SNP in the chromosome</li>
<li><code>Effect_size</code>, how strongly this variant affects the trait</li>
</ul>
<p>The effect size of the variant estimates the proportion of trait variation that is explained by this SNP. From our understanding of the <a href="https://en.wikipedia.org/wiki/Neutral_theory_of_molecular_evolution">Neutral Theory of Molecular Evolution</a>, we know that most variants are selectively neutral, i.e., they have little or no effect on fitness.</p>
<p>Given this, we might expect that most variants have small, or no, effects on traits.</p>
<div style="page-break-after: always;"></div>
</section>
<section id="manhattan-plot" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="manhattan-plot"><span class="header-section-number">3.2</span> Manhattan plot</h2>
<p>A Manhattan plot is a standard way of viewing GWAS results. They are named Manhattan plots because they resemble the skyline of a high-rise city, such as Manhattan (New York).</p>
<p>Manhattan plots show the position of the SNP on the x-axis, and the P-value of the association on the y-axis. P-values are plotted as −log10(P), so that very small P-values end up at the top of the plot and stand out.</p>
<p>The code below will make a Manhattan plot for the salt trait.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#set plotting parameters</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">layout</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>), <span class="dv">1</span>, <span class="dv">3</span>, <span class="at">byrow =</span> <span class="cn">TRUE</span>),<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">5579133</span>,<span class="dv">4539804</span>,<span class="dv">2452883</span>))</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">bty=</span><span class="st">"l"</span>,<span class="at">cex=</span><span class="fl">0.8</span>,<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">0</span>)<span class="sc">+</span><span class="dv">2</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co">#loop through chromosomes 1,2,3, plotting each one in turn</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>){</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>      <span class="co">#make a subset of SNPs from this chromosome</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>      temp <span class="ot">=</span> <span class="fu">subset</span>(salt, Chr <span class="sc">==</span> j)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>      <span class="co">#plot these</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>      <span class="fu">plot</span>(temp<span class="sc">$</span>BP, <span class="sc">-</span><span class="fu">log10</span>(temp<span class="sc">$</span>Wald_P),</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>           <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">"Chr"</span>,j),<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">10</span>),</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>           <span class="at">xlab=</span><span class="st">"position"</span>,<span class="at">ylab=</span><span class="st">"-log10(P)"</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>      <span class="co">#draw a threshold line</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>      <span class="co">#SNPs above this line are statistically associated with the trait</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>      <span class="fu">abline</span>(<span class="at">h=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">1e-5</span>),<span class="at">col=</span><span class="dv">2</span>,<span class="at">lty=</span><span class="dv">2</span>)</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Discussion points: Manhattan plot
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Where are the strongest association signals located (do peaks cluster on particular chromosomes)?</li>
<li>Why are most points close to the bottom of the plot? What does this tell you about polygenic traits?</li>
<li>How does the −log10(P) scale aid interpretation compared to plotting raw P-values?</li>
<li>What biological processes could lead to a few tall peaks versus many low signals (e.g., selection, drift, mutation rate, linkage)?</li>
</ul>
</div>
</div>
<p>You will usually see a small number of very low P-values; with large −log10 values they sit at the top of the y-axis. For these, there is strong support that they affect the trait.</p>
<p>You will also see many SNPs with P-values around 0.1 or greater. These are at the bottom of the y-axis. There is weak support that they affect the trait. These may have tiny effect sizes, or none at all.</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
−log10 scale plotting
</div>
</div>
<div class="callout-body-container callout-body">
<p>Note that we use <code>plot(temp$BP, -log10(temp$Wald_P))</code> to plot the Wald_P P-values on a negative log scale. This means that very significant P-values (e.g., P = 1 × 10^−9.5) will show as high points on the plot (e.g., at position 9.5 on the vertical y-axis in this case).</p>
</div>
</div>
<p>All the non-significant SNPs (e.g., P = 0.1, which is 1 × 10^−1) will be at the bottom of the y-axis. If most variants have small or no effects on traits, there will be many of these.</p>
<p>This illustrates polygenic architecture: many variants with small effects and a few with larger effects.</p>
</section>
<section id="snp-effect-sizes" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="snp-effect-sizes"><span class="header-section-number">3.3</span> SNP effect sizes</h2>
<p>The strength of the association between genotype and phenotype relates to the effect size (how strongly the variant affects the trait). Some alleles affect the trait strongly; most do not. Now make a histogram of all the effect sizes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(salt<span class="sc">$</span>Effect_size,<span class="at">br=</span><span class="dv">30</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Discussion points: effect sizes
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>What does the histogram shape tell you about the distribution of SNP effect sizes?</li>
<li>How does this pattern relate to the Neutral Theory (many neutral or near-neutral variants)?</li>
<li>Where would you expect to see evidence of purifying selection or positive selection in such a distribution?</li>
</ul>
</div>
</div>
</section>
<section id="how-does-this-influence-evolution" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="how-does-this-influence-evolution"><span class="header-section-number">3.4</span> How does this influence evolution?</h2>
<p>Effect size here is the proportion of phenotypic variance explained by that SNP. Most effect sizes are very small—often &lt; 0.05—meaning each explains far less than 5% of the trait variance.</p>
<p>Consider how such tiny-effect alleles influence evolutionary change.</p>
<p>A <a href="https://en.wikipedia.org/wiki/Selective_sweep">selective sweep</a> is a process where a beneficial mutation becomes fixed in a population rapidly, removing the genetic variation around the selected allele.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Discussion points: selective sweeps and polygenicity
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>If a trait is influenced by a few large-effect mutations, how likely is a classic selective sweep and why?</li>
<li>If a trait is influenced by many tiny-effect mutations, what evolutionary pattern replaces a sweep? (Think subtle allele frequency shifts.)</li>
<li>How can tight linkage to a deleterious allele modify or slow a sweep?</li>
<li>What genomic signatures (e.g., reduced diversity, extended haplotype homozygosity) would differ between oligogenic and highly polygenic adaptation?</li>
</ul>
</div>
</div>
</section>
<section id="another-trait-glucose-fructose-use" class="level2" data-number="3.5">
<h2 data-number="3.5" class="anchored" data-anchor-id="another-trait-glucose-fructose-use"><span class="header-section-number">3.5</span> Another trait: glucose &amp; fructose use</h2>
<p>In a vineyard environment, it is advantageous to metabolise glucose and fructose in rotting fruit. Glucose/fructose use is in the <code>gluc</code> data frame.</p>
<p>Again, we will look at the P-values (<code>Wald_P</code>) and how much each SNP is predicted to affect the trait (<code>Effect_size</code>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#set plotting parameters</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">layout</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>), <span class="dv">1</span>, <span class="dv">3</span>, <span class="at">byrow =</span> <span class="cn">TRUE</span>),<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">5579133</span>,<span class="dv">4539804</span>,<span class="dv">2452883</span>))</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">bty=</span><span class="st">"l"</span>,<span class="at">cex=</span><span class="fl">0.8</span>,<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">0</span>)<span class="sc">+</span><span class="dv">2</span>)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">#loop through chromosomes 1,2,3, plotting each one in turn</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>){</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>      temp <span class="ot">=</span> <span class="fu">subset</span>(gluc, Chr <span class="sc">==</span> j)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>      <span class="fu">plot</span>(temp<span class="sc">$</span>BP, <span class="sc">-</span><span class="fu">log10</span>(temp<span class="sc">$</span>Wald_P),</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">"Chr"</span>, j, <span class="st">"Gluc"</span>),<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">10</span>),</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>           <span class="at">xlab=</span><span class="st">"position"</span>,<span class="at">ylab=</span><span class="st">"-log10(P)"</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>      <span class="fu">abline</span>(<span class="at">h=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">1e-5</span>),<span class="at">col=</span><span class="dv">2</span>,<span class="at">lty=</span><span class="dv">2</span>)</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="sc">:::</span> callout<span class="sc">-</span>tip</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Discussion points: second trait comparison</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> Do the strongest glucose<span class="sc">/</span>fructose association peaks overlap with salt peaks? What could shared loci <span class="fu">imply</span> (pleiotropy vs linkage)?</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> Are there trait<span class="sc">-</span>specific peaks? What might that say about specialised metabolic pathways?</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> How could environmental <span class="fu">context</span> (salinity vs sugar availability) shift which alleles are favoured?</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="sc">:::</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>Again, you will see a small number of very low P<span class="sc">-</span>values at the top of the y<span class="sc">-</span><span class="fu">axis</span> (strong support) and many higher P<span class="sc">-</span>values at the <span class="fu">bottom</span> (weak or no support; tiny or no effect sizes).</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>Now, make a histogram of the effect sizes<span class="sc">:</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(gluc<span class="sc">$</span>Effect_size)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="looking-at-two-traits" class="level2" data-number="3.6">
<h2 data-number="3.6" class="anchored" data-anchor-id="looking-at-two-traits"><span class="header-section-number">3.6</span> Looking at two traits</h2>
<p>Species cannot adapt to a single condition (e.g., heat or salt) in isolation because environments are complex and variable. As a thought experiment, we will consider this using the fission yeast data.</p>
<p>We will compare two Manhattan plots by overlaying them: salt (black) and glucose/fructose metabolism (red). The output PDF will appear in your working directory.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#open the pdf writer</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"GWAS-plot3.pdf"</span>,<span class="at">width=</span><span class="dv">60</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co">#set up the plot layout</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="fu">layout</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>), <span class="dv">1</span>, <span class="dv">3</span>, <span class="at">byrow =</span> <span class="cn">TRUE</span>),<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">5579133</span>,<span class="dv">4539804</span>,<span class="dv">2452883</span>))</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">bty=</span><span class="st">"l"</span>,<span class="at">cex=</span><span class="fl">0.8</span>,<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">0</span>)<span class="sc">+</span><span class="dv">2</span>)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co">#loop through chromosomes</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>){</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>      <span class="co">#plot the salt GWAS points (black dots)</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>      temp <span class="ot">=</span> <span class="fu">subset</span>(salt, Chr <span class="sc">==</span> j)</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>      <span class="fu">plot</span>(temp<span class="sc">$</span>BP, <span class="sc">-</span><span class="fu">log10</span>(temp<span class="sc">$</span>Wald_P),<span class="at">main =</span> <span class="st">""</span>,<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">10</span>),<span class="at">xlab=</span><span class="st">"position"</span>,<span class="at">ylab=</span><span class="st">"-log10(P)"</span>)</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>      <span class="co">#add the glucose/fructose GWAS points (red crosses)</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>      temp2 <span class="ot">=</span> <span class="fu">subset</span>(gluc, Chr <span class="sc">==</span> j)</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>      <span class="fu">points</span>(temp2<span class="sc">$</span>BP, <span class="sc">-</span><span class="fu">log10</span>(temp2<span class="sc">$</span>Wald_P),<span class="at">main =</span> <span class="st">"gluc"</span>,<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">10</span>),<span class="at">xlab=</span><span class="st">"position"</span>,<span class="at">ylab=</span><span class="st">"-log10(P)"</span>,<span class="at">col=</span><span class="dv">2</span>,<span class="at">pch=</span><span class="dv">3</span>)</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>      <span class="co">#add the significance threshold</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>      <span class="fu">abline</span>(<span class="at">h=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">1e-5</span>),<span class="at">col=</span><span class="dv">2</span>,<span class="at">lty=</span><span class="dv">2</span>)</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="co">#close the pdf writer</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now find the pdf called <code>GWAS-plot3.pdf</code> and take a look.</p>
<section id="might-mutations-for-two-traits-conflict" class="level3" data-number="3.6.1">
<h3 data-number="3.6.1" class="anchored" data-anchor-id="might-mutations-for-two-traits-conflict"><span class="header-section-number">3.6.1</span> Might mutations for two traits conflict?</h3>
<p>Look at the significant peak on chromosome 3 (~1,500,000 bp). This likely has a relatively large effect size. There is also an important variant for glucose/fructose use nearby on chromosome 3.</p>
<p>Consider what might happen if alleles are linked and the environment is complex—salty yet fructose-rich (e.g., a coastal vineyard).</p>
<p>What might happen if helpful alleles for both traits are on the same haplotype?</p>
<p><img src="images/pombe2.jpg" class="img-fluid" style="width:100.0%"></p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Haplotypes
</div>
</div>
<div class="callout-body-container callout-body">
<p>Physically, a haplotype is a stretch of DNA from one chromosome. In diploids, each individual carries two homologous chromosomes. In haploid <em>S. pombe</em>, different haplotypes are simply different chromosome versions present across the population. Through meiotic recombination haplotypes mix, but SNPs very close together rarely separate, so they tend to travel together.</p>
</div>
</div>
</section>
<section id="haplotype-problems" class="level3" data-number="3.6.2">
<h3 data-number="3.6.2" class="anchored" data-anchor-id="haplotype-problems"><span class="header-section-number">3.6.2</span> Haplotype problems</h3>
<p>Consider these situations:</p>
<ul>
<li>What might happen if helpful alleles are on the same haplotype?</li>
<li>What might happen if helpful alleles are on different (opposing) haplotypes?</li>
</ul>
<p>The figures below will help to explain.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Discussion points: haplotypes and linkage
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>How does tight linkage between beneficial alleles influence the speed of joint adaptation?</li>
<li>What evolutionary constraint arises when beneficial alleles are on different haplotypes in low recombination regions?</li>
<li>How can recombination rate modulate the resolution of this “conflict” between opposing haplotypes?</li>
<li>Which scenario (single vs opposing haplotypes) better facilitates rapid multi-trait optimisation?</li>
</ul>
</div>
</div>
<p><img src="images/pombe2.jpg" class="img-fluid" style="width:100.0%"> <strong>Figure 2. Beneficial alleles on one haplotype.</strong> Would there be any conflict here?<br>
</p>
<p><img src="images/pombe3.jpg" class="img-fluid" style="width:100.0%"> <strong>Figure 3. Beneficial alleles on two different haplotypes.</strong> What might happen here? What is the potential problem?<br>
</p>
<div style="page-break-after: always;"></div>
</section>
</section>
</section>
<section id="summary-what-we-have-learned" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Summary: what we have learned</h1>
<p>This workshop used data from fission yeast, but the principles apply broadly to quantitative traits. Key points:</p>
<ol type="1">
<li>Most SNPs do not strongly associate with traits.</li>
<li>Most SNPs have very small effects on traits, but some do have detectable effects.</li>
<li>SNPs do not affect all traits at once. Some SNPs affect one trait; some affect another.</li>
<li>Genetic variants, like SNPs, travel through time and space in haplotypes.</li>
</ol>
<div style="page-break-after: always;"></div>
</section>
<section id="after-the-workshop-exam-style-questions" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> After the workshop: exam-style questions</h1>
<p>Exam questions often ask you to apply workshop principles to other species or scenarios. Reflecting on these helps deepen understanding.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Principles of quantitative genetics
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Most SNPs have no effect, or very small effects, on traits.</li>
<li>SNPs that do affect traits are generally responsible for less than 5% of the trait variation.</li>
<li>Effect sizes are also small for other genetic variants, such as transposon insertions, duplications/deletions and so on.</li>
<li>Genetic variants can affect one trait, or several.</li>
<li>In the short term, selection occurs on <em>haplotypes</em>, but over time recombination will separate linked variants.</li>
</ul>
</div>
</div>
<section id="question-1-10-points" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="question-1-10-points"><span class="header-section-number">5.1</span> Question 1 (10 points)</h2>
<p>This question checks your understanding of quantitative traits and why most variants have small effects.</p>
<ol type="a">
<li><p><strong>Define</strong> the difference between a simple (single-gene) trait and a quantitative trait. Give one example of each from this workshop. (2 points)</p></li>
<li><p>The histogram of SNP effect sizes for the salt trait is heavily skewed towards zero. <strong>Describe</strong> what this pattern tells us about the distribution of SNP effects in the population. (3 points)</p></li>
<li><p>Using the <a href="https://en.wikipedia.org/wiki/Neutral_theory_of_molecular_evolution">Neutral Theory of Molecular Evolution</a>, <strong>explain</strong> why most SNPs are expected to have little or no effect on fitness. (3 points)</p></li>
<li><p><strong>Suggest</strong> one practical way to increase the power of a GWAS to detect very small-effect variants and explain why it helps. (2 points)</p></li>
</ol>
</section>
<section id="question-2-10-points" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="question-2-10-points"><span class="header-section-number">5.2</span> Question 2 (10 points)</h2>
<p>You notice that the Manhattan plots for salt tolerance and glucose/fructose utilisation share a peak on chromosome 3, but also have trait-specific peaks.</p>
<ol type="a">
<li><p><strong>Describe</strong> what an overlapping peak for both traits suggests about the underlying genetics of those traits. (3 points)</p></li>
<li><p><strong>State</strong> one additional analysis you could carry out to test whether the same causal variant influences both traits. (2 points)</p></li>
<li><p><strong>Explain</strong> how recombination could help resolve a situation where beneficial alleles for two traits are on different haplotypes. (3 points)</p></li>
<li><p><strong>Identify</strong> one environmental measurement you would collect alongside the genomic data to interpret potential trade-offs between the two traits, and justify your choice. (2 points)</p></li>
</ol>


<!-- -->

</section>
</section>
</div>
</div>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb8" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "BIO00056I GWAS Workshop"</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> Daniel Jeffares</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2025-11-04</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="co">    html:</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="co">        code-copy: true</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="co">        code-tools: true</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="co">        code-link: true</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="co">        toc: true</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="co">        number-sections: true</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="co">        toc-location: right</span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a><span class="co">        toc-depth: 2</span></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="co">        highlight-style: github</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a><span class="co">    </span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a><span class="co">    pdf:</span></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a><span class="co">      toc: true</span></span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a><span class="co">      number-sections: true</span></span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a><span class="co">      colorlinks: true</span></span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> source</span></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a><span class="fu"># Learning objectives</span></span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>Understand the principles of quantitative genetics.</span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>Appreciate how this alters our perspective on evolution.</span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span>Learn how to design experiments that explore quantitative trait variation.</span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a><span class="fu"># Introduction</span></span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a><span class="fu">## Background</span></span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a>In the lectures, we learnt about quantitative traits and how to analyse them. To recap, quantitative traits are often described as complex traits because they are affected by genetic variation in many genes. The genetic variants in those genes each account for a small proportion of the trait variation. We usually assume that they contribute in an additive way (rather than more complex ways).</span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a>One good way to identify loci controlling quantitative traits is to perform a genome-wide association study (GWAS). GWAS experiments use diversity panels of individuals to identify genetic markers that are significantly associated with a trait of interest.</span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a>Using a diversity panel instead of a biparental mapping population ensures that linkage disequilibrium is low (segments of co-inherited DNA are small), and therefore resolution is high (QTL sizes are small), which helps to narrow down the regions containing the genes controlling the trait. Diversity panels also capture much more genetic variation.</span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a>:::callout-important</span>
<span id="cb8-42"><a href="#cb8-42" aria-hidden="true" tabindex="-1"></a><span class="fu"># Quantitative traits and evolution</span></span>
<span id="cb8-43"><a href="#cb8-43" aria-hidden="true" tabindex="-1"></a>Most traits are quantitative, and are caused by multiple genetic variants. This means that:</span>
<span id="cb8-44"><a href="#cb8-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-45"><a href="#cb8-45" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>We expect traits to adapt *gradually*, as a few variants change</span>
<span id="cb8-46"><a href="#cb8-46" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>We do not expect to observe many strong *selective sweeps*</span>
<span id="cb8-47"><a href="#cb8-47" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb8-48"><a href="#cb8-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-49"><a href="#cb8-49" aria-hidden="true" tabindex="-1"></a>:::callout-note</span>
<span id="cb8-50"><a href="#cb8-50" aria-hidden="true" tabindex="-1"></a><span class="fu"># Glossary</span></span>
<span id="cb8-51"><a href="#cb8-51" aria-hidden="true" tabindex="-1"></a>Technical definitions for this workshop.</span>
<span id="cb8-52"><a href="#cb8-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-53"><a href="#cb8-53" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>*Schizosaccharomyces pombe*: Fission yeast; a haploid, unicellular fungus widely used as a model organism.</span>
<span id="cb8-54"><a href="#cb8-54" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>*GWAS*: Genome-wide association study; a statistical approach to associate genetic variants with traits across a genome.</span>
<span id="cb8-55"><a href="#cb8-55" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>*trait*: A measurable characteristic (phenotype), such as growth rate at high temperature.</span>
<span id="cb8-56"><a href="#cb8-56" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>*Manhattan plot*: A plot of genomic position (x) versus −log10(P-value) (y) showing association signals across the genome.</span>
<span id="cb8-57"><a href="#cb8-57" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>*QTL*: Quantitative trait locus; a genomic region containing variants that influence a quantitative trait.</span>
<span id="cb8-58"><a href="#cb8-58" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>*haplotype*: A set of genetic variants along a single chromosome that tend to be inherited together.</span>
<span id="cb8-59"><a href="#cb8-59" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>*selective sweep*: Rapid increase in frequency of a beneficial allele, reducing nearby genetic variation via linkage.</span>
<span id="cb8-60"><a href="#cb8-60" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb8-61"><a href="#cb8-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-62"><a href="#cb8-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-63"><a href="#cb8-63" aria-hidden="true" tabindex="-1"></a><span class="fu">## Fission yeast</span></span>
<span id="cb8-64"><a href="#cb8-64" aria-hidden="true" tabindex="-1"></a>In this workshop we will use data from the fission yeast *Schizosaccharomyces pombe*, which is a model species for cell biology.</span>
<span id="cb8-65"><a href="#cb8-65" aria-hidden="true" tabindex="-1"></a>You can read more about *S. pombe* <span class="co">[</span><span class="ot">here</span><span class="co">](http://www-users.york.ac.uk/~dj757/BIO00047I/BIO00056I_index.html)</span>.</span>
<span id="cb8-66"><a href="#cb8-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-67"><a href="#cb8-67" aria-hidden="true" tabindex="-1"></a>When fission yeast is not being used in labs as a model species for cell biology, it usually grows on fruit. It is most often found in high-sugar fruits or in fermented beverages (wine, beer, and quite often in cachaça in Brazil).</span>
<span id="cb8-68"><a href="#cb8-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-69"><a href="#cb8-69" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/pombe1.jpg)</span>{width=50%}</span>
<span id="cb8-70"><a href="#cb8-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-71"><a href="#cb8-71" aria-hidden="true" tabindex="-1"></a>::: {style="font-size: 80%;"}</span>
<span id="cb8-72"><a href="#cb8-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-73"><a href="#cb8-73" aria-hidden="true" tabindex="-1"></a>**Figure 1. The fission yeast *Schizosaccharomyces pombe*.** Fission yeast is an excellent model system. It is easy to grow in the laboratory, easy to transform, and it has a very small genome (12 Mb). It has ~5,000 protein-coding genes and ~1,500 non-coding RNAs. It is haploid. Fission yeast and the budding yeast *Saccharomyces cerevisiae* are not closely related fungi, so study of both has contributed greatly to our understanding of molecular biology, evolutionary biology, and quantitative genetics.</span>
<span id="cb8-74"><a href="#cb8-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-75"><a href="#cb8-75" aria-hidden="true" tabindex="-1"></a>::: {style="font-size: 100%;"}</span>
<span id="cb8-76"><a href="#cb8-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-77"><a href="#cb8-77" aria-hidden="true" tabindex="-1"></a>\</span>
<span id="cb8-78"><a href="#cb8-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-79"><a href="#cb8-79" aria-hidden="true" tabindex="-1"></a><span class="fu">## GWAS for yeast</span></span>
<span id="cb8-80"><a href="#cb8-80" aria-hidden="true" tabindex="-1"></a>GWAS is often used for human or crop traits, but we can apply this method to any sexually recombining eukaryotic species for which we have trait data and polymorphism data (e.g., SNPs).</span>
<span id="cb8-81"><a href="#cb8-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-82"><a href="#cb8-82" aria-hidden="true" tabindex="-1"></a>The data we will use come from two papers: <span class="co">[</span><span class="ot">Nature Genetics</span><span class="co">](https://www.ncbi.nlm.nih.gov/pubmed/25665008)</span> and <span class="co">[</span><span class="ot">Nature Communications</span><span class="co">](http://europepmc.org/abstract/MED/28117401)</span>. These articles described the genetic diversity of 160 strains of this species by sequencing their genomes, and measured over 200 traits.</span>
<span id="cb8-83"><a href="#cb8-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-84"><a href="#cb8-84" aria-hidden="true" tabindex="-1"></a>In this workshop we will examine GWAS results for four traits and consider what they could mean for the evolution of this species.</span>
<span id="cb8-85"><a href="#cb8-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-86"><a href="#cb8-86" aria-hidden="true" tabindex="-1"></a><span class="fu">## The traits</span></span>
<span id="cb8-87"><a href="#cb8-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-88"><a href="#cb8-88" aria-hidden="true" tabindex="-1"></a>The traits we will look at today are:</span>
<span id="cb8-89"><a href="#cb8-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-90"><a href="#cb8-90" aria-hidden="true" tabindex="-1"></a>**Growth in 400 mM NaCl**</span>
<span id="cb8-91"><a href="#cb8-91" aria-hidden="true" tabindex="-1"></a>The growth rate of the strains in 400 mM salt. This could be important if strains grew in fruit near the sea or in salty substrates.</span>
<span id="cb8-92"><a href="#cb8-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-93"><a href="#cb8-93" aria-hidden="true" tabindex="-1"></a>**Growth at elevated temperature** </span>
<span id="cb8-94"><a href="#cb8-94" aria-hidden="true" tabindex="-1"></a>The growth rate of the strains at 40 °C. This is quite warm for this yeast, so it is a stressful condition.</span>
<span id="cb8-95"><a href="#cb8-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-96"><a href="#cb8-96" aria-hidden="true" tabindex="-1"></a>**Maximal cell density in rich medium** </span>
<span id="cb8-97"><a href="#cb8-97" aria-hidden="true" tabindex="-1"></a>This trait measures how well each strain grows when given plenty of sugar and other resources.</span>
<span id="cb8-98"><a href="#cb8-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-99"><a href="#cb8-99" aria-hidden="true" tabindex="-1"></a>**Glucose/fructose utilisation in wine** </span>
<span id="cb8-100"><a href="#cb8-100" aria-hidden="true" tabindex="-1"></a>How much of the glucose and fructose each strain used in grape must (used to make wine).</span>
<span id="cb8-101"><a href="#cb8-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-102"><a href="#cb8-102" aria-hidden="true" tabindex="-1"></a>::: callout-note</span>
<span id="cb8-103"><a href="#cb8-103" aria-hidden="true" tabindex="-1"></a><span class="fu">## Traits and trade-offs</span></span>
<span id="cb8-104"><a href="#cb8-104" aria-hidden="true" tabindex="-1"></a>A high value in one of these traits could be adaptive in one environment but suboptimal in another. Also, optimising one trait may limit another trait. This could be because one mutation results in multiple changes, or the alleles that influence two traits are closely linked.</span>
<span id="cb8-105"><a href="#cb8-105" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb8-106"><a href="#cb8-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-107"><a href="#cb8-107" aria-hidden="true" tabindex="-1"></a>{{&lt; pagebreak &gt;}}</span>
<span id="cb8-108"><a href="#cb8-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-109"><a href="#cb8-109" aria-hidden="true" tabindex="-1"></a><span class="fu"># Exercises</span></span>
<span id="cb8-110"><a href="#cb8-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-111"><a href="#cb8-111" aria-hidden="true" tabindex="-1"></a><span class="fu">## Exploring the data</span></span>
<span id="cb8-112"><a href="#cb8-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-113"><a href="#cb8-113" aria-hidden="true" tabindex="-1"></a>First, clear all the previous data you may have in R, set your working directory, and load the GWAS data.</span>
<span id="cb8-114"><a href="#cb8-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-115"><a href="#cb8-115" aria-hidden="true" tabindex="-1"></a>**To set a working directory** you can either use a command (as below), or use the menus; </span>
<span id="cb8-116"><a href="#cb8-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-117"><a href="#cb8-117" aria-hidden="true" tabindex="-1"></a>Session &gt; Set Working Directory &gt; Choose Directory.</span>
<span id="cb8-118"><a href="#cb8-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-121"><a href="#cb8-121" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-122"><a href="#cb8-122" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb8-123"><a href="#cb8-123" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb8-124"><a href="#cb8-124" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb8-125"><a href="#cb8-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-126"><a href="#cb8-126" aria-hidden="true" tabindex="-1"></a><span class="co">#clear all the previous data</span></span>
<span id="cb8-127"><a href="#cb8-127" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb8-128"><a href="#cb8-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-129"><a href="#cb8-129" aria-hidden="true" tabindex="-1"></a><span class="co">#set working directory</span></span>
<span id="cb8-130"><a href="#cb8-130" aria-hidden="true" tabindex="-1"></a><span class="co">#your working directory will be different!</span></span>
<span id="cb8-131"><a href="#cb8-131" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/Users/dj757/gd/modules/BIO56I/workshops/gwas"</span>)</span>
<span id="cb8-132"><a href="#cb8-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-133"><a href="#cb8-133" aria-hidden="true" tabindex="-1"></a><span class="co">#load the data</span></span>
<span id="cb8-134"><a href="#cb8-134" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">url</span>(<span class="st">"http://www-users.york.ac.uk/~dj757/BIO00056I/data/gwas-data.Rda"</span>))</span>
<span id="cb8-135"><a href="#cb8-135" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb8-136"><a href="#cb8-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-137"><a href="#cb8-137" aria-hidden="true" tabindex="-1"></a>After loading these data you will have four data frames called <span class="in">`gluc`</span>, <span class="in">`heat`</span>, <span class="in">`rich`</span>, and <span class="in">`salt`</span>. Each contains the GWAS results for one trait. Start by looking at <span class="in">`salt`</span>.</span>
<span id="cb8-138"><a href="#cb8-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-139"><a href="#cb8-139" aria-hidden="true" tabindex="-1"></a>Try these commands to see what they contain:</span>
<span id="cb8-140"><a href="#cb8-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-143"><a href="#cb8-143" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-144"><a href="#cb8-144" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb8-145"><a href="#cb8-145" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb8-146"><a href="#cb8-146" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb8-147"><a href="#cb8-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-148"><a href="#cb8-148" aria-hidden="true" tabindex="-1"></a><span class="co">#examine what is in the salt data frame</span></span>
<span id="cb8-149"><a href="#cb8-149" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(salt)</span>
<span id="cb8-150"><a href="#cb8-150" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(salt)</span>
<span id="cb8-151"><a href="#cb8-151" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(salt)</span>
<span id="cb8-152"><a href="#cb8-152" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(salt)</span>
<span id="cb8-153"><a href="#cb8-153" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb8-154"><a href="#cb8-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-155"><a href="#cb8-155" aria-hidden="true" tabindex="-1"></a>This table contains the output from a GWAS tool called <span class="co">[</span><span class="ot">LDAK</span><span class="co">](https://dougspeed.com/)</span>. This is one of many software tools that run GWAS analyses. It describes the association of each SNP with the trait of salt tolerance.</span>
<span id="cb8-156"><a href="#cb8-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-157"><a href="#cb8-157" aria-hidden="true" tabindex="-1"></a>Each row of the table contains information about one SNP. For our purposes, the important columns are:</span>
<span id="cb8-158"><a href="#cb8-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-159"><a href="#cb8-159" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`Wald_P`</span>, the P-value, the probability that this SNP is associated with this trait</span>
<span id="cb8-160"><a href="#cb8-160" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`Chr`</span>, which chromosome the SNP is on</span>
<span id="cb8-161"><a href="#cb8-161" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`BP`</span>, the position of the SNP in the chromosome</span>
<span id="cb8-162"><a href="#cb8-162" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span><span class="in">`Effect_size`</span>, how strongly this variant affects the trait</span>
<span id="cb8-163"><a href="#cb8-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-164"><a href="#cb8-164" aria-hidden="true" tabindex="-1"></a>The effect size of the variant estimates the proportion of trait variation that is explained by this SNP. From our understanding of the <span class="co">[</span><span class="ot">Neutral Theory of Molecular Evolution</span><span class="co">](https://en.wikipedia.org/wiki/Neutral_theory_of_molecular_evolution)</span>, we know that most variants are selectively neutral, i.e., they have little or no effect on fitness.</span>
<span id="cb8-165"><a href="#cb8-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-166"><a href="#cb8-166" aria-hidden="true" tabindex="-1"></a>Given this, we might expect that most variants have small, or no, effects on traits.</span>
<span id="cb8-167"><a href="#cb8-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-168"><a href="#cb8-168" aria-hidden="true" tabindex="-1"></a>{{&lt; pagebreak &gt;}}</span>
<span id="cb8-169"><a href="#cb8-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-170"><a href="#cb8-170" aria-hidden="true" tabindex="-1"></a><span class="fu">## Manhattan plot</span></span>
<span id="cb8-171"><a href="#cb8-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-172"><a href="#cb8-172" aria-hidden="true" tabindex="-1"></a>A Manhattan plot is a standard way of viewing GWAS results. They are named Manhattan plots because they resemble the skyline of a high-rise city, such as Manhattan (New York).</span>
<span id="cb8-173"><a href="#cb8-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-174"><a href="#cb8-174" aria-hidden="true" tabindex="-1"></a>Manhattan plots show the position of the SNP on the x-axis, and the P-value of the association on the y-axis. P-values are plotted as −log10(P), so that very small P-values end up at the top of the plot and stand out.</span>
<span id="cb8-175"><a href="#cb8-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-176"><a href="#cb8-176" aria-hidden="true" tabindex="-1"></a>The code below will make a Manhattan plot for the salt trait.</span>
<span id="cb8-177"><a href="#cb8-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-180"><a href="#cb8-180" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-181"><a href="#cb8-181" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb8-182"><a href="#cb8-182" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb8-183"><a href="#cb8-183" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb8-184"><a href="#cb8-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-185"><a href="#cb8-185" aria-hidden="true" tabindex="-1"></a><span class="co">#set plotting parameters</span></span>
<span id="cb8-186"><a href="#cb8-186" aria-hidden="true" tabindex="-1"></a><span class="fu">layout</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>), <span class="dv">1</span>, <span class="dv">3</span>, <span class="at">byrow =</span> <span class="cn">TRUE</span>),<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">5579133</span>,<span class="dv">4539804</span>,<span class="dv">2452883</span>))</span>
<span id="cb8-187"><a href="#cb8-187" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">bty=</span><span class="st">"l"</span>,<span class="at">cex=</span><span class="fl">0.8</span>,<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">0</span>)<span class="sc">+</span><span class="dv">2</span>)</span>
<span id="cb8-188"><a href="#cb8-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-189"><a href="#cb8-189" aria-hidden="true" tabindex="-1"></a><span class="co">#loop through chromosomes 1,2,3, plotting each one in turn</span></span>
<span id="cb8-190"><a href="#cb8-190" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>){</span>
<span id="cb8-191"><a href="#cb8-191" aria-hidden="true" tabindex="-1"></a>      <span class="co">#make a subset of SNPs from this chromosome</span></span>
<span id="cb8-192"><a href="#cb8-192" aria-hidden="true" tabindex="-1"></a>      temp <span class="ot">=</span> <span class="fu">subset</span>(salt, Chr <span class="sc">==</span> j)</span>
<span id="cb8-193"><a href="#cb8-193" aria-hidden="true" tabindex="-1"></a>      <span class="co">#plot these</span></span>
<span id="cb8-194"><a href="#cb8-194" aria-hidden="true" tabindex="-1"></a>      <span class="fu">plot</span>(temp<span class="sc">$</span>BP, <span class="sc">-</span><span class="fu">log10</span>(temp<span class="sc">$</span>Wald_P),</span>
<span id="cb8-195"><a href="#cb8-195" aria-hidden="true" tabindex="-1"></a>           <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">"Chr"</span>,j),<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">10</span>),</span>
<span id="cb8-196"><a href="#cb8-196" aria-hidden="true" tabindex="-1"></a>           <span class="at">xlab=</span><span class="st">"position"</span>,<span class="at">ylab=</span><span class="st">"-log10(P)"</span></span>
<span id="cb8-197"><a href="#cb8-197" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb8-198"><a href="#cb8-198" aria-hidden="true" tabindex="-1"></a>      <span class="co">#draw a threshold line</span></span>
<span id="cb8-199"><a href="#cb8-199" aria-hidden="true" tabindex="-1"></a>      <span class="co">#SNPs above this line are statistically associated with the trait</span></span>
<span id="cb8-200"><a href="#cb8-200" aria-hidden="true" tabindex="-1"></a>      <span class="fu">abline</span>(<span class="at">h=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">1e-5</span>),<span class="at">col=</span><span class="dv">2</span>,<span class="at">lty=</span><span class="dv">2</span>)</span>
<span id="cb8-201"><a href="#cb8-201" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-202"><a href="#cb8-202" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb8-203"><a href="#cb8-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-204"><a href="#cb8-204" aria-hidden="true" tabindex="-1"></a>::: callout-tip</span>
<span id="cb8-205"><a href="#cb8-205" aria-hidden="true" tabindex="-1"></a><span class="fu"># Discussion points: Manhattan plot</span></span>
<span id="cb8-206"><a href="#cb8-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-207"><a href="#cb8-207" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Where are the strongest association signals located (do peaks cluster on particular chromosomes)?</span>
<span id="cb8-208"><a href="#cb8-208" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Why are most points close to the bottom of the plot? What does this tell you about polygenic traits?</span>
<span id="cb8-209"><a href="#cb8-209" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>How does the −log10(P) scale aid interpretation compared to plotting raw P-values?</span>
<span id="cb8-210"><a href="#cb8-210" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>What biological processes could lead to a few tall peaks versus many low signals (e.g., selection, drift, mutation rate, linkage)?</span>
<span id="cb8-211"><a href="#cb8-211" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb8-212"><a href="#cb8-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-213"><a href="#cb8-213" aria-hidden="true" tabindex="-1"></a>You will usually see a small number of very low P-values; with large −log10 values they sit at the top of the y-axis. For these, there is strong support that they affect the trait.</span>
<span id="cb8-214"><a href="#cb8-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-215"><a href="#cb8-215" aria-hidden="true" tabindex="-1"></a>You will also see many SNPs with P-values around 0.1 or greater. These are at the bottom of the y-axis. There is weak support that they affect the trait. These may have tiny effect sizes, or none at all.</span>
<span id="cb8-216"><a href="#cb8-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-217"><a href="#cb8-217" aria-hidden="true" tabindex="-1"></a>:::callout-important</span>
<span id="cb8-218"><a href="#cb8-218" aria-hidden="true" tabindex="-1"></a><span class="fu"># −log10 scale plotting</span></span>
<span id="cb8-219"><a href="#cb8-219" aria-hidden="true" tabindex="-1"></a>Note that we use <span class="in">`plot(temp$BP, -log10(temp$Wald_P))`</span> to plot the Wald_P P-values on a negative log scale. This means that very significant P-values (e.g., P = 1 × 10^−9.5) will show as high points on the plot (e.g., at position 9.5 on the vertical y-axis in this case).</span>
<span id="cb8-220"><a href="#cb8-220" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb8-221"><a href="#cb8-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-222"><a href="#cb8-222" aria-hidden="true" tabindex="-1"></a>All the non-significant SNPs (e.g., P = 0.1, which is 1 × 10^−1) will be at the bottom of the y-axis. If most variants have small or no effects on traits, there will be many of these.</span>
<span id="cb8-223"><a href="#cb8-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-224"><a href="#cb8-224" aria-hidden="true" tabindex="-1"></a>This illustrates polygenic architecture: many variants with small effects and a few with larger effects.</span>
<span id="cb8-225"><a href="#cb8-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-226"><a href="#cb8-226" aria-hidden="true" tabindex="-1"></a><span class="fu">## SNP effect sizes </span></span>
<span id="cb8-227"><a href="#cb8-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-228"><a href="#cb8-228" aria-hidden="true" tabindex="-1"></a>The strength of the association between genotype and phenotype relates to the effect size (how strongly the variant affects the trait). Some alleles affect the trait strongly; most do not. Now make a histogram of all the effect sizes.</span>
<span id="cb8-229"><a href="#cb8-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-232"><a href="#cb8-232" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-233"><a href="#cb8-233" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb8-234"><a href="#cb8-234" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb8-235"><a href="#cb8-235" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb8-236"><a href="#cb8-236" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(salt<span class="sc">$</span>Effect_size,<span class="at">br=</span><span class="dv">30</span>)</span>
<span id="cb8-237"><a href="#cb8-237" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb8-238"><a href="#cb8-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-239"><a href="#cb8-239" aria-hidden="true" tabindex="-1"></a>::: callout-tip</span>
<span id="cb8-240"><a href="#cb8-240" aria-hidden="true" tabindex="-1"></a><span class="fu"># Discussion points: effect sizes</span></span>
<span id="cb8-241"><a href="#cb8-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-242"><a href="#cb8-242" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>What does the histogram shape tell you about the distribution of SNP effect sizes?</span>
<span id="cb8-243"><a href="#cb8-243" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>How does this pattern relate to the Neutral Theory (many neutral or near-neutral variants)?</span>
<span id="cb8-244"><a href="#cb8-244" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Where would you expect to see evidence of purifying selection or positive selection in such a distribution?</span>
<span id="cb8-245"><a href="#cb8-245" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb8-246"><a href="#cb8-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-247"><a href="#cb8-247" aria-hidden="true" tabindex="-1"></a><span class="fu">## How does this influence evolution?</span></span>
<span id="cb8-248"><a href="#cb8-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-249"><a href="#cb8-249" aria-hidden="true" tabindex="-1"></a>Effect size here is the proportion of phenotypic variance explained by that SNP. Most effect sizes are very small—often &lt; 0.05—meaning each explains far less than 5% of the trait variance.</span>
<span id="cb8-250"><a href="#cb8-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-251"><a href="#cb8-251" aria-hidden="true" tabindex="-1"></a>Consider how such tiny-effect alleles influence evolutionary change.</span>
<span id="cb8-252"><a href="#cb8-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-253"><a href="#cb8-253" aria-hidden="true" tabindex="-1"></a>A <span class="co">[</span><span class="ot">selective sweep</span><span class="co">](https://en.wikipedia.org/wiki/Selective_sweep)</span> is a process where a beneficial mutation becomes fixed in a population rapidly, removing the genetic variation around the selected allele.</span>
<span id="cb8-254"><a href="#cb8-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-255"><a href="#cb8-255" aria-hidden="true" tabindex="-1"></a>::: callout-tip</span>
<span id="cb8-256"><a href="#cb8-256" aria-hidden="true" tabindex="-1"></a><span class="fu"># Discussion points: selective sweeps and polygenicity</span></span>
<span id="cb8-257"><a href="#cb8-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-258"><a href="#cb8-258" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>If a trait is influenced by a few large-effect mutations, how likely is a classic selective sweep and why?</span>
<span id="cb8-259"><a href="#cb8-259" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>If a trait is influenced by many tiny-effect mutations, what evolutionary pattern replaces a sweep? (Think subtle allele frequency shifts.)</span>
<span id="cb8-260"><a href="#cb8-260" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>How can tight linkage to a deleterious allele modify or slow a sweep?</span>
<span id="cb8-261"><a href="#cb8-261" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>What genomic signatures (e.g., reduced diversity, extended haplotype homozygosity) would differ between oligogenic and highly polygenic adaptation?</span>
<span id="cb8-262"><a href="#cb8-262" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb8-263"><a href="#cb8-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-264"><a href="#cb8-264" aria-hidden="true" tabindex="-1"></a><span class="fu">## Another trait: glucose &amp; fructose use</span></span>
<span id="cb8-265"><a href="#cb8-265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-266"><a href="#cb8-266" aria-hidden="true" tabindex="-1"></a>In a vineyard environment, it is advantageous to metabolise glucose and fructose in rotting fruit. Glucose/fructose use is in the <span class="in">`gluc`</span> data frame.</span>
<span id="cb8-267"><a href="#cb8-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-268"><a href="#cb8-268" aria-hidden="true" tabindex="-1"></a>Again, we will look at the P-values (<span class="in">`Wald_P`</span>) and how much each SNP is predicted to affect the trait (<span class="in">`Effect_size`</span>).</span>
<span id="cb8-269"><a href="#cb8-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-272"><a href="#cb8-272" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-273"><a href="#cb8-273" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb8-274"><a href="#cb8-274" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb8-275"><a href="#cb8-275" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb8-276"><a href="#cb8-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-277"><a href="#cb8-277" aria-hidden="true" tabindex="-1"></a><span class="co">#set plotting parameters</span></span>
<span id="cb8-278"><a href="#cb8-278" aria-hidden="true" tabindex="-1"></a><span class="fu">layout</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>), <span class="dv">1</span>, <span class="dv">3</span>, <span class="at">byrow =</span> <span class="cn">TRUE</span>),<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">5579133</span>,<span class="dv">4539804</span>,<span class="dv">2452883</span>))</span>
<span id="cb8-279"><a href="#cb8-279" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">bty=</span><span class="st">"l"</span>,<span class="at">cex=</span><span class="fl">0.8</span>,<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">0</span>)<span class="sc">+</span><span class="dv">2</span>)</span>
<span id="cb8-280"><a href="#cb8-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-281"><a href="#cb8-281" aria-hidden="true" tabindex="-1"></a><span class="co">#loop through chromosomes 1,2,3, plotting each one in turn</span></span>
<span id="cb8-282"><a href="#cb8-282" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>){</span>
<span id="cb8-283"><a href="#cb8-283" aria-hidden="true" tabindex="-1"></a>      temp <span class="ot">=</span> <span class="fu">subset</span>(gluc, Chr <span class="sc">==</span> j)</span>
<span id="cb8-284"><a href="#cb8-284" aria-hidden="true" tabindex="-1"></a>      <span class="fu">plot</span>(temp<span class="sc">$</span>BP, <span class="sc">-</span><span class="fu">log10</span>(temp<span class="sc">$</span>Wald_P),</span>
<span id="cb8-285"><a href="#cb8-285" aria-hidden="true" tabindex="-1"></a>           <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">"Chr"</span>, j, <span class="st">"Gluc"</span>),<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">10</span>),</span>
<span id="cb8-286"><a href="#cb8-286" aria-hidden="true" tabindex="-1"></a>           <span class="at">xlab=</span><span class="st">"position"</span>,<span class="at">ylab=</span><span class="st">"-log10(P)"</span></span>
<span id="cb8-287"><a href="#cb8-287" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb8-288"><a href="#cb8-288" aria-hidden="true" tabindex="-1"></a>      <span class="fu">abline</span>(<span class="at">h=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">1e-5</span>),<span class="at">col=</span><span class="dv">2</span>,<span class="at">lty=</span><span class="dv">2</span>)</span>
<span id="cb8-289"><a href="#cb8-289" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-290"><a href="#cb8-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-291"><a href="#cb8-291" aria-hidden="true" tabindex="-1"></a><span class="sc">:::</span> callout<span class="sc">-</span>tip</span>
<span id="cb8-292"><a href="#cb8-292" aria-hidden="true" tabindex="-1"></a><span class="co"># Discussion points: second trait comparison</span></span>
<span id="cb8-293"><a href="#cb8-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-294"><a href="#cb8-294" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> Do the strongest glucose<span class="sc">/</span>fructose association peaks overlap with salt peaks? What could shared loci <span class="fu">imply</span> (pleiotropy vs linkage)?</span>
<span id="cb8-295"><a href="#cb8-295" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> Are there trait<span class="sc">-</span>specific peaks? What might that say about specialised metabolic pathways?</span>
<span id="cb8-296"><a href="#cb8-296" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> How could environmental <span class="fu">context</span> (salinity vs sugar availability) shift which alleles are favoured?</span>
<span id="cb8-297"><a href="#cb8-297" aria-hidden="true" tabindex="-1"></a><span class="sc">:::</span></span>
<span id="cb8-298"><a href="#cb8-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-299"><a href="#cb8-299" aria-hidden="true" tabindex="-1"></a>Again, you will see a small number of very low P<span class="sc">-</span>values at the top of the y<span class="sc">-</span><span class="fu">axis</span> (strong support) and many higher P<span class="sc">-</span>values at the <span class="fu">bottom</span> (weak or no support; tiny or no effect sizes).</span>
<span id="cb8-300"><a href="#cb8-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-301"><a href="#cb8-301" aria-hidden="true" tabindex="-1"></a>Now, make a histogram of the effect sizes<span class="sc">:</span></span>
<span id="cb8-302"><a href="#cb8-302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-305"><a href="#cb8-305" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-306"><a href="#cb8-306" aria-hidden="true" tabindex="-1"></a><span class="at">#| echo: true</span></span>
<span id="cb8-307"><a href="#cb8-307" aria-hidden="true" tabindex="-1"></a><span class="at">#| warning: false</span></span>
<span id="cb8-308"><a href="#cb8-308" aria-hidden="true" tabindex="-1"></a><span class="at">#| eval: false</span></span>
<span id="cb8-309"><a href="#cb8-309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-310"><a href="#cb8-310" aria-hidden="true" tabindex="-1"></a><span class="at">hist(gluc$Effect_size)</span></span>
<span id="cb8-311"><a href="#cb8-311" aria-hidden="true" tabindex="-1"></a><span class="st">```</span></span>
<span id="cb8-312"><a href="#cb8-312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-313"><a href="#cb8-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-314"><a href="#cb8-314" aria-hidden="true" tabindex="-1"></a><span class="do">## Looking at two traits</span></span>
<span id="cb8-315"><a href="#cb8-315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-316"><a href="#cb8-316" aria-hidden="true" tabindex="-1"></a>Species cannot adapt to a single <span class="fu">condition</span> (e.g., heat or salt) <span class="cf">in</span> isolation because environments are complex and variable. As a thought experiment, we will consider this using the fission yeast data.</span>
<span id="cb8-317"><a href="#cb8-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-318"><a href="#cb8-318" aria-hidden="true" tabindex="-1"></a>We will compare two Manhattan plots by overlaying them<span class="sc">:</span> <span class="fu">salt</span> (black) and glucose<span class="sc">/</span>fructose <span class="fu">metabolism</span> (red). The output PDF will appear <span class="cf">in</span> your working directory.</span>
<span id="cb8-319"><a href="#cb8-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-322"><a href="#cb8-322" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-323"><a href="#cb8-323" aria-hidden="true" tabindex="-1"></a><span class="at">#| echo: true</span></span>
<span id="cb8-324"><a href="#cb8-324" aria-hidden="true" tabindex="-1"></a><span class="at">#| warning: false</span></span>
<span id="cb8-325"><a href="#cb8-325" aria-hidden="true" tabindex="-1"></a><span class="at">#| eval: false</span></span>
<span id="cb8-326"><a href="#cb8-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-327"><a href="#cb8-327" aria-hidden="true" tabindex="-1"></a><span class="at">#open the pdf writer</span></span>
<span id="cb8-328"><a href="#cb8-328" aria-hidden="true" tabindex="-1"></a><span class="at">pdf("GWAS-plot3.pdf",width=60)</span></span>
<span id="cb8-329"><a href="#cb8-329" aria-hidden="true" tabindex="-1"></a><span class="at">#set up the plot layout</span></span>
<span id="cb8-330"><a href="#cb8-330" aria-hidden="true" tabindex="-1"></a><span class="at">layout(matrix(c(1:3), 1, 3, byrow = TRUE),widths = c(5579133,4539804,2452883))</span></span>
<span id="cb8-331"><a href="#cb8-331" aria-hidden="true" tabindex="-1"></a><span class="at">par(bty="l",cex=0.8,mar=c(2,2,2,0)+2)</span></span>
<span id="cb8-332"><a href="#cb8-332" aria-hidden="true" tabindex="-1"></a><span class="at">#loop through chromosomes</span></span>
<span id="cb8-333"><a href="#cb8-333" aria-hidden="true" tabindex="-1"></a><span class="at">for (j in 1:3){</span></span>
<span id="cb8-334"><a href="#cb8-334" aria-hidden="true" tabindex="-1"></a><span class="at">      #plot the salt GWAS points (black dots)</span></span>
<span id="cb8-335"><a href="#cb8-335" aria-hidden="true" tabindex="-1"></a><span class="at">      temp = subset(salt, Chr == j)</span></span>
<span id="cb8-336"><a href="#cb8-336" aria-hidden="true" tabindex="-1"></a><span class="at">      plot(temp$BP, -log10(temp$Wald_P),main = "",ylim=c(1,10),xlab="position",ylab="-log10(P)")</span></span>
<span id="cb8-337"><a href="#cb8-337" aria-hidden="true" tabindex="-1"></a><span class="at">      </span></span>
<span id="cb8-338"><a href="#cb8-338" aria-hidden="true" tabindex="-1"></a><span class="at">      #add the glucose/fructose GWAS points (red crosses)</span></span>
<span id="cb8-339"><a href="#cb8-339" aria-hidden="true" tabindex="-1"></a><span class="at">      temp2 = subset(gluc, Chr == j)</span></span>
<span id="cb8-340"><a href="#cb8-340" aria-hidden="true" tabindex="-1"></a><span class="at">      points(temp2$BP, -log10(temp2$Wald_P),main = "gluc",ylim=c(1,10),xlab="position",ylab="-log10(P)",col=2,pch=3)</span></span>
<span id="cb8-341"><a href="#cb8-341" aria-hidden="true" tabindex="-1"></a><span class="at">      </span></span>
<span id="cb8-342"><a href="#cb8-342" aria-hidden="true" tabindex="-1"></a><span class="at">      #add the significance threshold</span></span>
<span id="cb8-343"><a href="#cb8-343" aria-hidden="true" tabindex="-1"></a><span class="at">      abline(h=-log10(1e-5),col=2,lty=2)</span></span>
<span id="cb8-344"><a href="#cb8-344" aria-hidden="true" tabindex="-1"></a><span class="at">}</span></span>
<span id="cb8-345"><a href="#cb8-345" aria-hidden="true" tabindex="-1"></a><span class="at">#close the pdf writer</span></span>
<span id="cb8-346"><a href="#cb8-346" aria-hidden="true" tabindex="-1"></a><span class="at">dev.off()</span></span>
<span id="cb8-347"><a href="#cb8-347" aria-hidden="true" tabindex="-1"></a><span class="st">```</span></span>
<span id="cb8-348"><a href="#cb8-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-349"><a href="#cb8-349" aria-hidden="true" tabindex="-1"></a>Now find the pdf called <span class="st">`</span><span class="at">GWAS-plot3.pdf</span><span class="st">`</span> and take a look.</span>
<span id="cb8-350"><a href="#cb8-350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-351"><a href="#cb8-351" aria-hidden="true" tabindex="-1"></a><span class="do">### Might mutations for two traits conflict?</span></span>
<span id="cb8-352"><a href="#cb8-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-353"><a href="#cb8-353" aria-hidden="true" tabindex="-1"></a>Look at the significant peak on chromosome <span class="dv">3</span> (<span class="sc">~</span><span class="dv">1</span>,<span class="dv">500</span>,<span class="dv">000</span> bp). This likely has a relatively large effect size. There is also an important variant <span class="cf">for</span> glucose<span class="sc">/</span>fructose use nearby on chromosome <span class="fl">3.</span></span>
<span id="cb8-354"><a href="#cb8-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-355"><a href="#cb8-355" aria-hidden="true" tabindex="-1"></a>Consider what might happen <span class="cf">if</span> alleles are linked and the environment is complex—salty yet fructose<span class="sc">-</span><span class="fu">rich</span> (e.g., a coastal vineyard).</span>
<span id="cb8-356"><a href="#cb8-356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-357"><a href="#cb8-357" aria-hidden="true" tabindex="-1"></a>What might happen <span class="cf">if</span> helpful alleles <span class="cf">for</span> both traits are on the same haplotype?</span>
<span id="cb8-358"><a href="#cb8-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-359"><a href="#cb8-359" aria-hidden="true" tabindex="-1"></a><span class="sc">!</span>[](images<span class="sc">/</span>pombe2.jpg){width<span class="ot">=</span><span class="dv">100</span>%}</span>
<span id="cb8-360"><a href="#cb8-360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-361"><a href="#cb8-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-362"><a href="#cb8-362" aria-hidden="true" tabindex="-1"></a><span class="sc">:::</span>callout<span class="sc">-</span>note</span>
<span id="cb8-363"><a href="#cb8-363" aria-hidden="true" tabindex="-1"></a><span class="co"># Haplotypes</span></span>
<span id="cb8-364"><a href="#cb8-364" aria-hidden="true" tabindex="-1"></a>Physically, a haplotype is a stretch of DNA from one chromosome. In diploids, each individual carries two homologous chromosomes. In haploid <span class="sc">*</span>S. pombe<span class="sc">*</span>, different haplotypes are simply different chromosome versions present across the population. Through meiotic recombination haplotypes mix, but SNPs very close together rarely separate, so they tend to travel together.</span>
<span id="cb8-365"><a href="#cb8-365" aria-hidden="true" tabindex="-1"></a><span class="sc">:::</span></span>
<span id="cb8-366"><a href="#cb8-366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-367"><a href="#cb8-367" aria-hidden="true" tabindex="-1"></a><span class="do">### Haplotype problems</span></span>
<span id="cb8-368"><a href="#cb8-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-369"><a href="#cb8-369" aria-hidden="true" tabindex="-1"></a>Consider these situations<span class="sc">:</span></span>
<span id="cb8-370"><a href="#cb8-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-371"><a href="#cb8-371" aria-hidden="true" tabindex="-1"></a><span class="er">*</span> What might happen <span class="cf">if</span> helpful alleles are on the same haplotype?</span>
<span id="cb8-372"><a href="#cb8-372" aria-hidden="true" tabindex="-1"></a><span class="sc">*</span> What might happen <span class="cf">if</span> helpful alleles are on <span class="fu">different</span> (opposing) haplotypes?</span>
<span id="cb8-373"><a href="#cb8-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-374"><a href="#cb8-374" aria-hidden="true" tabindex="-1"></a>The figures below will help to explain.</span>
<span id="cb8-375"><a href="#cb8-375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-376"><a href="#cb8-376" aria-hidden="true" tabindex="-1"></a><span class="sc">:::</span> callout<span class="sc">-</span>tip</span>
<span id="cb8-377"><a href="#cb8-377" aria-hidden="true" tabindex="-1"></a><span class="co"># Discussion points: haplotypes and linkage</span></span>
<span id="cb8-378"><a href="#cb8-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-379"><a href="#cb8-379" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> How does tight linkage between beneficial alleles influence the speed of joint adaptation?</span>
<span id="cb8-380"><a href="#cb8-380" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> What evolutionary constraint arises when beneficial alleles are on different haplotypes <span class="cf">in</span> low recombination regions?</span>
<span id="cb8-381"><a href="#cb8-381" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> How can recombination rate modulate the resolution of this “conflict” between opposing haplotypes?</span>
<span id="cb8-382"><a href="#cb8-382" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span> Which <span class="fu">scenario</span> (single vs opposing haplotypes) better facilitates rapid multi<span class="sc">-</span>trait optimisation?</span>
<span id="cb8-383"><a href="#cb8-383" aria-hidden="true" tabindex="-1"></a><span class="sc">:::</span></span>
<span id="cb8-384"><a href="#cb8-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-385"><a href="#cb8-385" aria-hidden="true" tabindex="-1"></a><span class="sc">!</span>[](images<span class="sc">/</span>pombe2.jpg){width<span class="ot">=</span><span class="dv">100</span>%}</span>
<span id="cb8-386"><a href="#cb8-386" aria-hidden="true" tabindex="-1"></a><span class="sc">**</span>Figure <span class="fl">2.</span> Beneficial alleles on one haplotype.<span class="sc">**</span> Would there be any conflict here?</span>
<span id="cb8-387"><a href="#cb8-387" aria-hidden="true" tabindex="-1"></a>\</span>
<span id="cb8-388"><a href="#cb8-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-389"><a href="#cb8-389" aria-hidden="true" tabindex="-1"></a><span class="sc">!</span>[](images<span class="sc">/</span>pombe3.jpg){width<span class="ot">=</span><span class="dv">100</span>%}</span>
<span id="cb8-390"><a href="#cb8-390" aria-hidden="true" tabindex="-1"></a><span class="sc">**</span>Figure <span class="fl">3.</span> Beneficial alleles on two different haplotypes.<span class="sc">**</span> What might happen here? What is the potential problem?</span>
<span id="cb8-391"><a href="#cb8-391" aria-hidden="true" tabindex="-1"></a>\</span>
<span id="cb8-392"><a href="#cb8-392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-393"><a href="#cb8-393" aria-hidden="true" tabindex="-1"></a>{{<span class="sc">&lt;</span> pagebreak <span class="sc">&gt;</span>}}</span>
<span id="cb8-394"><a href="#cb8-394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-395"><a href="#cb8-395" aria-hidden="true" tabindex="-1"></a><span class="co"># Summary: what we have learned</span></span>
<span id="cb8-396"><a href="#cb8-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-397"><a href="#cb8-397" aria-hidden="true" tabindex="-1"></a>This workshop used data from fission yeast, but the principles apply broadly to quantitative traits. Key points<span class="sc">:</span></span>
<span id="cb8-398"><a href="#cb8-398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-399"><a href="#cb8-399" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Most SNPs do not strongly associate with traits.</span>
<span id="cb8-400"><a href="#cb8-400" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Most SNPs have very small effects on traits, but some do have detectable effects.</span>
<span id="cb8-401"><a href="#cb8-401" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> SNPs do not affect all traits at once. Some SNPs affect one trait; some affect another.</span>
<span id="cb8-402"><a href="#cb8-402" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Genetic variants, like SNPs, travel through time and space <span class="cf">in</span> haplotypes.</span>
<span id="cb8-403"><a href="#cb8-403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-404"><a href="#cb8-404" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-405"><a href="#cb8-405" aria-hidden="true" tabindex="-1"></a>{{<span class="sc">&lt;</span> pagebreak <span class="sc">&gt;</span>}}</span>
<span id="cb8-406"><a href="#cb8-406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-407"><a href="#cb8-407" aria-hidden="true" tabindex="-1"></a><span class="co"># After the workshop: exam-style questions</span></span>
<span id="cb8-408"><a href="#cb8-408" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-409"><a href="#cb8-409" aria-hidden="true" tabindex="-1"></a>Exam questions often ask you to apply workshop principles to other species or scenarios. Reflecting on these helps deepen understanding.</span>
<span id="cb8-410"><a href="#cb8-410" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-411"><a href="#cb8-411" aria-hidden="true" tabindex="-1"></a><span class="sc">:::</span>callout<span class="sc">-</span>note</span>
<span id="cb8-412"><a href="#cb8-412" aria-hidden="true" tabindex="-1"></a><span class="co"># Principles of quantitative genetics</span></span>
<span id="cb8-413"><a href="#cb8-413" aria-hidden="true" tabindex="-1"></a><span class="sc">*</span> Most SNPs have no effect, or very small effects, on traits.</span>
<span id="cb8-414"><a href="#cb8-414" aria-hidden="true" tabindex="-1"></a><span class="sc">*</span> SNPs that do affect traits are generally responsible <span class="cf">for</span> less than <span class="dv">5</span>% of the trait variation.</span>
<span id="cb8-415"><a href="#cb8-415" aria-hidden="true" tabindex="-1"></a><span class="sc">*</span> Effect sizes are also small <span class="cf">for</span> other genetic variants, such as transposon insertions, duplications<span class="sc">/</span>deletions and so on.</span>
<span id="cb8-416"><a href="#cb8-416" aria-hidden="true" tabindex="-1"></a><span class="sc">*</span> Genetic variants can affect one trait, or several.</span>
<span id="cb8-417"><a href="#cb8-417" aria-hidden="true" tabindex="-1"></a><span class="sc">*</span> In the short term, selection occurs on <span class="sc">*</span>haplotypes<span class="sc">*</span>, but over time recombination will separate linked variants.</span>
<span id="cb8-418"><a href="#cb8-418" aria-hidden="true" tabindex="-1"></a><span class="sc">:::</span></span>
<span id="cb8-419"><a href="#cb8-419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-420"><a href="#cb8-420" aria-hidden="true" tabindex="-1"></a><span class="do">## Question 1 (10 points)</span></span>
<span id="cb8-421"><a href="#cb8-421" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-422"><a href="#cb8-422" aria-hidden="true" tabindex="-1"></a>This question checks your understanding of quantitative traits and why most variants have small effects.</span>
<span id="cb8-423"><a href="#cb8-423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-424"><a href="#cb8-424" aria-hidden="true" tabindex="-1"></a>a<span class="er">)</span> <span class="sc">**</span>Define<span class="sc">**</span> the difference between a <span class="fu">simple</span> (single<span class="sc">-</span>gene) trait and a quantitative trait. Give one example of each from this <span class="fu">workshop.</span> (<span class="dv">2</span> points)</span>
<span id="cb8-425"><a href="#cb8-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-426"><a href="#cb8-426" aria-hidden="true" tabindex="-1"></a>b<span class="er">)</span> The histogram of SNP effect sizes <span class="cf">for</span> the salt trait is heavily skewed towards zero. <span class="sc">**</span>Describe<span class="sc">**</span> what this pattern tells us about the distribution of SNP effects <span class="cf">in</span> the <span class="fu">population.</span> (<span class="dv">3</span> points)</span>
<span id="cb8-427"><a href="#cb8-427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-428"><a href="#cb8-428" aria-hidden="true" tabindex="-1"></a>c<span class="er">)</span> Using the [Neutral Theory of Molecular Evolution](https<span class="sc">:</span><span class="er">//</span>en.wikipedia.org<span class="sc">/</span>wiki<span class="sc">/</span>Neutral_theory_of_molecular_evolution), <span class="sc">**</span>explain<span class="sc">**</span> why most SNPs are expected to have little or no effect on <span class="fu">fitness.</span> (<span class="dv">3</span> points)</span>
<span id="cb8-429"><a href="#cb8-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-430"><a href="#cb8-430" aria-hidden="true" tabindex="-1"></a>d<span class="er">)</span> <span class="sc">**</span>Suggest<span class="sc">**</span> one practical way to increase the power of a GWAS to detect very small<span class="sc">-</span>effect variants and explain why it <span class="fu">helps.</span> (<span class="dv">2</span> points)</span>
<span id="cb8-431"><a href="#cb8-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-432"><a href="#cb8-432" aria-hidden="true" tabindex="-1"></a><span class="do">## Question 2 (10 points)</span></span>
<span id="cb8-433"><a href="#cb8-433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-434"><a href="#cb8-434" aria-hidden="true" tabindex="-1"></a>You notice that the Manhattan plots <span class="cf">for</span> salt tolerance and glucose<span class="sc">/</span>fructose utilisation share a peak on chromosome <span class="dv">3</span>, but also have trait<span class="sc">-</span>specific peaks.</span>
<span id="cb8-435"><a href="#cb8-435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-436"><a href="#cb8-436" aria-hidden="true" tabindex="-1"></a>a<span class="er">)</span> <span class="sc">**</span>Describe<span class="sc">**</span> what an overlapping peak <span class="cf">for</span> both traits suggests about the underlying genetics of those <span class="fu">traits.</span> (<span class="dv">3</span> points)</span>
<span id="cb8-437"><a href="#cb8-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-438"><a href="#cb8-438" aria-hidden="true" tabindex="-1"></a>b<span class="er">)</span> <span class="sc">**</span>State<span class="sc">**</span> one additional analysis you could carry out to test whether the same causal variant influences both <span class="fu">traits.</span> (<span class="dv">2</span> points)</span>
<span id="cb8-439"><a href="#cb8-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-440"><a href="#cb8-440" aria-hidden="true" tabindex="-1"></a>c<span class="er">)</span> <span class="sc">**</span>Explain<span class="sc">**</span> how recombination could help resolve a situation where beneficial alleles <span class="cf">for</span> two traits are on different <span class="fu">haplotypes.</span> (<span class="dv">3</span> points)</span>
<span id="cb8-441"><a href="#cb8-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-442"><a href="#cb8-442" aria-hidden="true" tabindex="-1"></a>d<span class="er">)</span> <span class="sc">**</span>Identify<span class="sc">**</span> one environmental measurement you would collect alongside the genomic data to interpret potential trade<span class="sc">-</span>offs between the two traits, and justify your <span class="fu">choice.</span> (<span class="dv">2</span> points)</span>
<span id="cb8-443"><a href="#cb8-443" aria-hidden="true" tabindex="-1"></a></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>